// telnyx/resources/networks/networks.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as DefaultGatewayAPI from "./default-gateway.mjs";
import { DefaultGateway, } from "./default-gateway.mjs";
import { DefaultFlatPagination } from "../../core/pagination.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * Network operations
 */
export class Networks extends APIResource {
    constructor() {
        super(...arguments);
        this.defaultGateway = new DefaultGatewayAPI.DefaultGateway(this._client);
    }
    /**
     * Create a new Network.
     *
     * @example
     * ```ts
     * const network = await client.networks.create({
     *   name: 'test network',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/networks', { body, ...options });
    }
    /**
     * Retrieve a Network.
     *
     * @example
     * ```ts
     * const network = await client.networks.retrieve(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/networks/${id}`, options);
    }
    /**
     * Update a Network.
     *
     * @example
     * ```ts
     * const network = await client.networks.update(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *   { name: 'test network' },
     * );
     * ```
     */
    update(networkID, body, options) {
        return this._client.patch(path `/networks/${networkID}`, { body, ...options });
    }
    /**
     * List all Networks.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const networkListResponse of client.networks.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/networks', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete a Network.
     *
     * @example
     * ```ts
     * const network = await client.networks.delete(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/networks/${id}`, options);
    }
    /**
     * List all Interfaces for a Network.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const networkListInterfacesResponse of client.networks.listInterfaces(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * )) {
     *   // ...
     * }
     * ```
     */
    listInterfaces(id, query = {}, options) {
        return this._client.getAPIList(path `/networks/${id}/network_interfaces`, (DefaultFlatPagination), { query, ...options });
    }
}
Networks.DefaultGateway = DefaultGateway;
