// telnyx/resources/networks/networks.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Networks = void 0;
const tslib_1 = require("../../internal/tslib.js");
const resource_1 = require("../../core/resource.js");
const DefaultGatewayAPI = tslib_1.__importStar(require("./default-gateway.js"));
const default_gateway_1 = require("./default-gateway.js");
const pagination_1 = require("../../core/pagination.js");
const path_1 = require("../../internal/utils/path.js");
/**
 * Network operations
 */
class Networks extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.defaultGateway = new DefaultGatewayAPI.DefaultGateway(this._client);
    }
    /**
     * Create a new Network.
     *
     * @example
     * ```ts
     * const network = await client.networks.create({
     *   name: 'test network',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/networks', { body, ...options });
    }
    /**
     * Retrieve a Network.
     *
     * @example
     * ```ts
     * const network = await client.networks.retrieve(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/networks/${id}`, options);
    }
    /**
     * Update a Network.
     *
     * @example
     * ```ts
     * const network = await client.networks.update(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *   { name: 'test network' },
     * );
     * ```
     */
    update(networkID, body, options) {
        return this._client.patch((0, path_1.path) `/networks/${networkID}`, { body, ...options });
    }
    /**
     * List all Networks.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const networkListResponse of client.networks.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/networks', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete a Network.
     *
     * @example
     * ```ts
     * const network = await client.networks.delete(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/networks/${id}`, options);
    }
    /**
     * List all Interfaces for a Network.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const networkListInterfacesResponse of client.networks.listInterfaces(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * )) {
     *   // ...
     * }
     * ```
     */
    listInterfaces(id, query = {}, options) {
        return this._client.getAPIList((0, path_1.path) `/networks/${id}/network_interfaces`, (pagination_1.DefaultFlatPagination), { query, ...options });
    }
}
exports.Networks = Networks;
Networks.DefaultGateway = default_gateway_1.DefaultGateway;
