// telnyx/resources/bundle-pricing/user-bundles.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import { DefaultFlatPagination } from "../../core/pagination.mjs";
import { buildHeaders } from "../../internal/headers.mjs";
import { path } from "../../internal/utils/path.mjs";
export class UserBundles extends APIResource {
    /**
     * Creates multiple user bundles for the user.
     *
     * @example
     * ```ts
     * const userBundle =
     *   await client.bundlePricing.userBundles.create();
     * ```
     */
    create(params, options) {
        const { authorization_bearer, ...body } = params;
        return this._client.post('/bundle_pricing/user_bundles/bulk', {
            body,
            ...options,
            headers: buildHeaders([
                { ...(authorization_bearer != null ? { authorization_bearer: authorization_bearer } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Retrieves a user bundle by its ID.
     *
     * @example
     * ```ts
     * const userBundle =
     *   await client.bundlePricing.userBundles.retrieve(
     *     'ca1d2263-d1f1-43ac-ba53-248e7a4bb26a',
     *   );
     * ```
     */
    retrieve(userBundleID, params = {}, options) {
        const { authorization_bearer } = params ?? {};
        return this._client.get(path `/bundle_pricing/user_bundles/${userBundleID}`, {
            ...options,
            headers: buildHeaders([
                { ...(authorization_bearer != null ? { authorization_bearer: authorization_bearer } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Get a paginated list of user bundles.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const userBundle of client.bundlePricing.userBundles.list()) {
     *   // ...
     * }
     * ```
     */
    list(params = {}, options) {
        const { authorization_bearer, ...query } = params ?? {};
        return this._client.getAPIList('/bundle_pricing/user_bundles', (DefaultFlatPagination), {
            query,
            ...options,
            headers: buildHeaders([
                { ...(authorization_bearer != null ? { authorization_bearer: authorization_bearer } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Deactivates a user bundle by its ID.
     *
     * @example
     * ```ts
     * const response =
     *   await client.bundlePricing.userBundles.deactivate(
     *     'ca1d2263-d1f1-43ac-ba53-248e7a4bb26a',
     *   );
     * ```
     */
    deactivate(userBundleID, params = {}, options) {
        const { authorization_bearer } = params ?? {};
        return this._client.delete(path `/bundle_pricing/user_bundles/${userBundleID}`, {
            ...options,
            headers: buildHeaders([
                { ...(authorization_bearer != null ? { authorization_bearer: authorization_bearer } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Retrieves the resources of a user bundle by its ID.
     *
     * @example
     * ```ts
     * const response =
     *   await client.bundlePricing.userBundles.listResources(
     *     'ca1d2263-d1f1-43ac-ba53-248e7a4bb26a',
     *   );
     * ```
     */
    listResources(userBundleID, params = {}, options) {
        const { authorization_bearer } = params ?? {};
        return this._client.get(path `/bundle_pricing/user_bundles/${userBundleID}/resources`, {
            ...options,
            headers: buildHeaders([
                { ...(authorization_bearer != null ? { authorization_bearer: authorization_bearer } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Returns all user bundles that aren't in use.
     *
     * @example
     * ```ts
     * const response =
     *   await client.bundlePricing.userBundles.listUnused();
     * ```
     */
    listUnused(params = {}, options) {
        const { authorization_bearer, ...query } = params ?? {};
        return this._client.get('/bundle_pricing/user_bundles/unused', {
            query,
            ...options,
            headers: buildHeaders([
                { ...(authorization_bearer != null ? { authorization_bearer: authorization_bearer } : undefined) },
                options?.headers,
            ]),
        });
    }
}
