// telnyx/resources/bundle-pricing/user-bundles.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserBundles = void 0;
const resource_1 = require("../../core/resource.js");
const pagination_1 = require("../../core/pagination.js");
const headers_1 = require("../../internal/headers.js");
const path_1 = require("../../internal/utils/path.js");
class UserBundles extends resource_1.APIResource {
    /**
     * Creates multiple user bundles for the user.
     *
     * @example
     * ```ts
     * const userBundle =
     *   await client.bundlePricing.userBundles.create();
     * ```
     */
    create(params, options) {
        const { authorization_bearer, ...body } = params;
        return this._client.post('/bundle_pricing/user_bundles/bulk', {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([
                { ...(authorization_bearer != null ? { authorization_bearer: authorization_bearer } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Retrieves a user bundle by its ID.
     *
     * @example
     * ```ts
     * const userBundle =
     *   await client.bundlePricing.userBundles.retrieve(
     *     'ca1d2263-d1f1-43ac-ba53-248e7a4bb26a',
     *   );
     * ```
     */
    retrieve(userBundleID, params = {}, options) {
        const { authorization_bearer } = params ?? {};
        return this._client.get((0, path_1.path) `/bundle_pricing/user_bundles/${userBundleID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([
                { ...(authorization_bearer != null ? { authorization_bearer: authorization_bearer } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Get a paginated list of user bundles.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const userBundle of client.bundlePricing.userBundles.list()) {
     *   // ...
     * }
     * ```
     */
    list(params = {}, options) {
        const { authorization_bearer, ...query } = params ?? {};
        return this._client.getAPIList('/bundle_pricing/user_bundles', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
            headers: (0, headers_1.buildHeaders)([
                { ...(authorization_bearer != null ? { authorization_bearer: authorization_bearer } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Deactivates a user bundle by its ID.
     *
     * @example
     * ```ts
     * const response =
     *   await client.bundlePricing.userBundles.deactivate(
     *     'ca1d2263-d1f1-43ac-ba53-248e7a4bb26a',
     *   );
     * ```
     */
    deactivate(userBundleID, params = {}, options) {
        const { authorization_bearer } = params ?? {};
        return this._client.delete((0, path_1.path) `/bundle_pricing/user_bundles/${userBundleID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([
                { ...(authorization_bearer != null ? { authorization_bearer: authorization_bearer } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Retrieves the resources of a user bundle by its ID.
     *
     * @example
     * ```ts
     * const response =
     *   await client.bundlePricing.userBundles.listResources(
     *     'ca1d2263-d1f1-43ac-ba53-248e7a4bb26a',
     *   );
     * ```
     */
    listResources(userBundleID, params = {}, options) {
        const { authorization_bearer } = params ?? {};
        return this._client.get((0, path_1.path) `/bundle_pricing/user_bundles/${userBundleID}/resources`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([
                { ...(authorization_bearer != null ? { authorization_bearer: authorization_bearer } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Returns all user bundles that aren't in use.
     *
     * @example
     * ```ts
     * const response =
     *   await client.bundlePricing.userBundles.listUnused();
     * ```
     */
    listUnused(params = {}, options) {
        const { authorization_bearer, ...query } = params ?? {};
        return this._client.get('/bundle_pricing/user_bundles/unused', {
            query,
            ...options,
            headers: (0, headers_1.buildHeaders)([
                { ...(authorization_bearer != null ? { authorization_bearer: authorization_bearer } : undefined) },
                options?.headers,
            ]),
        });
    }
}
exports.UserBundles = UserBundles;
