// telnyx/resources/messaging-tollfree/verification/requests.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Requests = void 0;
const resource_1 = require("../../../core/resource.js");
const pagination_1 = require("../../../core/pagination.js");
const headers_1 = require("../../../internal/headers.js");
const path_1 = require("../../../internal/utils/path.js");
/**
 * Manage your tollfree verification requests
 */
class Requests extends resource_1.APIResource {
    /**
     * Submit a new tollfree verification request
     *
     * @example
     * ```ts
     * const verificationRequestEgress =
     *   await client.messagingTollfree.verification.requests.create({
     *     additionalInformation: 'additionalInformation',
     *     businessAddr1: '600 Congress Avenue',
     *     businessCity: 'Austin',
     *     businessContactEmail: 'email@example.com',
     *     businessContactFirstName: 'John',
     *     businessContactLastName: 'Doe',
     *     businessContactPhone: '+18005550100',
     *     businessName: 'Telnyx LLC',
     *     businessState: 'Texas',
     *     businessZip: '78701',
     *     corporateWebsite: 'http://example.com',
     *     messageVolume: '100,000',
     *     optInWorkflow:
     *       "User signs into the Telnyx portal, enters a number and is prompted to select whether they want to use 2FA verification for security purposes. If they've opted in a confirmation message is sent out to the handset",
     *     optInWorkflowImageURLs: [
     *       { url: 'https://telnyx.com/sign-up' },
     *       { url: 'https://telnyx.com/company/data-privacy' },
     *     ],
     *     phoneNumbers: [{ phoneNumber: '+18773554398' }, { phoneNumber: '+18773554399' }],
     *     productionMessageContent: 'Your Telnyx OTP is XXXX',
     *     useCase: '2FA',
     *     useCaseSummary:
     *       'This is a use case where Telnyx sends out 2FA codes to portal users to verify their identity in order to sign into the portal',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/messaging_tollfree/verification/requests', { body, ...options });
    }
    /**
     * Get a single verification request by its ID.
     *
     * @example
     * ```ts
     * const verificationRequestStatus =
     *   await client.messagingTollfree.verification.requests.retrieve(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/messaging_tollfree/verification/requests/${id}`, options);
    }
    /**
     * Update an existing tollfree verification request. This is particularly useful
     * when there are pending customer actions to be taken.
     *
     * @example
     * ```ts
     * const verificationRequestEgress =
     *   await client.messagingTollfree.verification.requests.update(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     {
     *       additionalInformation: 'additionalInformation',
     *       businessAddr1: '600 Congress Avenue',
     *       businessCity: 'Austin',
     *       businessContactEmail: 'email@example.com',
     *       businessContactFirstName: 'John',
     *       businessContactLastName: 'Doe',
     *       businessContactPhone: '+18005550100',
     *       businessName: 'Telnyx LLC',
     *       businessState: 'Texas',
     *       businessZip: '78701',
     *       corporateWebsite: 'http://example.com',
     *       messageVolume: '100,000',
     *       optInWorkflow:
     *         "User signs into the Telnyx portal, enters a number and is prompted to select whether they want to use 2FA verification for security purposes. If they've opted in a confirmation message is sent out to the handset",
     *       optInWorkflowImageURLs: [
     *         { url: 'https://telnyx.com/sign-up' },
     *         { url: 'https://telnyx.com/company/data-privacy' },
     *       ],
     *       phoneNumbers: [{ phoneNumber: '+18773554398' }, { phoneNumber: '+18773554399' }],
     *       productionMessageContent: 'Your Telnyx OTP is XXXX',
     *       useCase: '2FA',
     *       useCaseSummary:
     *         'This is a use case where Telnyx sends out 2FA codes to portal users to verify their identity in order to sign into the portal',
     *     },
     *   );
     * ```
     */
    update(id, body, options) {
        return this._client.patch((0, path_1.path) `/messaging_tollfree/verification/requests/${id}`, { body, ...options });
    }
    /**
     * Get a list of previously-submitted tollfree verification requests
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const verificationRequestStatus of client.messagingTollfree.verification.requests.list(
     *   { page: 1, page_size: 1 },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(query, options) {
        return this._client.getAPIList('/messaging_tollfree/verification/requests', (pagination_1.DefaultPaginationForMessagingTollfree), { query, ...options });
    }
    /**
     * Delete a verification request
     *
     * A request may only be deleted when when the request is in the "rejected" state.
     *
     * - `HTTP 200`: request successfully deleted
     * - `HTTP 400`: request exists but can't be deleted (i.e. not rejected)
     * - `HTTP 404`: request unknown or already deleted
     *
     * @example
     * ```ts
     * await client.messagingTollfree.verification.requests.delete(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/messaging_tollfree/verification/requests/${id}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Get the history of status changes for a verification request.
     *
     * Returns a paginated list of historical status changes including the reason for
     * each change and when it occurred.
     *
     * @example
     * ```ts
     * const response =
     *   await client.messagingTollfree.verification.requests.retrieveStatusHistory(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { 'page[number]': 1, 'page[size]': 1 },
     *   );
     * ```
     */
    retrieveStatusHistory(id, query, options) {
        return this._client.get((0, path_1.path) `/messaging_tollfree/verification/requests/${id}/status_history`, {
            query,
            ...options,
        });
    }
}
exports.Requests = Requests;
