// telnyx/resources/uac-connections/uac-connections.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.UacConnections = void 0;
const tslib_1 = require("../../internal/tslib.js");
const resource_1 = require("../../core/resource.js");
const ActionsAPI = tslib_1.__importStar(require("./actions.js"));
const actions_1 = require("./actions.js");
const pagination_1 = require("../../core/pagination.js");
const path_1 = require("../../internal/utils/path.js");
/**
 * UAC connection operations
 */
class UacConnections extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.actions = new ActionsAPI.Actions(this._client);
    }
    /**
     * Creates a UAC connection. A UAC (User Agent Client) Connection registers Telnyx
     * to your PBX — the opposite of a standard SIP trunk, where the PBX registers to
     * Telnyx. Use UAC when your PBX doesn’t support outbound SIP registration or you
     * need Telnyx to maintain the registration.
     *
     * @example
     * ```ts
     * const uacConnection = await client.uacConnections.create({
     *   connection_name: 'my name',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/uac_connections', { body, ...options });
    }
    /**
     * Retrieves the details of an existing UAC connection.
     *
     * @example
     * ```ts
     * const uacConnection = await client.uacConnections.retrieve(
     *   'id',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/uac_connections/${id}`, options);
    }
    /**
     * Updates settings of an existing UAC connection.
     *
     * @example
     * ```ts
     * const uacConnection = await client.uacConnections.update(
     *   'id',
     * );
     * ```
     */
    update(id, body, options) {
        return this._client.patch((0, path_1.path) `/uac_connections/${id}`, { body, ...options });
    }
    /**
     * Returns a list of your UAC connections. A UAC (User Agent Client) Connection
     * registers Telnyx to your PBX — the opposite of a standard SIP trunk, where the
     * PBX registers to Telnyx. Use UAC when your PBX doesn’t support outbound SIP
     * registration or you need Telnyx to maintain the registration.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const uacConnection of client.uacConnections.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/uac_connections', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Deletes an existing UAC connection.
     *
     * @example
     * ```ts
     * const uacConnection = await client.uacConnections.delete(
     *   'id',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/uac_connections/${id}`, options);
    }
}
exports.UacConnections = UacConnections;
UacConnections.Actions = actions_1.Actions;
