// telnyx/resources/uac-connections/uac-connections.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as ActionsAPI from "./actions.mjs";
import { Actions } from "./actions.mjs";
import { DefaultFlatPagination } from "../../core/pagination.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * UAC connection operations
 */
export class UacConnections extends APIResource {
    constructor() {
        super(...arguments);
        this.actions = new ActionsAPI.Actions(this._client);
    }
    /**
     * Creates a UAC connection. A UAC (User Agent Client) Connection registers Telnyx
     * to your PBX — the opposite of a standard SIP trunk, where the PBX registers to
     * Telnyx. Use UAC when your PBX doesn’t support outbound SIP registration or you
     * need Telnyx to maintain the registration.
     *
     * @example
     * ```ts
     * const uacConnection = await client.uacConnections.create({
     *   connection_name: 'my name',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/uac_connections', { body, ...options });
    }
    /**
     * Retrieves the details of an existing UAC connection.
     *
     * @example
     * ```ts
     * const uacConnection = await client.uacConnections.retrieve(
     *   'id',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/uac_connections/${id}`, options);
    }
    /**
     * Updates settings of an existing UAC connection.
     *
     * @example
     * ```ts
     * const uacConnection = await client.uacConnections.update(
     *   'id',
     * );
     * ```
     */
    update(id, body, options) {
        return this._client.patch(path `/uac_connections/${id}`, { body, ...options });
    }
    /**
     * Returns a list of your UAC connections. A UAC (User Agent Client) Connection
     * registers Telnyx to your PBX — the opposite of a standard SIP trunk, where the
     * PBX registers to Telnyx. Use UAC when your PBX doesn’t support outbound SIP
     * registration or you need Telnyx to maintain the registration.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const uacConnection of client.uacConnections.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/uac_connections', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Deletes an existing UAC connection.
     *
     * @example
     * ```ts
     * const uacConnection = await client.uacConnections.delete(
     *   'id',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/uac_connections/${id}`, options);
    }
}
UacConnections.Actions = Actions;
