// telnyx/resources/addresses/addresses.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as ActionsAPI from "./actions.mjs";
import { Actions, } from "./actions.mjs";
import { DefaultFlatPagination } from "../../core/pagination.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * Operations to work with Address records. Address records are emergency-validated addresses meant to be associated with phone numbers. They are validated for emergency usage purposes at creation time, although you may validate them separately with a custom workflow using the ValidateAddress operation separately. Address records are not usable for physical orders, such as for Telnyx SIM cards, please use UserAddress for that. It is not possible to entirely skip emergency service validation for Address records; if an emergency provider for a phone number rejects the address then it cannot be used on a phone number. To prevent records from getting out of sync, Address records are immutable and cannot be altered once created. If you realize you need to alter an address, a new record must be created with the differing address.
 */
export class Addresses extends APIResource {
    constructor() {
        super(...arguments);
        this.actions = new ActionsAPI.Actions(this._client);
    }
    /**
     * Creates an address.
     *
     * @example
     * ```ts
     * const address = await client.addresses.create({
     *   business_name: "Toy-O'Kon",
     *   country_code: 'US',
     *   first_name: 'Alfred',
     *   last_name: 'Foster',
     *   locality: 'Austin',
     *   street_address: '600 Congress Avenue',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/addresses', { body, ...options });
    }
    /**
     * Retrieves the details of an existing address.
     *
     * @example
     * ```ts
     * const address = await client.addresses.retrieve('id');
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/addresses/${id}`, options);
    }
    /**
     * Returns a list of your addresses.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const address of client.addresses.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/addresses', (DefaultFlatPagination), { query, ...options });
    }
    /**
     * Deletes an existing address.
     *
     * @example
     * ```ts
     * const address = await client.addresses.delete('id');
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/addresses/${id}`, options);
    }
}
Addresses.Actions = Actions;
