// telnyx/resources/credential-connections/credential-connections.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.CredentialConnections = void 0;
const tslib_1 = require("../../internal/tslib.js");
const resource_1 = require("../../core/resource.js");
const ActionsAPI = tslib_1.__importStar(require("./actions.js"));
const actions_1 = require("./actions.js");
const pagination_1 = require("../../core/pagination.js");
const path_1 = require("../../internal/utils/path.js");
/**
 * Credential connection operations
 */
class CredentialConnections extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.actions = new ActionsAPI.Actions(this._client);
    }
    /**
     * Creates a credential connection.
     *
     * @example
     * ```ts
     * const credentialConnection =
     *   await client.credentialConnections.create({
     *     connection_name: 'my name',
     *     password: 'my123secure456password789',
     *     user_name: 'myusername123',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/credential_connections', { body, ...options });
    }
    /**
     * Retrieves the details of an existing credential connection.
     *
     * @example
     * ```ts
     * const credentialConnection =
     *   await client.credentialConnections.retrieve('id');
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/credential_connections/${id}`, options);
    }
    /**
     * Updates settings of an existing credential connection.
     *
     * @example
     * ```ts
     * const credentialConnection =
     *   await client.credentialConnections.update('id');
     * ```
     */
    update(id, body, options) {
        return this._client.patch((0, path_1.path) `/credential_connections/${id}`, { body, ...options });
    }
    /**
     * Returns a list of your credential connections.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const credentialConnection of client.credentialConnections.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/credential_connections', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Deletes an existing credential connection.
     *
     * @example
     * ```ts
     * const credentialConnection =
     *   await client.credentialConnections.delete('id');
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/credential_connections/${id}`, options);
    }
}
exports.CredentialConnections = CredentialConnections;
CredentialConnections.Actions = actions_1.Actions;
