// telnyx/resources/credential-connections/credential-connections.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as ActionsAPI from "./actions.mjs";
import { Actions } from "./actions.mjs";
import { DefaultFlatPagination } from "../../core/pagination.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * Credential connection operations
 */
export class CredentialConnections extends APIResource {
    constructor() {
        super(...arguments);
        this.actions = new ActionsAPI.Actions(this._client);
    }
    /**
     * Creates a credential connection.
     *
     * @example
     * ```ts
     * const credentialConnection =
     *   await client.credentialConnections.create({
     *     connection_name: 'my name',
     *     password: 'my123secure456password789',
     *     user_name: 'myusername123',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/credential_connections', { body, ...options });
    }
    /**
     * Retrieves the details of an existing credential connection.
     *
     * @example
     * ```ts
     * const credentialConnection =
     *   await client.credentialConnections.retrieve('id');
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/credential_connections/${id}`, options);
    }
    /**
     * Updates settings of an existing credential connection.
     *
     * @example
     * ```ts
     * const credentialConnection =
     *   await client.credentialConnections.update('id');
     * ```
     */
    update(id, body, options) {
        return this._client.patch(path `/credential_connections/${id}`, { body, ...options });
    }
    /**
     * Returns a list of your credential connections.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const credentialConnection of client.credentialConnections.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/credential_connections', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Deletes an existing credential connection.
     *
     * @example
     * ```ts
     * const credentialConnection =
     *   await client.credentialConnections.delete('id');
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/credential_connections/${id}`, options);
    }
}
CredentialConnections.Actions = Actions;
