// telnyx/resources/messages/messages.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as RcsAPI from "./rcs.mjs";
import { Rcs, } from "./rcs.mjs";
import { path } from "../../internal/utils/path.mjs";
export class Messages extends APIResource {
    constructor() {
        super(...arguments);
        this.rcs = new RcsAPI.Rcs(this._client);
    }
    /**
     * Note: This API endpoint can only retrieve messages that are no older than 10
     * days since their creation. If you require messages older than this, please
     * generate an
     * [MDR report.](https://developers.telnyx.com/api-reference/mdr-usage-reports/create-mdr-usage-report)
     *
     * @example
     * ```ts
     * const message = await client.messages.retrieve(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/messages/${id}`, options);
    }
    /**
     * Cancel a scheduled message that has not yet been sent. Only messages with
     * `status=scheduled` and `send_at` more than a minute from now can be cancelled.
     *
     * @example
     * ```ts
     * const response = await client.messages.cancelScheduled(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    cancelScheduled(id, options) {
        return this._client.delete(path `/messages/${id}`, options);
    }
    /**
     * Retrieve all messages in a group MMS conversation by the group message ID.
     *
     * @example
     * ```ts
     * const response =
     *   await client.messages.retrieveGroupMessages(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    retrieveGroupMessages(messageID, options) {
        return this._client.get(path `/messages/group/${messageID}`, options);
    }
    /**
     * Schedule a message with a Phone Number, Alphanumeric Sender ID, Short Code or
     * Number Pool.
     *
     * This endpoint allows you to schedule a message with any messaging resource.
     * Current messaging resources include: long-code, short-code, number-pool, and
     * alphanumeric-sender-id.
     *
     * @example
     * ```ts
     * const response = await client.messages.schedule({
     *   to: '+18445550001',
     * });
     * ```
     */
    schedule(body, options) {
        return this._client.post('/messages/schedule', { body, ...options });
    }
    /**
     * Send a message with a Phone Number, Alphanumeric Sender ID, Short Code or Number
     * Pool.
     *
     * This endpoint allows you to send a message with any messaging resource. Current
     * messaging resources include: long-code, short-code, number-pool, and
     * alphanumeric-sender-id.
     *
     * @example
     * ```ts
     * const response = await client.messages.send({
     *   to: '+18445550001',
     * });
     * ```
     */
    send(body, options) {
        return this._client.post('/messages', { body, ...options });
    }
    /**
     * Send a group MMS message
     *
     * @example
     * ```ts
     * const response = await client.messages.sendGroupMms({
     *   from: '+13125551234',
     *   to: ['+18655551234', '+14155551234'],
     * });
     * ```
     */
    sendGroupMms(body, options) {
        return this._client.post('/messages/group_mms', { body, ...options });
    }
    /**
     * Send a long code message
     *
     * @example
     * ```ts
     * const response = await client.messages.sendLongCode({
     *   from: '+18445550001',
     *   to: '+13125550002',
     * });
     * ```
     */
    sendLongCode(body, options) {
        return this._client.post('/messages/long_code', { body, ...options });
    }
    /**
     * Send a message using number pool
     *
     * @example
     * ```ts
     * const response = await client.messages.sendNumberPool({
     *   messaging_profile_id:
     *     'abc85f64-5717-4562-b3fc-2c9600000000',
     *   to: '+13125550002',
     * });
     * ```
     */
    sendNumberPool(body, options) {
        return this._client.post('/messages/number_pool', { body, ...options });
    }
    /**
     * Send a short code message
     *
     * @example
     * ```ts
     * const response = await client.messages.sendShortCode({
     *   from: '+18445550001',
     *   to: '+18445550001',
     * });
     * ```
     */
    sendShortCode(body, options) {
        return this._client.post('/messages/short_code', { body, ...options });
    }
    /**
     * Send a Whatsapp message
     *
     * @example
     * ```ts
     * const response = await client.messages.sendWhatsapp({
     *   from: '+13125551234',
     *   to: '+13125551234',
     *   whatsapp_message: {},
     * });
     * ```
     */
    sendWhatsapp(body, options) {
        return this._client.post('/messages/whatsapp', { body, ...options });
    }
    /**
     * Send an SMS message using an alphanumeric sender ID. This is SMS only.
     *
     * @example
     * ```ts
     * const response =
     *   await client.messages.sendWithAlphanumericSender({
     *     from: 'MyCompany',
     *     messaging_profile_id:
     *       '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     text: 'text',
     *     to: '+E.164',
     *   });
     * ```
     */
    sendWithAlphanumericSender(body, options) {
        return this._client.post('/messages/alphanumeric_sender_id', { body, ...options });
    }
}
Messages.Rcs = Rcs;
