// telnyx/resources/enterprises/reputation/numbers.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Numbers = void 0;
const resource_1 = require("../../../core/resource.js");
const pagination_1 = require("../../../core/pagination.js");
const headers_1 = require("../../../internal/headers.js");
const path_1 = require("../../../internal/utils/path.js");
/**
 * Associate phone numbers with an enterprise for reputation monitoring and retrieve reputation scores
 */
class Numbers extends resource_1.APIResource {
    /**
     * Get detailed reputation data for a specific phone number associated with an
     * enterprise.
     *
     * **Query Parameters:**
     *
     * - `fresh` (default: `false`): When `true`, fetches fresh reputation data (incurs
     *   API cost). When `false`, returns cached data. If no cached data exists, fresh
     *   data is automatically fetched.
     *
     * **Returns:**
     *
     * - `spam_risk`: Overall spam risk level (`low`, `medium`, `high`)
     * - `spam_category`: Spam category classification
     * - `maturity_score`: Maturity metric (0–100)
     * - `connection_score`: Connection quality metric (0–100)
     * - `engagement_score`: Engagement metric (0–100)
     * - `sentiment_score`: Sentiment metric (0–100)
     * - `last_refreshed_at`: Timestamp of last data refresh
     *
     * @example
     * ```ts
     * const number =
     *   await client.enterprises.reputation.numbers.retrieve(
     *     '+16035551234',
     *     {
     *       enterprise_id: '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *     },
     *   );
     * ```
     */
    retrieve(phoneNumber, params, options) {
        const { enterprise_id, ...query } = params;
        return this._client.get((0, path_1.path) `/enterprises/${enterprise_id}/reputation/numbers/${phoneNumber}`, {
            query,
            ...options,
        });
    }
    /**
     * List all phone numbers associated with an enterprise for Number Reputation
     * monitoring.
     *
     * Returns phone numbers with their cached reputation data (if available). Supports
     * pagination and filtering by phone number.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const reputationPhoneNumberWithReputationData of client.enterprises.reputation.numbers.list(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * )) {
     *   // ...
     * }
     * ```
     */
    list(enterpriseID, query = {}, options) {
        return this._client.getAPIList((0, path_1.path) `/enterprises/${enterpriseID}/reputation/numbers`, (pagination_1.DefaultFlatPagination), { query, ...options });
    }
    /**
     * Associate one or more phone numbers with an enterprise for Number Reputation
     * monitoring.
     *
     * **Validations:**
     *
     * - Phone numbers must be in E.164 format (e.g., `+16035551234`)
     * - Phone numbers must be in-service and belong to your account (verified via
     *   Warehouse)
     * - Phone numbers must be US local numbers
     * - Phone numbers cannot already be associated with any enterprise
     *
     * **Note:** This operation is atomic — if any number fails validation, the entire
     * request fails.
     *
     * **Maximum:** 100 phone numbers per request.
     *
     * @example
     * ```ts
     * const response =
     *   await client.enterprises.reputation.numbers.associate(
     *     '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *     { phone_numbers: ['+16035551234'] },
     *   );
     * ```
     */
    associate(enterpriseID, body, options) {
        return this._client.post((0, path_1.path) `/enterprises/${enterpriseID}/reputation/numbers`, { body, ...options });
    }
    /**
     * Remove a phone number from Number Reputation monitoring for an enterprise.
     *
     * The number will no longer be tracked and reputation data will no longer be
     * refreshed.
     *
     * @example
     * ```ts
     * await client.enterprises.reputation.numbers.disassociate(
     *   '+16035551234',
     *   { enterprise_id: '6a09cdc3-8948-47f0-aa62-74ac943d6c58' },
     * );
     * ```
     */
    disassociate(phoneNumber, params, options) {
        const { enterprise_id } = params;
        return this._client.delete((0, path_1.path) `/enterprises/${enterprise_id}/reputation/numbers/${phoneNumber}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.Numbers = Numbers;
