// telnyx/resources/enterprises/reputation/reputation.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../core/resource.mjs";
import * as NumbersAPI from "./numbers.mjs";
import { Numbers, } from "./numbers.mjs";
import { buildHeaders } from "../../../internal/headers.mjs";
import { path } from "../../../internal/utils/path.mjs";
/**
 * Manage Number Reputation enrollment and check frequency settings for an enterprise
 */
export class Reputation extends APIResource {
    constructor() {
        super(...arguments);
        this.numbers = new NumbersAPI.Numbers(this._client);
    }
    /**
     * Retrieve the current Number Reputation settings for an enterprise.
     *
     * Returns the enrollment status (`pending`, `approved`, `rejected`, `deleted`),
     * check frequency, and any rejection reasons.
     *
     * Returns `404` if reputation has not been enabled for this enterprise.
     *
     * @example
     * ```ts
     * const reputation =
     *   await client.enterprises.reputation.retrieve(
     *     '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *   );
     * ```
     */
    retrieve(enterpriseID, options) {
        return this._client.get(path `/enterprises/${enterpriseID}/reputation`, options);
    }
    /**
     * Disable Number Reputation for an enterprise.
     *
     * This will:
     *
     * - Delete the reputation settings record
     * - Log the deletion for audit purposes
     * - Stop all future automated reputation checks
     *
     * **Note:** Can only be performed on `approved` reputation settings.
     *
     * @example
     * ```ts
     * await client.enterprises.reputation.disable(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    disable(enterpriseID, options) {
        return this._client.delete(path `/enterprises/${enterpriseID}/reputation`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Enable Number Reputation service for an enterprise.
     *
     * **Requirements:**
     *
     * - Signed LOA (Letter of Authorization) document ID
     * - Reputation check frequency (defaults to `business_daily`)
     * - Number Reputation Terms of Service must be accepted
     *
     * **Flow:**
     *
     * 1. Registers the enterprise for reputation monitoring
     * 2. Creates reputation settings with `pending` status
     * 3. Awaits admin approval before monitoring begins
     *
     * **Resubmission After Rejection:** If a previously rejected record exists, this
     * endpoint will delete it and create a new `pending` record.
     *
     * **Available Frequencies:**
     *
     * - `business_daily` — Monday–Friday
     * - `daily` — Every day
     * - `weekly` — Once per week
     * - `biweekly` — Once every two weeks
     * - `monthly` — Once per month
     * - `never` — Manual refresh only
     *
     * @example
     * ```ts
     * const response = await client.enterprises.reputation.enable(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *   { loa_document_id: 'doc_01HXYZ1234ABCDEF' },
     * );
     * ```
     */
    enable(enterpriseID, body, options) {
        return this._client.post(path `/enterprises/${enterpriseID}/reputation`, { body, ...options });
    }
    /**
     * Update how often reputation data is automatically refreshed.
     *
     * **Note:** The enterprise must have `approved` reputation settings. Updating
     * frequency on `pending` or `rejected` settings will return an error.
     *
     * **Available Frequencies:**
     *
     * - `business_daily` — Monday–Friday
     * - `daily` — Every day including weekends
     * - `weekly` — Once per week
     * - `biweekly` — Once every two weeks
     * - `monthly` — Once per month
     * - `never` — Manual refresh only (no automatic checks)
     *
     * @example
     * ```ts
     * const response =
     *   await client.enterprises.reputation.updateFrequency(
     *     '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *     { check_frequency: 'business_daily' },
     *   );
     * ```
     */
    updateFrequency(enterpriseID, body, options) {
        return this._client.patch(path `/enterprises/${enterpriseID}/reputation/frequency`, { body, ...options });
    }
}
Reputation.Numbers = Numbers;
