// telnyx/resources/ai/assistants/assistants.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../core/resource.mjs";
import * as CanaryDeploysAPI from "./canary-deploys.mjs";
import { CanaryDeploys, } from "./canary-deploys.mjs";
import * as ScheduledEventsAPI from "./scheduled-events.mjs";
import { ScheduledEvents, } from "./scheduled-events.mjs";
import * as TagsAPI from "./tags.mjs";
import { Tags, } from "./tags.mjs";
import * as ToolsAPI from "./tools.mjs";
import { Tools, } from "./tools.mjs";
import * as VersionsAPI from "./versions.mjs";
import { Versions, } from "./versions.mjs";
import * as TestsAPI from "./tests/tests.mjs";
import { Tests, } from "./tests/tests.mjs";
import { path } from "../../../internal/utils/path.mjs";
/**
 * Configure AI assistant specifications
 */
export class Assistants extends APIResource {
    constructor() {
        super(...arguments);
        this.tests = new TestsAPI.Tests(this._client);
        this.canaryDeploys = new CanaryDeploysAPI.CanaryDeploys(this._client);
        this.scheduledEvents = new ScheduledEventsAPI.ScheduledEvents(this._client);
        this.tools = new ToolsAPI.Tools(this._client);
        this.versions = new VersionsAPI.Versions(this._client);
        this.tags = new TagsAPI.Tags(this._client);
    }
    /**
     * Create a new AI Assistant.
     *
     * @example
     * ```ts
     * const inferenceEmbedding =
     *   await client.ai.assistants.create({
     *     instructions: 'instructions',
     *     name: 'name',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/ai/assistants', { body, ...options });
    }
    /**
     * Retrieve an AI Assistant configuration by `assistant_id`.
     *
     * @example
     * ```ts
     * const inferenceEmbedding =
     *   await client.ai.assistants.retrieve('assistant_id');
     * ```
     */
    retrieve(assistantID, query = {}, options) {
        return this._client.get(path `/ai/assistants/${assistantID}`, { query, ...options });
    }
    /**
     * Update an AI Assistant's attributes.
     *
     * @example
     * ```ts
     * const inferenceEmbedding =
     *   await client.ai.assistants.update('assistant_id');
     * ```
     */
    update(assistantID, body, options) {
        return this._client.post(path `/ai/assistants/${assistantID}`, { body, ...options });
    }
    /**
     * Retrieve a list of all AI Assistants configured by the user.
     *
     * @example
     * ```ts
     * const assistantsList = await client.ai.assistants.list();
     * ```
     */
    list(options) {
        return this._client.get('/ai/assistants', options);
    }
    /**
     * Delete an AI Assistant by `assistant_id`.
     *
     * @example
     * ```ts
     * const assistant = await client.ai.assistants.delete(
     *   'assistant_id',
     * );
     * ```
     */
    delete(assistantID, options) {
        return this._client.delete(path `/ai/assistants/${assistantID}`, options);
    }
    /**
     * This endpoint allows a client to send a chat message to a specific AI Assistant.
     * The assistant processes the message and returns a relevant reply based on the
     * current conversation context. Refer to the Conversation API to
     * [create a conversation](https://developers.telnyx.com/api-reference/conversations/create-a-conversation),
     * [filter existing conversations](https://developers.telnyx.com/api-reference/conversations/list-conversations),
     * [fetch messages for a conversation](https://developers.telnyx.com/api-reference/conversations/get-conversation-messages),
     * and
     * [manually add messages to a conversation](https://developers.telnyx.com/api-reference/conversations/create-message).
     *
     * @example
     * ```ts
     * const response = await client.ai.assistants.chat(
     *   'assistant_id',
     *   {
     *     content: 'Tell me a joke about cats',
     *     conversation_id: '42b20469-1215-4a9a-8964-c36f66b406f4',
     *   },
     * );
     * ```
     */
    chat(assistantID, body, options) {
        return this._client.post(path `/ai/assistants/${assistantID}/chat`, { body, ...options });
    }
    /**
     * Clone an existing assistant, excluding telephony and messaging settings.
     *
     * @example
     * ```ts
     * const inferenceEmbedding = await client.ai.assistants.clone(
     *   'assistant_id',
     * );
     * ```
     */
    clone(assistantID, options) {
        return this._client.post(path `/ai/assistants/${assistantID}/clone`, options);
    }
    /**
     * Get an assistant texml by `assistant_id`.
     *
     * @example
     * ```ts
     * const response = await client.ai.assistants.getTexml(
     *   'assistant_id',
     * );
     * ```
     */
    getTexml(assistantID, options) {
        return this._client.get(path `/ai/assistants/${assistantID}/texml`, options);
    }
    /**
     * Import assistants from external providers. Any assistant that has already been
     * imported will be overwritten with its latest version from the importing
     * provider.
     *
     * @example
     * ```ts
     * const assistantsList = await client.ai.assistants.imports({
     *   api_key_ref: 'api_key_ref',
     *   provider: 'elevenlabs',
     * });
     * ```
     */
    imports(body, options) {
        return this._client.post('/ai/assistants/import', { body, ...options });
    }
    /**
     * Send an SMS message for an assistant. This endpoint:
     *
     * 1. Validates the assistant exists and has messaging profile configured
     * 2. If should_create_conversation is true, creates a new conversation with
     *    metadata
     * 3. Sends the SMS message (If `text` is set, this will be sent. Otherwise, if
     *    this is the first message in the conversation and the assistant has a
     *    `greeting` configured, this will be sent. Otherwise the assistant will
     *    generate the text to send.)
     * 4. Updates conversation metadata if provided
     * 5. Returns the conversation ID
     *
     * @example
     * ```ts
     * const response = await client.ai.assistants.sendSMS(
     *   'assistant_id',
     *   { from: 'from', to: 'to' },
     * );
     * ```
     */
    sendSMS(assistantID, body, options) {
        return this._client.post(path `/ai/assistants/${assistantID}/chat/sms`, { body, ...options });
    }
}
Assistants.Tests = Tests;
Assistants.CanaryDeploys = CanaryDeploys;
Assistants.ScheduledEvents = ScheduledEvents;
Assistants.Tools = Tools;
Assistants.Versions = Versions;
Assistants.Tags = Tags;
