// telnyx/resources/ai/assistants/canary-deploys.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.CanaryDeploys = void 0;
const resource_1 = require("../../../core/resource.js");
const headers_1 = require("../../../internal/headers.js");
const path_1 = require("../../../internal/utils/path.js");
/**
 * Configure AI assistant specifications
 */
class CanaryDeploys extends resource_1.APIResource {
    /**
     * Endpoint to create a canary deploy configuration for an assistant.
     *
     * Creates a new canary deploy configuration with multiple version IDs and their
     * traffic percentages for A/B testing or gradual rollouts of assistant versions.
     *
     * @example
     * ```ts
     * const canaryDeployResponse =
     *   await client.ai.assistants.canaryDeploys.create(
     *     'assistant_id',
     *   );
     * ```
     */
    create(assistantID, body, options) {
        return this._client.post((0, path_1.path) `/ai/assistants/${assistantID}/canary-deploys`, { body, ...options });
    }
    /**
     * Endpoint to get a canary deploy configuration for an assistant.
     *
     * Retrieves the current canary deploy configuration with all version IDs and their
     * traffic percentages for the specified assistant.
     *
     * @example
     * ```ts
     * const canaryDeployResponse =
     *   await client.ai.assistants.canaryDeploys.retrieve(
     *     'assistant_id',
     *   );
     * ```
     */
    retrieve(assistantID, options) {
        return this._client.get((0, path_1.path) `/ai/assistants/${assistantID}/canary-deploys`, options);
    }
    /**
     * Endpoint to update a canary deploy configuration for an assistant.
     *
     * Updates the existing canary deploy configuration with new version IDs and
     * percentages. All old versions and percentages are replaces by new ones from this
     * request.
     *
     * @example
     * ```ts
     * const canaryDeployResponse =
     *   await client.ai.assistants.canaryDeploys.update(
     *     'assistant_id',
     *   );
     * ```
     */
    update(assistantID, body, options) {
        return this._client.put((0, path_1.path) `/ai/assistants/${assistantID}/canary-deploys`, { body, ...options });
    }
    /**
     * Endpoint to delete a canary deploy configuration for an assistant.
     *
     * Removes all canary deploy configurations for the specified assistant.
     *
     * @example
     * ```ts
     * await client.ai.assistants.canaryDeploys.delete(
     *   'assistant_id',
     * );
     * ```
     */
    delete(assistantID, options) {
        return this._client.delete((0, path_1.path) `/ai/assistants/${assistantID}/canary-deploys`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.CanaryDeploys = CanaryDeploys;
