// telnyx/resources/ai/assistants/scheduled-events.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduledEvents = void 0;
const resource_1 = require("../../../core/resource.js");
const pagination_1 = require("../../../core/pagination.js");
const headers_1 = require("../../../internal/headers.js");
const path_1 = require("../../../internal/utils/path.js");
/**
 * Configure AI assistant specifications
 */
class ScheduledEvents extends resource_1.APIResource {
    /**
     * Create a scheduled event for an assistant
     *
     * @example
     * ```ts
     * const scheduledEventResponse =
     *   await client.ai.assistants.scheduledEvents.create(
     *     'assistant_id',
     *     {
     *       scheduled_at_fixed_datetime:
     *         '2025-04-15T13:07:28.764Z',
     *       telnyx_agent_target: 'telnyx_agent_target',
     *       telnyx_conversation_channel: 'phone_call',
     *       telnyx_end_user_target: 'telnyx_end_user_target',
     *     },
     *   );
     * ```
     */
    create(assistantID, body, options) {
        return this._client.post((0, path_1.path) `/ai/assistants/${assistantID}/scheduled_events`, { body, ...options });
    }
    /**
     * Retrieve a scheduled event by event ID
     *
     * @example
     * ```ts
     * const scheduledEventResponse =
     *   await client.ai.assistants.scheduledEvents.retrieve(
     *     'event_id',
     *     { assistant_id: 'assistant_id' },
     *   );
     * ```
     */
    retrieve(eventID, params, options) {
        const { assistant_id } = params;
        return this._client.get((0, path_1.path) `/ai/assistants/${assistant_id}/scheduled_events/${eventID}`, options);
    }
    /**
     * Get scheduled events for an assistant with pagination and filtering
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const scheduledEventListResponse of client.ai.assistants.scheduledEvents.list(
     *   'assistant_id',
     * )) {
     *   // ...
     * }
     * ```
     */
    list(assistantID, query = {}, options) {
        return this._client.getAPIList((0, path_1.path) `/ai/assistants/${assistantID}/scheduled_events`, (pagination_1.DefaultFlatPagination), { query, ...options });
    }
    /**
     * If the event is pending, this will cancel the event. Otherwise, this will simply
     * remove the record of the event.
     *
     * @example
     * ```ts
     * await client.ai.assistants.scheduledEvents.delete(
     *   'event_id',
     *   { assistant_id: 'assistant_id' },
     * );
     * ```
     */
    delete(eventID, params, options) {
        const { assistant_id } = params;
        return this._client.delete((0, path_1.path) `/ai/assistants/${assistant_id}/scheduled_events/${eventID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.ScheduledEvents = ScheduledEvents;
