// telnyx/resources/ai/assistants/scheduled-events.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../core/resource.mjs";
import { DefaultFlatPagination, } from "../../../core/pagination.mjs";
import { buildHeaders } from "../../../internal/headers.mjs";
import { path } from "../../../internal/utils/path.mjs";
/**
 * Configure AI assistant specifications
 */
export class ScheduledEvents extends APIResource {
    /**
     * Create a scheduled event for an assistant
     *
     * @example
     * ```ts
     * const scheduledEventResponse =
     *   await client.ai.assistants.scheduledEvents.create(
     *     'assistant_id',
     *     {
     *       scheduled_at_fixed_datetime:
     *         '2025-04-15T13:07:28.764Z',
     *       telnyx_agent_target: 'telnyx_agent_target',
     *       telnyx_conversation_channel: 'phone_call',
     *       telnyx_end_user_target: 'telnyx_end_user_target',
     *     },
     *   );
     * ```
     */
    create(assistantID, body, options) {
        return this._client.post(path `/ai/assistants/${assistantID}/scheduled_events`, { body, ...options });
    }
    /**
     * Retrieve a scheduled event by event ID
     *
     * @example
     * ```ts
     * const scheduledEventResponse =
     *   await client.ai.assistants.scheduledEvents.retrieve(
     *     'event_id',
     *     { assistant_id: 'assistant_id' },
     *   );
     * ```
     */
    retrieve(eventID, params, options) {
        const { assistant_id } = params;
        return this._client.get(path `/ai/assistants/${assistant_id}/scheduled_events/${eventID}`, options);
    }
    /**
     * Get scheduled events for an assistant with pagination and filtering
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const scheduledEventListResponse of client.ai.assistants.scheduledEvents.list(
     *   'assistant_id',
     * )) {
     *   // ...
     * }
     * ```
     */
    list(assistantID, query = {}, options) {
        return this._client.getAPIList(path `/ai/assistants/${assistantID}/scheduled_events`, (DefaultFlatPagination), { query, ...options });
    }
    /**
     * If the event is pending, this will cancel the event. Otherwise, this will simply
     * remove the record of the event.
     *
     * @example
     * ```ts
     * await client.ai.assistants.scheduledEvents.delete(
     *   'event_id',
     *   { assistant_id: 'assistant_id' },
     * );
     * ```
     */
    delete(eventID, params, options) {
        const { assistant_id } = params;
        return this._client.delete(path `/ai/assistants/${assistant_id}/scheduled_events/${eventID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
