// telnyx/resources/ai/assistants/versions.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Versions = void 0;
const resource_1 = require("../../../core/resource.js");
const headers_1 = require("../../../internal/headers.js");
const path_1 = require("../../../internal/utils/path.js");
/**
 * Configure AI assistant specifications
 */
class Versions extends resource_1.APIResource {
    /**
     * Retrieves a specific version of an assistant by assistant_id and version_id
     *
     * @example
     * ```ts
     * const inferenceEmbedding =
     *   await client.ai.assistants.versions.retrieve(
     *     'version_id',
     *     { assistant_id: 'assistant_id' },
     *   );
     * ```
     */
    retrieve(versionID, params, options) {
        const { assistant_id, ...query } = params;
        return this._client.get((0, path_1.path) `/ai/assistants/${assistant_id}/versions/${versionID}`, {
            query,
            ...options,
        });
    }
    /**
     * Updates the configuration of a specific assistant version. Can not update main
     * version
     *
     * @example
     * ```ts
     * const inferenceEmbedding =
     *   await client.ai.assistants.versions.update('version_id', {
     *     assistant_id: 'assistant_id',
     *   });
     * ```
     */
    update(versionID, params, options) {
        const { assistant_id, ...body } = params;
        return this._client.post((0, path_1.path) `/ai/assistants/${assistant_id}/versions/${versionID}`, {
            body,
            ...options,
        });
    }
    /**
     * Retrieves all versions of a specific assistant with complete configuration and
     * metadata
     *
     * @example
     * ```ts
     * const assistantsList =
     *   await client.ai.assistants.versions.list('assistant_id');
     * ```
     */
    list(assistantID, options) {
        return this._client.get((0, path_1.path) `/ai/assistants/${assistantID}/versions`, options);
    }
    /**
     * Permanently removes a specific version of an assistant. Can not delete main
     * version
     *
     * @example
     * ```ts
     * await client.ai.assistants.versions.delete('version_id', {
     *   assistant_id: 'assistant_id',
     * });
     * ```
     */
    delete(versionID, params, options) {
        const { assistant_id } = params;
        return this._client.delete((0, path_1.path) `/ai/assistants/${assistant_id}/versions/${versionID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Promotes a specific version to be the main/current version of the assistant.
     * This will delete any existing canary deploy configuration and send all live
     * production traffic to this version.
     *
     * @example
     * ```ts
     * const inferenceEmbedding =
     *   await client.ai.assistants.versions.promote(
     *     'version_id',
     *     { assistant_id: 'assistant_id' },
     *   );
     * ```
     */
    promote(versionID, params, options) {
        const { assistant_id } = params;
        return this._client.post((0, path_1.path) `/ai/assistants/${assistant_id}/versions/${versionID}/promote`, options);
    }
}
exports.Versions = Versions;
