// telnyx/resources/messaging-hosted-number-orders/messaging-hosted-number-orders.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as ActionsAPI from "./actions.mjs";
import { Actions } from "./actions.mjs";
import { DefaultFlatPagination } from "../../core/pagination.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * Manage your messaging hosted numbers
 */
export class MessagingHostedNumberOrders extends APIResource {
    constructor() {
        super(...arguments);
        this.actions = new ActionsAPI.Actions(this._client);
    }
    /**
     * Create a messaging hosted number order
     *
     * @example
     * ```ts
     * const messagingHostedNumberOrder =
     *   await client.messagingHostedNumberOrders.create();
     * ```
     */
    create(body = {}, options) {
        return this._client.post('/messaging_hosted_number_orders', { body, ...options });
    }
    /**
     * Retrieve a messaging hosted number order
     *
     * @example
     * ```ts
     * const messagingHostedNumberOrder =
     *   await client.messagingHostedNumberOrders.retrieve('id');
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/messaging_hosted_number_orders/${id}`, options);
    }
    /**
     * List messaging hosted number orders
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const messagingHostedNumberOrder of client.messagingHostedNumberOrders.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/messaging_hosted_number_orders', (DefaultFlatPagination), { query, ...options });
    }
    /**
     * Delete a messaging hosted number order and all associated phone numbers.
     *
     * @example
     * ```ts
     * const messagingHostedNumberOrder =
     *   await client.messagingHostedNumberOrders.delete('id');
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/messaging_hosted_number_orders/${id}`, options);
    }
    /**
     * Check hosted messaging eligibility
     *
     * @example
     * ```ts
     * const response =
     *   await client.messagingHostedNumberOrders.checkEligibility(
     *     { phone_numbers: ['string'] },
     *   );
     * ```
     */
    checkEligibility(body, options) {
        return this._client.post('/messaging_hosted_number_orders/eligibility_numbers_check', {
            body,
            ...options,
        });
    }
    /**
     * Create verification codes to validate numbers of the hosted order. The
     * verification codes will be sent to the numbers of the hosted order.
     *
     * @example
     * ```ts
     * const response =
     *   await client.messagingHostedNumberOrders.createVerificationCodes(
     *     'id',
     *     {
     *       phone_numbers: ['string'],
     *       verification_method: 'sms',
     *     },
     *   );
     * ```
     */
    createVerificationCodes(id, body, options) {
        return this._client.post(path `/messaging_hosted_number_orders/${id}/verification_codes`, {
            body,
            ...options,
        });
    }
    /**
     * Validate the verification codes sent to the numbers of the hosted order. The
     * verification codes must be created in the verification codes endpoint.
     *
     * @example
     * ```ts
     * const response =
     *   await client.messagingHostedNumberOrders.validateCodes(
     *     'id',
     *     {
     *       verification_codes: [
     *         { code: 'code', phone_number: 'phone_number' },
     *       ],
     *     },
     *   );
     * ```
     */
    validateCodes(id, body, options) {
        return this._client.post(path `/messaging_hosted_number_orders/${id}/validation_codes`, {
            body,
            ...options,
        });
    }
}
MessagingHostedNumberOrders.Actions = Actions;
