// telnyx/resources/ai/missions/runs/plan.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../../core/resource.mjs";
import { path } from "../../../../internal/utils/path.mjs";
export class Plan extends APIResource {
    /**
     * Create the initial plan for a run
     *
     * @example
     * ```ts
     * const plan = await client.ai.missions.runs.plan.create(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   {
     *     mission_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     steps: [
     *       {
     *         description: 'description',
     *         sequence: 0,
     *         step_id: 'step_id',
     *       },
     *     ],
     *   },
     * );
     * ```
     */
    create(runID, params, options) {
        const { mission_id, ...body } = params;
        return this._client.post(path `/ai/missions/${mission_id}/runs/${runID}/plan`, { body, ...options });
    }
    /**
     * Get the plan (all steps) for a run
     *
     * @example
     * ```ts
     * const plan = await client.ai.missions.runs.plan.retrieve(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   { mission_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e' },
     * );
     * ```
     */
    retrieve(runID, params, options) {
        const { mission_id } = params;
        return this._client.get(path `/ai/missions/${mission_id}/runs/${runID}/plan`, options);
    }
    /**
     * Add one or more steps to an existing plan
     *
     * @example
     * ```ts
     * const response =
     *   await client.ai.missions.runs.plan.addStepsToPlan(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     {
     *       mission_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *       steps: [
     *         {
     *           description: 'description',
     *           sequence: 0,
     *           step_id: 'step_id',
     *         },
     *       ],
     *     },
     *   );
     * ```
     */
    addStepsToPlan(runID, params, options) {
        const { mission_id, ...body } = params;
        return this._client.post(path `/ai/missions/${mission_id}/runs/${runID}/plan/steps`, { body, ...options });
    }
    /**
     * Get details of a specific plan step
     *
     * @example
     * ```ts
     * const response =
     *   await client.ai.missions.runs.plan.getStepDetails(
     *     'step_id',
     *     {
     *       mission_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *       run_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     },
     *   );
     * ```
     */
    getStepDetails(stepID, params, options) {
        const { mission_id, run_id } = params;
        return this._client.get(path `/ai/missions/${mission_id}/runs/${run_id}/plan/steps/${stepID}`, options);
    }
    /**
     * Update the status of a plan step
     *
     * @example
     * ```ts
     * const response =
     *   await client.ai.missions.runs.plan.updateStep('step_id', {
     *     mission_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     run_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   });
     * ```
     */
    updateStep(stepID, params, options) {
        const { mission_id, run_id, ...body } = params;
        return this._client.patch(path `/ai/missions/${mission_id}/runs/${run_id}/plan/steps/${stepID}`, {
            body,
            ...options,
        });
    }
}
