// telnyx/resources/ai/missions/runs/runs.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../../core/resource.mjs";
import * as EventsAPI from "./events.mjs";
import { Events, } from "./events.mjs";
import * as PlanAPI from "./plan.mjs";
import { Plan, } from "./plan.mjs";
import * as TelnyxAgentsAPI from "./telnyx-agents.mjs";
import { TelnyxAgents, } from "./telnyx-agents.mjs";
import { DefaultFlatPagination, } from "../../../../core/pagination.mjs";
import { path } from "../../../../internal/utils/path.mjs";
export class Runs extends APIResource {
    constructor() {
        super(...arguments);
        this.events = new EventsAPI.Events(this._client);
        this.plan = new PlanAPI.Plan(this._client);
        this.telnyxAgents = new TelnyxAgentsAPI.TelnyxAgents(this._client);
    }
    /**
     * Start a new run for a mission
     *
     * @example
     * ```ts
     * const run = await client.ai.missions.runs.create(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    create(missionID, body = {}, options) {
        return this._client.post(path `/ai/missions/${missionID}/runs`, { body, ...options });
    }
    /**
     * Get details of a specific run
     *
     * @example
     * ```ts
     * const run = await client.ai.missions.runs.retrieve(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   { mission_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e' },
     * );
     * ```
     */
    retrieve(runID, params, options) {
        const { mission_id } = params;
        return this._client.get(path `/ai/missions/${mission_id}/runs/${runID}`, options);
    }
    /**
     * Update run status and/or result
     *
     * @example
     * ```ts
     * const run = await client.ai.missions.runs.update(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   { mission_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e' },
     * );
     * ```
     */
    update(runID, params, options) {
        const { mission_id, ...body } = params;
        return this._client.patch(path `/ai/missions/${mission_id}/runs/${runID}`, { body, ...options });
    }
    /**
     * List all runs for a specific mission
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const missionRunData of client.ai.missions.runs.list(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * )) {
     *   // ...
     * }
     * ```
     */
    list(missionID, query = {}, options) {
        return this._client.getAPIList(path `/ai/missions/${missionID}/runs`, (DefaultFlatPagination), { query, ...options });
    }
    /**
     * Cancel a running or paused run
     *
     * @example
     * ```ts
     * const response = await client.ai.missions.runs.cancelRun(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   { mission_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e' },
     * );
     * ```
     */
    cancelRun(runID, params, options) {
        const { mission_id } = params;
        return this._client.post(path `/ai/missions/${mission_id}/runs/${runID}/cancel`, options);
    }
    /**
     * List recent runs across all missions
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const missionRunData of client.ai.missions.runs.listRuns()) {
     *   // ...
     * }
     * ```
     */
    listRuns(query = {}, options) {
        return this._client.getAPIList('/ai/missions/runs', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Pause a running run
     *
     * @example
     * ```ts
     * const response = await client.ai.missions.runs.pauseRun(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   { mission_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e' },
     * );
     * ```
     */
    pauseRun(runID, params, options) {
        const { mission_id } = params;
        return this._client.post(path `/ai/missions/${mission_id}/runs/${runID}/pause`, options);
    }
    /**
     * Resume a paused run
     *
     * @example
     * ```ts
     * const response = await client.ai.missions.runs.resumeRun(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   { mission_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e' },
     * );
     * ```
     */
    resumeRun(runID, params, options) {
        const { mission_id } = params;
        return this._client.post(path `/ai/missions/${mission_id}/runs/${runID}/resume`, options);
    }
}
Runs.Events = Events;
Runs.Plan = Plan;
Runs.TelnyxAgents = TelnyxAgents;
