// telnyx/resources/ai/missions/runs/runs.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Runs = void 0;
const tslib_1 = require("../../../../internal/tslib.js");
const resource_1 = require("../../../../core/resource.js");
const EventsAPI = tslib_1.__importStar(require("./events.js"));
const events_1 = require("./events.js");
const PlanAPI = tslib_1.__importStar(require("./plan.js"));
const plan_1 = require("./plan.js");
const TelnyxAgentsAPI = tslib_1.__importStar(require("./telnyx-agents.js"));
const telnyx_agents_1 = require("./telnyx-agents.js");
const pagination_1 = require("../../../../core/pagination.js");
const path_1 = require("../../../../internal/utils/path.js");
class Runs extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.events = new EventsAPI.Events(this._client);
        this.plan = new PlanAPI.Plan(this._client);
        this.telnyxAgents = new TelnyxAgentsAPI.TelnyxAgents(this._client);
    }
    /**
     * Start a new run for a mission
     *
     * @example
     * ```ts
     * const run = await client.ai.missions.runs.create(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    create(missionID, body = {}, options) {
        return this._client.post((0, path_1.path) `/ai/missions/${missionID}/runs`, { body, ...options });
    }
    /**
     * Get details of a specific run
     *
     * @example
     * ```ts
     * const run = await client.ai.missions.runs.retrieve(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   { mission_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e' },
     * );
     * ```
     */
    retrieve(runID, params, options) {
        const { mission_id } = params;
        return this._client.get((0, path_1.path) `/ai/missions/${mission_id}/runs/${runID}`, options);
    }
    /**
     * Update run status and/or result
     *
     * @example
     * ```ts
     * const run = await client.ai.missions.runs.update(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   { mission_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e' },
     * );
     * ```
     */
    update(runID, params, options) {
        const { mission_id, ...body } = params;
        return this._client.patch((0, path_1.path) `/ai/missions/${mission_id}/runs/${runID}`, { body, ...options });
    }
    /**
     * List all runs for a specific mission
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const missionRunData of client.ai.missions.runs.list(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * )) {
     *   // ...
     * }
     * ```
     */
    list(missionID, query = {}, options) {
        return this._client.getAPIList((0, path_1.path) `/ai/missions/${missionID}/runs`, (pagination_1.DefaultFlatPagination), { query, ...options });
    }
    /**
     * Cancel a running or paused run
     *
     * @example
     * ```ts
     * const response = await client.ai.missions.runs.cancelRun(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   { mission_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e' },
     * );
     * ```
     */
    cancelRun(runID, params, options) {
        const { mission_id } = params;
        return this._client.post((0, path_1.path) `/ai/missions/${mission_id}/runs/${runID}/cancel`, options);
    }
    /**
     * List recent runs across all missions
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const missionRunData of client.ai.missions.runs.listRuns()) {
     *   // ...
     * }
     * ```
     */
    listRuns(query = {}, options) {
        return this._client.getAPIList('/ai/missions/runs', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Pause a running run
     *
     * @example
     * ```ts
     * const response = await client.ai.missions.runs.pauseRun(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   { mission_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e' },
     * );
     * ```
     */
    pauseRun(runID, params, options) {
        const { mission_id } = params;
        return this._client.post((0, path_1.path) `/ai/missions/${mission_id}/runs/${runID}/pause`, options);
    }
    /**
     * Resume a paused run
     *
     * @example
     * ```ts
     * const response = await client.ai.missions.runs.resumeRun(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   { mission_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e' },
     * );
     * ```
     */
    resumeRun(runID, params, options) {
        const { mission_id } = params;
        return this._client.post((0, path_1.path) `/ai/missions/${mission_id}/runs/${runID}/resume`, options);
    }
}
exports.Runs = Runs;
Runs.Events = events_1.Events;
Runs.Plan = plan_1.Plan;
Runs.TelnyxAgents = telnyx_agents_1.TelnyxAgents;
