// telnyx/resources/phone-numbers/index.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Voicemail = exports.Voice = exports.PhoneNumbers = exports.Messaging = exports.Jobs = exports.CsvDownloads = exports.Actions = void 0;
var actions_1 = require("./actions.js");
Object.defineProperty(exports, "Actions", { enumerable: true, get: function () { return actions_1.Actions; } });
var csv_downloads_1 = require("./csv-downloads.js");
Object.defineProperty(exports, "CsvDownloads", { enumerable: true, get: function () { return csv_downloads_1.CsvDownloads; } });
var jobs_1 = require("./jobs.js");
Object.defineProperty(exports, "Jobs", { enumerable: true, get: function () { return jobs_1.Jobs; } });
var messaging_1 = require("./messaging.js");
Object.defineProperty(exports, "Messaging", { enumerable: true, get: function () { return messaging_1.Messaging; } });
var phone_numbers_1 = require("./phone-numbers.js");
Object.defineProperty(exports, "PhoneNumbers", { enumerable: true, get: function () { return phone_numbers_1.PhoneNumbers; } });
var voice_1 = require("./voice.js");
Object.defineProperty(exports, "Voice", { enumerable: true, get: function () { return voice_1.Voice; } });
var voicemail_1 = require("./voicemail.js");
Object.defineProperty(exports, "Voicemail", { enumerable: true, get: function () { return voicemail_1.Voicemail; } });
