// telnyx/resources/phone-numbers/jobs.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import { DefaultFlatPagination } from "../../core/pagination.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * Background jobs performed over a batch of phone numbers
 */
export class Jobs extends APIResource {
    /**
     * Retrieve a phone numbers job
     *
     * @example
     * ```ts
     * const job = await client.phoneNumbers.jobs.retrieve('id');
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/phone_numbers/jobs/${id}`, options);
    }
    /**
     * Lists the phone numbers jobs
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const phoneNumbersJob of client.phoneNumbers.jobs.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/phone_numbers/jobs', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Creates a new background job to delete a batch of numbers. At most one thousand
     * numbers can be updated per API call.
     *
     * @example
     * ```ts
     * const response = await client.phoneNumbers.jobs.deleteBatch(
     *   {
     *     phone_numbers: [
     *       '+19705555098',
     *       '+19715555098',
     *       '32873127836',
     *     ],
     *   },
     * );
     * ```
     */
    deleteBatch(body, options) {
        return this._client.post('/phone_numbers/jobs/delete_phone_numbers', { body, ...options });
    }
    /**
     * Creates a new background job to update a batch of numbers. At most one thousand
     * numbers can be updated per API call. At least one of the updateable fields must
     * be submitted. IMPORTANT: You must either specify filters (using the filter
     * parameters) or specific phone numbers (using the phone_numbers parameter in the
     * request body). If you specify filters, ALL phone numbers that match the given
     * filters (up to 1000 at a time) will be updated. If you want to update only
     * specific numbers, you must use the phone_numbers parameter in the request body.
     * When using the phone_numbers parameter, ensure you follow the correct format as
     * shown in the example (either phone number IDs or phone numbers in E164 format).
     *
     * @example
     * ```ts
     * const response = await client.phoneNumbers.jobs.updateBatch(
     *   {
     *     phone_numbers: ['1583466971586889004', '+13127367254'],
     *   },
     * );
     * ```
     */
    updateBatch(params, options) {
        const { filter, ...body } = params;
        return this._client.post('/phone_numbers/jobs/update_phone_numbers', {
            query: { filter },
            body,
            ...options,
        });
    }
    /**
     * Creates a background job to update the emergency settings of a collection of
     * phone numbers. At most one thousand numbers can be updated per API call.
     *
     * @example
     * ```ts
     * const response =
     *   await client.phoneNumbers.jobs.updateEmergencySettingsBatch(
     *     {
     *       emergency_enabled: true,
     *       phone_numbers: [
     *         '+19705555098',
     *         '+19715555098',
     *         '32873127836',
     *       ],
     *     },
     *   );
     * ```
     */
    updateEmergencySettingsBatch(body, options) {
        return this._client.post('/phone_numbers/jobs/update_emergency_settings', { body, ...options });
    }
}
