// telnyx/resources/phone-numbers/phone-numbers.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as ActionsAPI from "./actions.mjs";
import { Actions, } from "./actions.mjs";
import * as CsvDownloadsAPI from "./csv-downloads.mjs";
import { CsvDownloads, } from "./csv-downloads.mjs";
import * as JobsAPI from "./jobs.mjs";
import { Jobs, } from "./jobs.mjs";
import * as MessagingAPI from "./messaging.mjs";
import { Messaging, } from "./messaging.mjs";
import * as VoiceAPI from "./voice.mjs";
import { Voice, } from "./voice.mjs";
import * as VoicemailAPI from "./voicemail.mjs";
import { Voicemail, } from "./voicemail.mjs";
import { DefaultFlatPagination } from "../../core/pagination.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * Configure your phone numbers
 */
export class PhoneNumbers extends APIResource {
    constructor() {
        super(...arguments);
        this.actions = new ActionsAPI.Actions(this._client);
        this.csvDownloads = new CsvDownloadsAPI.CsvDownloads(this._client);
        this.jobs = new JobsAPI.Jobs(this._client);
        this.messaging = new MessagingAPI.Messaging(this._client);
        this.voice = new VoiceAPI.Voice(this._client);
        this.voicemail = new VoicemailAPI.Voicemail(this._client);
    }
    /**
     * Retrieve a phone number
     *
     * @example
     * ```ts
     * const phoneNumber = await client.phoneNumbers.retrieve(
     *   '1293384261075731499',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/phone_numbers/${id}`, options);
    }
    /**
     * Update a phone number
     *
     * @example
     * ```ts
     * const phoneNumber = await client.phoneNumbers.update(
     *   '1293384261075731499',
     * );
     * ```
     */
    update(phoneNumberID, body, options) {
        return this._client.patch(path `/phone_numbers/${phoneNumberID}`, { body, ...options });
    }
    /**
     * List phone numbers
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const phoneNumberDetailed of client.phoneNumbers.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/phone_numbers', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete a phone number
     *
     * @example
     * ```ts
     * const phoneNumber = await client.phoneNumbers.delete(
     *   '1293384261075731499',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/phone_numbers/${id}`, options);
    }
    /**
     * List phone numbers, This endpoint is a lighter version of the /phone_numbers
     * endpoint having higher performance and rate limit.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const phoneNumberSlimListResponse of client.phoneNumbers.slimList()) {
     *   // ...
     * }
     * ```
     */
    slimList(query = {}, options) {
        return this._client.getAPIList('/phone_numbers/slim', (DefaultFlatPagination), { query, ...options });
    }
}
PhoneNumbers.Actions = Actions;
PhoneNumbers.CsvDownloads = CsvDownloads;
PhoneNumbers.Jobs = Jobs;
PhoneNumbers.Messaging = Messaging;
PhoneNumbers.Voice = Voice;
PhoneNumbers.Voicemail = Voicemail;
