// telnyx/resources/phone-numbers/phone-numbers.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.PhoneNumbers = void 0;
const tslib_1 = require("../../internal/tslib.js");
const resource_1 = require("../../core/resource.js");
const ActionsAPI = tslib_1.__importStar(require("./actions.js"));
const actions_1 = require("./actions.js");
const CsvDownloadsAPI = tslib_1.__importStar(require("./csv-downloads.js"));
const csv_downloads_1 = require("./csv-downloads.js");
const JobsAPI = tslib_1.__importStar(require("./jobs.js"));
const jobs_1 = require("./jobs.js");
const MessagingAPI = tslib_1.__importStar(require("./messaging.js"));
const messaging_1 = require("./messaging.js");
const VoiceAPI = tslib_1.__importStar(require("./voice.js"));
const voice_1 = require("./voice.js");
const VoicemailAPI = tslib_1.__importStar(require("./voicemail.js"));
const voicemail_1 = require("./voicemail.js");
const pagination_1 = require("../../core/pagination.js");
const path_1 = require("../../internal/utils/path.js");
/**
 * Configure your phone numbers
 */
class PhoneNumbers extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.actions = new ActionsAPI.Actions(this._client);
        this.csvDownloads = new CsvDownloadsAPI.CsvDownloads(this._client);
        this.jobs = new JobsAPI.Jobs(this._client);
        this.messaging = new MessagingAPI.Messaging(this._client);
        this.voice = new VoiceAPI.Voice(this._client);
        this.voicemail = new VoicemailAPI.Voicemail(this._client);
    }
    /**
     * Retrieve a phone number
     *
     * @example
     * ```ts
     * const phoneNumber = await client.phoneNumbers.retrieve(
     *   '1293384261075731499',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/phone_numbers/${id}`, options);
    }
    /**
     * Update a phone number
     *
     * @example
     * ```ts
     * const phoneNumber = await client.phoneNumbers.update(
     *   '1293384261075731499',
     * );
     * ```
     */
    update(phoneNumberID, body, options) {
        return this._client.patch((0, path_1.path) `/phone_numbers/${phoneNumberID}`, { body, ...options });
    }
    /**
     * List phone numbers
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const phoneNumberDetailed of client.phoneNumbers.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/phone_numbers', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete a phone number
     *
     * @example
     * ```ts
     * const phoneNumber = await client.phoneNumbers.delete(
     *   '1293384261075731499',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/phone_numbers/${id}`, options);
    }
    /**
     * List phone numbers, This endpoint is a lighter version of the /phone_numbers
     * endpoint having higher performance and rate limit.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const phoneNumberSlimListResponse of client.phoneNumbers.slimList()) {
     *   // ...
     * }
     * ```
     */
    slimList(query = {}, options) {
        return this._client.getAPIList('/phone_numbers/slim', (pagination_1.DefaultFlatPagination), { query, ...options });
    }
}
exports.PhoneNumbers = PhoneNumbers;
PhoneNumbers.Actions = actions_1.Actions;
PhoneNumbers.CsvDownloads = csv_downloads_1.CsvDownloads;
PhoneNumbers.Jobs = jobs_1.Jobs;
PhoneNumbers.Messaging = messaging_1.Messaging;
PhoneNumbers.Voice = voice_1.Voice;
PhoneNumbers.Voicemail = voicemail_1.Voicemail;
