// telnyx/resources/whatsapp/business-accounts/business-accounts.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../core/resource.mjs";
import * as PhoneNumbersAPI from "./phone-numbers.mjs";
import { PhoneNumbers, } from "./phone-numbers.mjs";
import * as SettingsAPI from "./settings.mjs";
import { Settings, } from "./settings.mjs";
import { DefaultFlatPagination, } from "../../../core/pagination.mjs";
import { buildHeaders } from "../../../internal/headers.mjs";
import { path } from "../../../internal/utils/path.mjs";
/**
 * Manage Whatsapp business accounts
 */
export class BusinessAccounts extends APIResource {
    constructor() {
        super(...arguments);
        this.phoneNumbers = new PhoneNumbersAPI.PhoneNumbers(this._client);
        this.settings = new SettingsAPI.Settings(this._client);
    }
    /**
     * Get a single Whatsapp Business Account
     *
     * @example
     * ```ts
     * const businessAccount =
     *   await client.whatsapp.businessAccounts.retrieve('id');
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/v2/whatsapp/business_accounts/${id}`, options);
    }
    /**
     * List Whatsapp Business Accounts
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const businessAccountListResponse of client.whatsapp.businessAccounts.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/v2/whatsapp/business_accounts', (DefaultFlatPagination), { query, ...options });
    }
    /**
     * Delete a Whatsapp Business Account
     *
     * @example
     * ```ts
     * await client.whatsapp.businessAccounts.delete('id');
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/v2/whatsapp/business_accounts/${id}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
BusinessAccounts.PhoneNumbers = PhoneNumbers;
BusinessAccounts.Settings = Settings;
