// telnyx/resources/whatsapp/business-accounts/business-accounts.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.BusinessAccounts = void 0;
const tslib_1 = require("../../../internal/tslib.js");
const resource_1 = require("../../../core/resource.js");
const PhoneNumbersAPI = tslib_1.__importStar(require("./phone-numbers.js"));
const phone_numbers_1 = require("./phone-numbers.js");
const SettingsAPI = tslib_1.__importStar(require("./settings.js"));
const settings_1 = require("./settings.js");
const pagination_1 = require("../../../core/pagination.js");
const headers_1 = require("../../../internal/headers.js");
const path_1 = require("../../../internal/utils/path.js");
/**
 * Manage Whatsapp business accounts
 */
class BusinessAccounts extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.phoneNumbers = new PhoneNumbersAPI.PhoneNumbers(this._client);
        this.settings = new SettingsAPI.Settings(this._client);
    }
    /**
     * Get a single Whatsapp Business Account
     *
     * @example
     * ```ts
     * const businessAccount =
     *   await client.whatsapp.businessAccounts.retrieve('id');
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/v2/whatsapp/business_accounts/${id}`, options);
    }
    /**
     * List Whatsapp Business Accounts
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const businessAccountListResponse of client.whatsapp.businessAccounts.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/v2/whatsapp/business_accounts', (pagination_1.DefaultFlatPagination), { query, ...options });
    }
    /**
     * Delete a Whatsapp Business Account
     *
     * @example
     * ```ts
     * await client.whatsapp.businessAccounts.delete('id');
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/v2/whatsapp/business_accounts/${id}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.BusinessAccounts = BusinessAccounts;
BusinessAccounts.PhoneNumbers = phone_numbers_1.PhoneNumbers;
BusinessAccounts.Settings = settings_1.Settings;
