// telnyx/resources/external-connections/external-connections.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExternalConnections = void 0;
const tslib_1 = require("../../internal/tslib.js");
const resource_1 = require("../../core/resource.js");
const CivicAddressesAPI = tslib_1.__importStar(require("./civic-addresses.js"));
const civic_addresses_1 = require("./civic-addresses.js");
const LogMessagesAPI = tslib_1.__importStar(require("./log-messages.js"));
const log_messages_1 = require("./log-messages.js");
const PhoneNumbersAPI = tslib_1.__importStar(require("./phone-numbers.js"));
const phone_numbers_1 = require("./phone-numbers.js");
const ReleasesAPI = tslib_1.__importStar(require("./releases.js"));
const releases_1 = require("./releases.js");
const UploadsAPI = tslib_1.__importStar(require("./uploads.js"));
const uploads_1 = require("./uploads.js");
const pagination_1 = require("../../core/pagination.js");
const path_1 = require("../../internal/utils/path.js");
/**
 * External Connections operations
 */
class ExternalConnections extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.logMessages = new LogMessagesAPI.LogMessages(this._client);
        this.civicAddresses = new CivicAddressesAPI.CivicAddresses(this._client);
        this.phoneNumbers = new PhoneNumbersAPI.PhoneNumbers(this._client);
        this.releases = new ReleasesAPI.Releases(this._client);
        this.uploads = new UploadsAPI.Uploads(this._client);
    }
    /**
     * Creates a new External Connection based on the parameters sent in the request.
     * The external_sip_connection and outbound voice profile id are required. Once
     * created, you can assign phone numbers to your application using the
     * `/phone_numbers` endpoint.
     *
     * @example
     * ```ts
     * const externalConnection =
     *   await client.externalConnections.create({
     *     external_sip_connection: 'zoom',
     *     outbound: {},
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/external_connections', { body, ...options });
    }
    /**
     * Return the details of an existing External Connection inside the 'data'
     * attribute of the response.
     *
     * @example
     * ```ts
     * const externalConnection =
     *   await client.externalConnections.retrieve(
     *     '1293384261075731499',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/external_connections/${id}`, options);
    }
    /**
     * Updates settings of an existing External Connection based on the parameters of
     * the request.
     *
     * @example
     * ```ts
     * const externalConnection =
     *   await client.externalConnections.update(
     *     '1293384261075731499',
     *     {
     *       outbound: {
     *         outbound_voice_profile_id: '1911630617284445511',
     *       },
     *     },
     *   );
     * ```
     */
    update(id, body, options) {
        return this._client.patch((0, path_1.path) `/external_connections/${id}`, { body, ...options });
    }
    /**
     * This endpoint returns a list of your External Connections inside the 'data'
     * attribute of the response. External Connections are used by Telnyx customers to
     * seamless configure SIP trunking integrations with Telnyx Partners, through
     * External Voice Integrations in Mission Control Portal.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const externalConnection of client.externalConnections.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/external_connections', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Permanently deletes an External Connection. Deletion may be prevented if the
     * application is in use by phone numbers, is active, or if it is an Operator
     * Connect connection. To remove an Operator Connect integration please contact
     * Telnyx support.
     *
     * @example
     * ```ts
     * const externalConnection =
     *   await client.externalConnections.delete(
     *     '1293384261075731499',
     *   );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/external_connections/${id}`, options);
    }
    /**
     * Update a location's static emergency address
     *
     * @example
     * ```ts
     * const response =
     *   await client.externalConnections.updateLocation(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     {
     *       id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *       static_emergency_address_id:
     *         '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     },
     *   );
     * ```
     */
    updateLocation(locationID, params, options) {
        const { id, ...body } = params;
        return this._client.patch((0, path_1.path) `/external_connections/${id}/locations/${locationID}`, {
            body,
            ...options,
        });
    }
}
exports.ExternalConnections = ExternalConnections;
ExternalConnections.LogMessages = log_messages_1.LogMessages;
ExternalConnections.CivicAddresses = civic_addresses_1.CivicAddresses;
ExternalConnections.PhoneNumbers = phone_numbers_1.PhoneNumbers;
ExternalConnections.Releases = releases_1.Releases;
ExternalConnections.Uploads = uploads_1.Uploads;
