// telnyx/resources/external-connections/external-connections.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as CivicAddressesAPI from "./civic-addresses.mjs";
import { CivicAddresses, } from "./civic-addresses.mjs";
import * as LogMessagesAPI from "./log-messages.mjs";
import { LogMessages, } from "./log-messages.mjs";
import * as PhoneNumbersAPI from "./phone-numbers.mjs";
import { PhoneNumbers, } from "./phone-numbers.mjs";
import * as ReleasesAPI from "./releases.mjs";
import { Releases, } from "./releases.mjs";
import * as UploadsAPI from "./uploads.mjs";
import { Uploads, } from "./uploads.mjs";
import { DefaultFlatPagination } from "../../core/pagination.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * External Connections operations
 */
export class ExternalConnections extends APIResource {
    constructor() {
        super(...arguments);
        this.logMessages = new LogMessagesAPI.LogMessages(this._client);
        this.civicAddresses = new CivicAddressesAPI.CivicAddresses(this._client);
        this.phoneNumbers = new PhoneNumbersAPI.PhoneNumbers(this._client);
        this.releases = new ReleasesAPI.Releases(this._client);
        this.uploads = new UploadsAPI.Uploads(this._client);
    }
    /**
     * Creates a new External Connection based on the parameters sent in the request.
     * The external_sip_connection and outbound voice profile id are required. Once
     * created, you can assign phone numbers to your application using the
     * `/phone_numbers` endpoint.
     *
     * @example
     * ```ts
     * const externalConnection =
     *   await client.externalConnections.create({
     *     external_sip_connection: 'zoom',
     *     outbound: {},
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/external_connections', { body, ...options });
    }
    /**
     * Return the details of an existing External Connection inside the 'data'
     * attribute of the response.
     *
     * @example
     * ```ts
     * const externalConnection =
     *   await client.externalConnections.retrieve(
     *     '1293384261075731499',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/external_connections/${id}`, options);
    }
    /**
     * Updates settings of an existing External Connection based on the parameters of
     * the request.
     *
     * @example
     * ```ts
     * const externalConnection =
     *   await client.externalConnections.update(
     *     '1293384261075731499',
     *     {
     *       outbound: {
     *         outbound_voice_profile_id: '1911630617284445511',
     *       },
     *     },
     *   );
     * ```
     */
    update(id, body, options) {
        return this._client.patch(path `/external_connections/${id}`, { body, ...options });
    }
    /**
     * This endpoint returns a list of your External Connections inside the 'data'
     * attribute of the response. External Connections are used by Telnyx customers to
     * seamless configure SIP trunking integrations with Telnyx Partners, through
     * External Voice Integrations in Mission Control Portal.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const externalConnection of client.externalConnections.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/external_connections', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Permanently deletes an External Connection. Deletion may be prevented if the
     * application is in use by phone numbers, is active, or if it is an Operator
     * Connect connection. To remove an Operator Connect integration please contact
     * Telnyx support.
     *
     * @example
     * ```ts
     * const externalConnection =
     *   await client.externalConnections.delete(
     *     '1293384261075731499',
     *   );
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/external_connections/${id}`, options);
    }
    /**
     * Update a location's static emergency address
     *
     * @example
     * ```ts
     * const response =
     *   await client.externalConnections.updateLocation(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     {
     *       id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *       static_emergency_address_id:
     *         '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     },
     *   );
     * ```
     */
    updateLocation(locationID, params, options) {
        const { id, ...body } = params;
        return this._client.patch(path `/external_connections/${id}/locations/${locationID}`, {
            body,
            ...options,
        });
    }
}
ExternalConnections.LogMessages = LogMessages;
ExternalConnections.CivicAddresses = CivicAddresses;
ExternalConnections.PhoneNumbers = PhoneNumbers;
ExternalConnections.Releases = Releases;
ExternalConnections.Uploads = Uploads;
