// telnyx/resources/external-connections/index.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Uploads = exports.Releases = exports.PhoneNumbers = exports.LogMessages = exports.ExternalConnections = exports.CivicAddresses = void 0;
var civic_addresses_1 = require("./civic-addresses.js");
Object.defineProperty(exports, "CivicAddresses", { enumerable: true, get: function () { return civic_addresses_1.CivicAddresses; } });
var external_connections_1 = require("./external-connections.js");
Object.defineProperty(exports, "ExternalConnections", { enumerable: true, get: function () { return external_connections_1.ExternalConnections; } });
var log_messages_1 = require("./log-messages.js");
Object.defineProperty(exports, "LogMessages", { enumerable: true, get: function () { return log_messages_1.LogMessages; } });
var phone_numbers_1 = require("./phone-numbers.js");
Object.defineProperty(exports, "PhoneNumbers", { enumerable: true, get: function () { return phone_numbers_1.PhoneNumbers; } });
var releases_1 = require("./releases.js");
Object.defineProperty(exports, "Releases", { enumerable: true, get: function () { return releases_1.Releases; } });
var uploads_1 = require("./uploads.js");
Object.defineProperty(exports, "Uploads", { enumerable: true, get: function () { return uploads_1.Uploads; } });
