// telnyx/resources/external-connections/uploads.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Uploads = void 0;
const resource_1 = require("../../core/resource.js");
const pagination_1 = require("../../core/pagination.js");
const path_1 = require("../../internal/utils/path.js");
/**
 * External Connections operations
 */
class Uploads extends resource_1.APIResource {
    /**
     * Creates a new Upload request to Microsoft teams with the included phone numbers.
     * Only one of civic_address_id or location_id must be provided, not both. The
     * maximum allowed phone numbers for the numbers_ids array is 1000.
     *
     * @example
     * ```ts
     * const upload =
     *   await client.externalConnections.uploads.create(
     *     '1293384261075731499',
     *     {
     *       number_ids: [
     *         '3920457616934164700',
     *         '3920457616934164701',
     *         '3920457616934164702',
     *         '3920457616934164703',
     *       ],
     *     },
     *   );
     * ```
     */
    create(id, body, options) {
        return this._client.post((0, path_1.path) `/external_connections/${id}/uploads`, { body, ...options });
    }
    /**
     * Return the details of an Upload request and its phone numbers.
     *
     * @example
     * ```ts
     * const upload =
     *   await client.externalConnections.uploads.retrieve(
     *     '7b6a6449-b055-45a6-81f6-f6f0dffa4cc6',
     *     { id: '1293384261075731499' },
     *   );
     * ```
     */
    retrieve(ticketID, params, options) {
        const { id } = params;
        return this._client.get((0, path_1.path) `/external_connections/${id}/uploads/${ticketID}`, options);
    }
    /**
     * Returns a list of your Upload requests for the given external connection.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const upload of client.externalConnections.uploads.list(
     *   '1293384261075731499',
     * )) {
     *   // ...
     * }
     * ```
     */
    list(id, query = {}, options) {
        return this._client.getAPIList((0, path_1.path) `/external_connections/${id}/uploads`, (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Returns the count of all pending upload requests for the given external
     * connection.
     *
     * @example
     * ```ts
     * const response =
     *   await client.externalConnections.uploads.pendingCount(
     *     '1293384261075731499',
     *   );
     * ```
     */
    pendingCount(id, options) {
        return this._client.get((0, path_1.path) `/external_connections/${id}/uploads/status`, options);
    }
    /**
     * Forces a recheck of the status of all pending Upload requests for the given
     * external connection in the background.
     *
     * @example
     * ```ts
     * const response =
     *   await client.externalConnections.uploads.refreshStatus(
     *     '1293384261075731499',
     *   );
     * ```
     */
    refreshStatus(id, options) {
        return this._client.post((0, path_1.path) `/external_connections/${id}/uploads/refresh`, options);
    }
    /**
     * If there were any errors during the upload process, this endpoint will retry the
     * upload request. In some cases this will reattempt the existing upload request,
     * in other cases it may create a new upload request. Please check the ticket_id in
     * the response to determine if a new upload request was created.
     *
     * @example
     * ```ts
     * const response =
     *   await client.externalConnections.uploads.retry(
     *     '7b6a6449-b055-45a6-81f6-f6f0dffa4cc6',
     *     { id: '1293384261075731499' },
     *   );
     * ```
     */
    retry(ticketID, params, options) {
        const { id } = params;
        return this._client.post((0, path_1.path) `/external_connections/${id}/uploads/${ticketID}/retry`, options);
    }
}
exports.Uploads = Uploads;
