// telnyx/resources/messaging-10dlc/campaign/campaign.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Campaign = void 0;
const tslib_1 = require("../../../internal/tslib.js");
const resource_1 = require("../../../core/resource.js");
const OsrAPI = tslib_1.__importStar(require("./osr.js"));
const osr_1 = require("./osr.js");
const UsecaseAPI = tslib_1.__importStar(require("./usecase.js"));
const usecase_1 = require("./usecase.js");
const pagination_1 = require("../../../core/pagination.js");
const path_1 = require("../../../internal/utils/path.js");
/**
 * Campaign operations
 */
class Campaign extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.usecase = new UsecaseAPI.Usecase(this._client);
        this.osr = new OsrAPI.Osr(this._client);
    }
    /**
     * Retrieve campaign details by `campaignId`.
     *
     * @example
     * ```ts
     * const telnyxCampaignCsp =
     *   await client.messaging10dlc.campaign.retrieve(
     *     'campaignId',
     *   );
     * ```
     */
    retrieve(campaignID, options) {
        return this._client.get((0, path_1.path) `/10dlc/campaign/${campaignID}`, options);
    }
    /**
     * Update a campaign's properties by `campaignId`. **Please note:** only sample
     * messages are editable.
     *
     * @example
     * ```ts
     * const telnyxCampaignCsp =
     *   await client.messaging10dlc.campaign.update('campaignId');
     * ```
     */
    update(campaignID, body, options) {
        return this._client.put((0, path_1.path) `/10dlc/campaign/${campaignID}`, { body, ...options });
    }
    /**
     * Retrieve a list of campaigns associated with a supplied `brandId`.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const campaignListResponse of client.messaging10dlc.campaign.list(
     *   { brandId: 'brandId' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(query, options) {
        return this._client.getAPIList('/10dlc/campaign', (pagination_1.PerPagePaginationV2), {
            query,
            ...options,
        });
    }
    /**
     * Manually accept a campaign shared with Telnyx
     *
     * @example
     * ```ts
     * const response =
     *   await client.messaging10dlc.campaign.acceptSharing(
     *     'C26F1KLZN',
     *   );
     * ```
     */
    acceptSharing(campaignID, options) {
        return this._client.post((0, path_1.path) `/10dlc/campaign/acceptSharing/${campaignID}`, options);
    }
    /**
     * Terminate a campaign. Note that once deactivated, a campaign cannot be restored.
     *
     * @example
     * ```ts
     * const response =
     *   await client.messaging10dlc.campaign.deactivate(
     *     'campaignId',
     *   );
     * ```
     */
    deactivate(campaignID, options) {
        return this._client.delete((0, path_1.path) `/10dlc/campaign/${campaignID}`, options);
    }
    /**
     * Get the campaign metadata for each MNO it was submitted to.
     *
     * @example
     * ```ts
     * const response =
     *   await client.messaging10dlc.campaign.getMnoMetadata(
     *     'campaignId',
     *   );
     * ```
     */
    getMnoMetadata(campaignID, options) {
        return this._client.get((0, path_1.path) `/10dlc/campaign/${campaignID}/mnoMetadata`, options);
    }
    /**
     * Retrieve campaign's operation status at MNO level.
     *
     * @example
     * ```ts
     * const response =
     *   await client.messaging10dlc.campaign.getOperationStatus(
     *     'campaignId',
     *   );
     * ```
     */
    getOperationStatus(campaignID, options) {
        return this._client.get((0, path_1.path) `/10dlc/campaign/${campaignID}/operationStatus`, options);
    }
    /**
     * Get Sharing Status
     *
     * @example
     * ```ts
     * const response =
     *   await client.messaging10dlc.campaign.getSharingStatus(
     *     'campaignId',
     *   );
     * ```
     */
    getSharingStatus(campaignID, options) {
        return this._client.get((0, path_1.path) `/10dlc/campaign/${campaignID}/sharing`, options);
    }
    /**
     * Submits an appeal for rejected native campaigns in TELNYX_FAILED or MNO_REJECTED
     * status. The appeal is recorded for manual compliance team review and the
     * campaign status is reset to TCR_ACCEPTED. Note: Appeal forwarding is handled
     * manually to allow proper review before incurring upstream charges.
     *
     * @example
     * ```ts
     * const response =
     *   await client.messaging10dlc.campaign.submitAppeal(
     *     '5eb13888-32b7-4cab-95e6-d834dde21d64',
     *     {
     *       appeal_reason:
     *         'The website has been updated to include the required privacy policy and terms of service.',
     *     },
     *   );
     * ```
     */
    submitAppeal(campaignID, body, options) {
        return this._client.post((0, path_1.path) `/10dlc/campaign/${campaignID}/appeal`, { body, ...options });
    }
}
exports.Campaign = Campaign;
Campaign.Usecase = usecase_1.Usecase;
Campaign.Osr = osr_1.Osr;
