// telnyx/resources/reports/mdr-usage-reports.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.MdrUsageReports = void 0;
const resource_1 = require("../../core/resource.js");
const pagination_1 = require("../../core/pagination.js");
const headers_1 = require("../../internal/headers.js");
const path_1 = require("../../internal/utils/path.js");
/**
 * Messaging usage reports
 */
class MdrUsageReports extends resource_1.APIResource {
    /**
     * Submit request for new new messaging usage report. This endpoint will pull and
     * aggregate messaging data in specified time period.
     *
     * @example
     * ```ts
     * const mdrUsageReport =
     *   await client.reports.mdrUsageReports.create({
     *     aggregation_type: 'NO_AGGREGATION',
     *     end_date: '2020-07-01T00:00:00-06:00',
     *     start_date: '2020-07-01T00:00:00-06:00',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/reports/mdr_usage_reports', {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ 'Content-Type': '*/*' }, options?.headers]),
        });
    }
    /**
     * Fetch a single messaging usage report by id
     *
     * @example
     * ```ts
     * const mdrUsageReport =
     *   await client.reports.mdrUsageReports.retrieve(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/reports/mdr_usage_reports/${id}`, options);
    }
    /**
     * Fetch all messaging usage reports. Usage reports are aggregated messaging data
     * for specified time period and breakdown
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const mdrUsageReport of client.reports.mdrUsageReports.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/reports/mdr_usage_reports', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete messaging usage report by id
     *
     * @example
     * ```ts
     * const mdrUsageReport =
     *   await client.reports.mdrUsageReports.delete(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/reports/mdr_usage_reports/${id}`, options);
    }
    /**
     * Generate and fetch messaging usage report synchronously. This endpoint will both
     * generate and fetch the messaging report over a specified time period. No polling
     * is necessary but the response may take up to a couple of minutes.
     *
     * @example
     * ```ts
     * const response =
     *   await client.reports.mdrUsageReports.fetchSync({
     *     aggregation_type: 'PROFILE',
     *   });
     * ```
     */
    fetchSync(query, options) {
        return this._client.get('/reports/mdr_usage_reports/sync', { query, ...options });
    }
}
exports.MdrUsageReports = MdrUsageReports;
