// telnyx/resources/authentication-providers.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { DefaultFlatPagination } from "../core/pagination.mjs";
import { path } from "../internal/utils/path.mjs";
export class AuthenticationProviders extends APIResource {
    /**
     * Creates an authentication provider.
     *
     * @example
     * ```ts
     * const authenticationProvider =
     *   await client.authenticationProviders.create({
     *     name: 'Okta',
     *     settings: {
     *       idp_cert_fingerprint:
     *         '13:38:C7:BB:C9:FF:4A:70:38:3A:E3:D9:5C:CD:DB:2E:50:1E:80:A7',
     *       idp_entity_id:
     *         'https://myorg.myidp.com/saml/metadata',
     *       idp_sso_target_url:
     *         'https://myorg.myidp.com/trust/saml2/http-post/sso',
     *     },
     *     short_name: 'myorg',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/authentication_providers', { body, ...options });
    }
    /**
     * Retrieves the details of an existing authentication provider.
     *
     * @example
     * ```ts
     * const authenticationProvider =
     *   await client.authenticationProviders.retrieve('id');
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/authentication_providers/${id}`, options);
    }
    /**
     * Updates settings of an existing authentication provider.
     *
     * @example
     * ```ts
     * const authenticationProvider =
     *   await client.authenticationProviders.update('id', {
     *     active: true,
     *     name: 'Okta',
     *     settings: {
     *       idp_entity_id:
     *         'https://myorg.myidp.com/saml/metadata',
     *       idp_sso_target_url:
     *         'https://myorg.myidp.com/trust/saml2/http-post/sso',
     *       idp_cert_fingerprint:
     *         '13:38:C7:BB:C9:FF:4A:70:38:3A:E3:D9:5C:CD:DB:2E:50:1E:80:A7',
     *       idp_cert_fingerprint_algorithm: 'sha1',
     *     },
     *     short_name: 'myorg',
     *   });
     * ```
     */
    update(id, body, options) {
        return this._client.patch(path `/authentication_providers/${id}`, { body, ...options });
    }
    /**
     * Returns a list of your SSO authentication providers.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const authenticationProvider of client.authenticationProviders.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/authentication_providers', (DefaultFlatPagination), { query, ...options });
    }
    /**
     * Deletes an existing authentication provider.
     *
     * @example
     * ```ts
     * const authenticationProvider =
     *   await client.authenticationProviders.delete('id');
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/authentication_providers/${id}`, options);
    }
}
