// telnyx/resources/billing-groups.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { DefaultFlatPagination } from "../core/pagination.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Billing groups operations
 */
export class BillingGroups extends APIResource {
    /**
     * Create a billing group
     *
     * @example
     * ```ts
     * const billingGroup = await client.billingGroups.create({
     *   name: 'string',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/billing_groups', { body, ...options });
    }
    /**
     * Get a billing group
     *
     * @example
     * ```ts
     * const billingGroup = await client.billingGroups.retrieve(
     *   'f5586561-8ff0-4291-a0ac-84fe544797bd',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/billing_groups/${id}`, options);
    }
    /**
     * Update a billing group
     *
     * @example
     * ```ts
     * const billingGroup = await client.billingGroups.update(
     *   'f5586561-8ff0-4291-a0ac-84fe544797bd',
     *   { name: 'string' },
     * );
     * ```
     */
    update(id, body, options) {
        return this._client.patch(path `/billing_groups/${id}`, { body, ...options });
    }
    /**
     * List all billing groups
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const billingGroup of client.billingGroups.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/billing_groups', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete a billing group
     *
     * @example
     * ```ts
     * const billingGroup = await client.billingGroups.delete(
     *   'f5586561-8ff0-4291-a0ac-84fe544797bd',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/billing_groups/${id}`, options);
    }
}
