// telnyx/resources/billing-groups.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.BillingGroups = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Billing groups operations
 */
class BillingGroups extends resource_1.APIResource {
    /**
     * Create a billing group
     *
     * @example
     * ```ts
     * const billingGroup = await client.billingGroups.create({
     *   name: 'string',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/billing_groups', { body, ...options });
    }
    /**
     * Get a billing group
     *
     * @example
     * ```ts
     * const billingGroup = await client.billingGroups.retrieve(
     *   'f5586561-8ff0-4291-a0ac-84fe544797bd',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/billing_groups/${id}`, options);
    }
    /**
     * Update a billing group
     *
     * @example
     * ```ts
     * const billingGroup = await client.billingGroups.update(
     *   'f5586561-8ff0-4291-a0ac-84fe544797bd',
     *   { name: 'string' },
     * );
     * ```
     */
    update(id, body, options) {
        return this._client.patch((0, path_1.path) `/billing_groups/${id}`, { body, ...options });
    }
    /**
     * List all billing groups
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const billingGroup of client.billingGroups.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/billing_groups', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete a billing group
     *
     * @example
     * ```ts
     * const billingGroup = await client.billingGroups.delete(
     *   'f5586561-8ff0-4291-a0ac-84fe544797bd',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/billing_groups/${id}`, options);
    }
}
exports.BillingGroups = BillingGroups;
