// telnyx/resources/call-control-applications.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.CallControlApplications = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Call Control applications operations
 */
class CallControlApplications extends resource_1.APIResource {
    /**
     * Create a call control application.
     *
     * @example
     * ```ts
     * const callControlApplication =
     *   await client.callControlApplications.create({
     *     application_name: 'call-router',
     *     webhook_event_url: 'https://example.com',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/call_control_applications', { body, ...options });
    }
    /**
     * Retrieves the details of an existing call control application.
     *
     * @example
     * ```ts
     * const callControlApplication =
     *   await client.callControlApplications.retrieve(
     *     '1293384261075731499',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/call_control_applications/${id}`, options);
    }
    /**
     * Updates settings of an existing call control application.
     *
     * @example
     * ```ts
     * const callControlApplication =
     *   await client.callControlApplications.update(
     *     '1293384261075731499',
     *     {
     *       application_name: 'call-router',
     *       webhook_event_url: 'https://example.com',
     *     },
     *   );
     * ```
     */
    update(id, body, options) {
        return this._client.patch((0, path_1.path) `/call_control_applications/${id}`, { body, ...options });
    }
    /**
     * Return a list of call control applications.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const callControlApplication of client.callControlApplications.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/call_control_applications', (pagination_1.DefaultFlatPagination), { query, ...options });
    }
    /**
     * Deletes a call control application.
     *
     * @example
     * ```ts
     * const callControlApplication =
     *   await client.callControlApplications.delete(
     *     '1293384261075731499',
     *   );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/call_control_applications/${id}`, options);
    }
}
exports.CallControlApplications = CallControlApplications;
