// telnyx/resources/call-control-applications.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { DefaultFlatPagination } from "../core/pagination.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Call Control applications operations
 */
export class CallControlApplications extends APIResource {
    /**
     * Create a call control application.
     *
     * @example
     * ```ts
     * const callControlApplication =
     *   await client.callControlApplications.create({
     *     application_name: 'call-router',
     *     webhook_event_url: 'https://example.com',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/call_control_applications', { body, ...options });
    }
    /**
     * Retrieves the details of an existing call control application.
     *
     * @example
     * ```ts
     * const callControlApplication =
     *   await client.callControlApplications.retrieve(
     *     '1293384261075731499',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/call_control_applications/${id}`, options);
    }
    /**
     * Updates settings of an existing call control application.
     *
     * @example
     * ```ts
     * const callControlApplication =
     *   await client.callControlApplications.update(
     *     '1293384261075731499',
     *     {
     *       application_name: 'call-router',
     *       webhook_event_url: 'https://example.com',
     *     },
     *   );
     * ```
     */
    update(id, body, options) {
        return this._client.patch(path `/call_control_applications/${id}`, { body, ...options });
    }
    /**
     * Return a list of call control applications.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const callControlApplication of client.callControlApplications.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/call_control_applications', (DefaultFlatPagination), { query, ...options });
    }
    /**
     * Deletes a call control application.
     *
     * @example
     * ```ts
     * const callControlApplication =
     *   await client.callControlApplications.delete(
     *     '1293384261075731499',
     *   );
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/call_control_applications/${id}`, options);
    }
}
