// telnyx/resources/custom-storage-credentials.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomStorageCredentials = void 0;
const resource_1 = require("../core/resource.js");
const headers_1 = require("../internal/headers.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Call Recordings operations.
 */
class CustomStorageCredentials extends resource_1.APIResource {
    /**
     * Creates a custom storage credentials configuration.
     *
     * @example
     * ```ts
     * const customStorageCredential =
     *   await client.customStorageCredentials.create(
     *     'connection_id',
     *     {
     *       backend: 'gcs',
     *       configuration: { backend: 'gcs' },
     *     },
     *   );
     * ```
     */
    create(connectionID, body, options) {
        return this._client.post((0, path_1.path) `/custom_storage_credentials/${connectionID}`, { body, ...options });
    }
    /**
     * Returns the information about custom storage credentials.
     *
     * @example
     * ```ts
     * const customStorageCredential =
     *   await client.customStorageCredentials.retrieve(
     *     'connection_id',
     *   );
     * ```
     */
    retrieve(connectionID, options) {
        return this._client.get((0, path_1.path) `/custom_storage_credentials/${connectionID}`, options);
    }
    /**
     * Updates a stored custom credentials configuration.
     *
     * @example
     * ```ts
     * const customStorageCredential =
     *   await client.customStorageCredentials.update(
     *     'connection_id',
     *     {
     *       backend: 'gcs',
     *       configuration: { backend: 'gcs' },
     *     },
     *   );
     * ```
     */
    update(connectionID, body, options) {
        return this._client.put((0, path_1.path) `/custom_storage_credentials/${connectionID}`, { body, ...options });
    }
    /**
     * Deletes a stored custom credentials configuration.
     *
     * @example
     * ```ts
     * await client.customStorageCredentials.delete(
     *   'connection_id',
     * );
     * ```
     */
    delete(connectionID, options) {
        return this._client.delete((0, path_1.path) `/custom_storage_credentials/${connectionID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.CustomStorageCredentials = CustomStorageCredentials;
