// telnyx/resources/custom-storage-credentials.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { buildHeaders } from "../internal/headers.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Call Recordings operations.
 */
export class CustomStorageCredentials extends APIResource {
    /**
     * Creates a custom storage credentials configuration.
     *
     * @example
     * ```ts
     * const customStorageCredential =
     *   await client.customStorageCredentials.create(
     *     'connection_id',
     *     {
     *       backend: 'gcs',
     *       configuration: { backend: 'gcs' },
     *     },
     *   );
     * ```
     */
    create(connectionID, body, options) {
        return this._client.post(path `/custom_storage_credentials/${connectionID}`, { body, ...options });
    }
    /**
     * Returns the information about custom storage credentials.
     *
     * @example
     * ```ts
     * const customStorageCredential =
     *   await client.customStorageCredentials.retrieve(
     *     'connection_id',
     *   );
     * ```
     */
    retrieve(connectionID, options) {
        return this._client.get(path `/custom_storage_credentials/${connectionID}`, options);
    }
    /**
     * Updates a stored custom credentials configuration.
     *
     * @example
     * ```ts
     * const customStorageCredential =
     *   await client.customStorageCredentials.update(
     *     'connection_id',
     *     {
     *       backend: 'gcs',
     *       configuration: { backend: 'gcs' },
     *     },
     *   );
     * ```
     */
    update(connectionID, body, options) {
        return this._client.put(path `/custom_storage_credentials/${connectionID}`, { body, ...options });
    }
    /**
     * Deletes a stored custom credentials configuration.
     *
     * @example
     * ```ts
     * await client.customStorageCredentials.delete(
     *   'connection_id',
     * );
     * ```
     */
    delete(connectionID, options) {
        return this._client.delete(path `/custom_storage_credentials/${connectionID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
