// telnyx/resources/dialogflow-connections.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { buildHeaders } from "../internal/headers.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Dialogflow Connection Operations.
 */
export class DialogflowConnections extends APIResource {
    /**
     * Save Dialogflow Credentiails to Telnyx, so it can be used with other Telnyx
     * services.
     *
     * @example
     * ```ts
     * const dialogflowConnection =
     *   await client.dialogflowConnections.create(
     *     'connection_id',
     *     {
     *       service_account: {
     *         type: 'bar',
     *         project_id: 'bar',
     *         private_key_id: 'bar',
     *         private_key: 'bar',
     *         client_email: 'bar',
     *         client_id: 'bar',
     *         auth_uri: 'bar',
     *         token_uri: 'bar',
     *         auth_provider_x509_cert_url: 'bar',
     *         client_x509_cert_url: 'bar',
     *       },
     *     },
     *   );
     * ```
     */
    create(connectionID, body, options) {
        return this._client.post(path `/dialogflow_connections/${connectionID}`, { body, ...options });
    }
    /**
     * Return details of the Dialogflow connection associated with the given
     * CallControl connection.
     *
     * @example
     * ```ts
     * const dialogflowConnection =
     *   await client.dialogflowConnections.retrieve(
     *     'connection_id',
     *   );
     * ```
     */
    retrieve(connectionID, options) {
        return this._client.get(path `/dialogflow_connections/${connectionID}`, options);
    }
    /**
     * Updates a stored Dialogflow Connection.
     *
     * @example
     * ```ts
     * const dialogflowConnection =
     *   await client.dialogflowConnections.update(
     *     'connection_id',
     *     {
     *       service_account: {
     *         type: 'bar',
     *         project_id: 'bar',
     *         private_key_id: 'bar',
     *         private_key: 'bar',
     *         client_email: 'bar',
     *         client_id: 'bar',
     *         auth_uri: 'bar',
     *         token_uri: 'bar',
     *         auth_provider_x509_cert_url: 'bar',
     *         client_x509_cert_url: 'bar',
     *       },
     *     },
     *   );
     * ```
     */
    update(connectionID, body, options) {
        return this._client.put(path `/dialogflow_connections/${connectionID}`, { body, ...options });
    }
    /**
     * Deletes a stored Dialogflow Connection.
     *
     * @example
     * ```ts
     * await client.dialogflowConnections.delete('connection_id');
     * ```
     */
    delete(connectionID, options) {
        return this._client.delete(path `/dialogflow_connections/${connectionID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
