// telnyx/resources/dialogflow-connections.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.DialogflowConnections = void 0;
const resource_1 = require("../core/resource.js");
const headers_1 = require("../internal/headers.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Dialogflow Connection Operations.
 */
class DialogflowConnections extends resource_1.APIResource {
    /**
     * Save Dialogflow Credentiails to Telnyx, so it can be used with other Telnyx
     * services.
     *
     * @example
     * ```ts
     * const dialogflowConnection =
     *   await client.dialogflowConnections.create(
     *     'connection_id',
     *     {
     *       service_account: {
     *         type: 'bar',
     *         project_id: 'bar',
     *         private_key_id: 'bar',
     *         private_key: 'bar',
     *         client_email: 'bar',
     *         client_id: 'bar',
     *         auth_uri: 'bar',
     *         token_uri: 'bar',
     *         auth_provider_x509_cert_url: 'bar',
     *         client_x509_cert_url: 'bar',
     *       },
     *     },
     *   );
     * ```
     */
    create(connectionID, body, options) {
        return this._client.post((0, path_1.path) `/dialogflow_connections/${connectionID}`, { body, ...options });
    }
    /**
     * Return details of the Dialogflow connection associated with the given
     * CallControl connection.
     *
     * @example
     * ```ts
     * const dialogflowConnection =
     *   await client.dialogflowConnections.retrieve(
     *     'connection_id',
     *   );
     * ```
     */
    retrieve(connectionID, options) {
        return this._client.get((0, path_1.path) `/dialogflow_connections/${connectionID}`, options);
    }
    /**
     * Updates a stored Dialogflow Connection.
     *
     * @example
     * ```ts
     * const dialogflowConnection =
     *   await client.dialogflowConnections.update(
     *     'connection_id',
     *     {
     *       service_account: {
     *         type: 'bar',
     *         project_id: 'bar',
     *         private_key_id: 'bar',
     *         private_key: 'bar',
     *         client_email: 'bar',
     *         client_id: 'bar',
     *         auth_uri: 'bar',
     *         token_uri: 'bar',
     *         auth_provider_x509_cert_url: 'bar',
     *         client_x509_cert_url: 'bar',
     *       },
     *     },
     *   );
     * ```
     */
    update(connectionID, body, options) {
        return this._client.put((0, path_1.path) `/dialogflow_connections/${connectionID}`, { body, ...options });
    }
    /**
     * Deletes a stored Dialogflow Connection.
     *
     * @example
     * ```ts
     * await client.dialogflowConnections.delete('connection_id');
     * ```
     */
    delete(connectionID, options) {
        return this._client.delete((0, path_1.path) `/dialogflow_connections/${connectionID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.DialogflowConnections = DialogflowConnections;
