// telnyx/resources/documents.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { DefaultFlatPagination } from "../core/pagination.mjs";
import { buildHeaders } from "../internal/headers.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Documents
 */
export class Documents extends APIResource {
    /**
     * Retrieve a document.
     *
     * @example
     * ```ts
     * const document = await client.documents.retrieve(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/documents/${id}`, options);
    }
    /**
     * Update a document.
     *
     * @example
     * ```ts
     * const document = await client.documents.update(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    update(documentID, body, options) {
        return this._client.patch(path `/documents/${documentID}`, { body, ...options });
    }
    /**
     * List all documents ordered by created_at descending.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const docServiceDocument of client.documents.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/documents', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete a document.<br /><br />A document can only be deleted if it's not linked
     * to a service. If it is linked to a service, it must be unlinked prior to
     * deleting.
     *
     * @example
     * ```ts
     * const document = await client.documents.delete(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/documents/${id}`, options);
    }
    /**
     * Download a document.
     *
     * @example
     * ```ts
     * const response = await client.documents.download(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     *
     * const content = await response.blob();
     * console.log(content);
     * ```
     */
    download(id, options) {
        return this._client.get(path `/documents/${id}/download`, {
            ...options,
            headers: buildHeaders([{ Accept: 'application/octet-stream' }, options?.headers]),
            __binaryResponse: true,
        });
    }
    /**
     * Generates a temporary pre-signed URL that can be used to download the document
     * directly from the storage backend without authentication.
     *
     * @example
     * ```ts
     * const response =
     *   await client.documents.generateDownloadLink(
     *     '550e8400-e29b-41d4-a716-446655440000',
     *   );
     * ```
     */
    generateDownloadLink(id, options) {
        return this._client.get(path `/documents/${id}/download_link`, options);
    }
    /**
     * Upload a document.<br /><br />Uploaded files must be linked to a service within
     * 30 minutes or they will be automatically deleted.
     *
     * @example
     * ```ts
     * const response = await client.documents.upload({
     *   document: {},
     * });
     * ```
     */
    upload(params, options) {
        const { document } = params;
        return this._client.post('/documents?content-type=multipart', { body: document, ...options });
    }
    /**
     * Upload a document.<br /><br />Uploaded files must be linked to a service within
     * 30 minutes or they will be automatically deleted.
     *
     * @example
     * ```ts
     * const response = await client.documents.uploadJson({
     *   document: {},
     * });
     * ```
     */
    uploadJson(params, options) {
        const { document } = params;
        return this._client.post('/documents', { body: document, ...options });
    }
}
