// telnyx/resources/fax-applications.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.FaxApplications = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Fax Applications operations
 */
class FaxApplications extends resource_1.APIResource {
    /**
     * Creates a new Fax Application based on the parameters sent in the request. The
     * application name and webhook URL are required. Once created, you can assign
     * phone numbers to your application using the `/phone_numbers` endpoint.
     *
     * @example
     * ```ts
     * const faxApplication = await client.faxApplications.create({
     *   application_name: 'fax-router',
     *   webhook_event_url: 'https://example.com',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/fax_applications', { body, ...options });
    }
    /**
     * Return the details of an existing Fax Application inside the 'data' attribute of
     * the response.
     *
     * @example
     * ```ts
     * const faxApplication =
     *   await client.faxApplications.retrieve(
     *     '1293384261075731499',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/fax_applications/${id}`, options);
    }
    /**
     * Updates settings of an existing Fax Application based on the parameters of the
     * request.
     *
     * @example
     * ```ts
     * const faxApplication = await client.faxApplications.update(
     *   '1293384261075731499',
     *   {
     *     application_name: 'fax-router',
     *     webhook_event_url: 'https://example.com',
     *   },
     * );
     * ```
     */
    update(id, body, options) {
        return this._client.patch((0, path_1.path) `/fax_applications/${id}`, { body, ...options });
    }
    /**
     * This endpoint returns a list of your Fax Applications inside the 'data'
     * attribute of the response. You can adjust which applications are listed by using
     * filters. Fax Applications are used to configure how you send and receive faxes
     * using the Programmable Fax API with Telnyx.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const faxApplication of client.faxApplications.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/fax_applications', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Permanently deletes a Fax Application. Deletion may be prevented if the
     * application is in use by phone numbers.
     *
     * @example
     * ```ts
     * const faxApplication = await client.faxApplications.delete(
     *   '1293384261075731499',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/fax_applications/${id}`, options);
    }
}
exports.FaxApplications = FaxApplications;
