// telnyx/resources/fqdns.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { DefaultFlatPagination } from "../core/pagination.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * FQDN operations
 */
export class Fqdns extends APIResource {
    /**
     * Create a new FQDN object.
     *
     * @example
     * ```ts
     * const fqdn = await client.fqdns.create({
     *   connection_id: '1516447646313612565',
     *   dns_record_type: 'a',
     *   fqdn: 'example.com',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/fqdns', { body, ...options });
    }
    /**
     * Return the details regarding a specific FQDN.
     *
     * @example
     * ```ts
     * const fqdn = await client.fqdns.retrieve(
     *   '1517907029795014409',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/fqdns/${id}`, options);
    }
    /**
     * Update the details of a specific FQDN.
     *
     * @example
     * ```ts
     * const fqdn = await client.fqdns.update(
     *   '1517907029795014409',
     * );
     * ```
     */
    update(id, body = {}, options) {
        return this._client.patch(path `/fqdns/${id}`, { body, ...options });
    }
    /**
     * Get all FQDNs belonging to the user that match the given filters.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const fqdn of client.fqdns.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/fqdns', (DefaultFlatPagination), { query, ...options });
    }
    /**
     * Delete an FQDN.
     *
     * @example
     * ```ts
     * const fqdn = await client.fqdns.delete(
     *   '1517907029795014409',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/fqdns/${id}`, options);
    }
}
