// telnyx/resources/fqdn-connections.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { DefaultFlatPagination } from "../core/pagination.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * FQDN connection operations
 */
export class FqdnConnections extends APIResource {
    /**
     * Creates a FQDN connection.
     *
     * @example
     * ```ts
     * const fqdnConnection = await client.fqdnConnections.create({
     *   connection_name: 'string',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/fqdn_connections', { body, ...options });
    }
    /**
     * Retrieves the details of an existing FQDN connection.
     *
     * @example
     * ```ts
     * const fqdnConnection =
     *   await client.fqdnConnections.retrieve(
     *     '1293384261075731499',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/fqdn_connections/${id}`, options);
    }
    /**
     * Updates settings of an existing FQDN connection.
     *
     * @example
     * ```ts
     * const fqdnConnection = await client.fqdnConnections.update(
     *   '1293384261075731499',
     * );
     * ```
     */
    update(id, body, options) {
        return this._client.patch(path `/fqdn_connections/${id}`, { body, ...options });
    }
    /**
     * Returns a list of your FQDN connections.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const fqdnConnection of client.fqdnConnections.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/fqdn_connections', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Deletes an FQDN connection.
     *
     * @example
     * ```ts
     * const fqdnConnection = await client.fqdnConnections.delete(
     *   '1293384261075731499',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/fqdn_connections/${id}`, options);
    }
}
