// telnyx/resources/fqdn-connections.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.FqdnConnections = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const path_1 = require("../internal/utils/path.js");
/**
 * FQDN connection operations
 */
class FqdnConnections extends resource_1.APIResource {
    /**
     * Creates a FQDN connection.
     *
     * @example
     * ```ts
     * const fqdnConnection = await client.fqdnConnections.create({
     *   connection_name: 'string',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/fqdn_connections', { body, ...options });
    }
    /**
     * Retrieves the details of an existing FQDN connection.
     *
     * @example
     * ```ts
     * const fqdnConnection =
     *   await client.fqdnConnections.retrieve(
     *     '1293384261075731499',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/fqdn_connections/${id}`, options);
    }
    /**
     * Updates settings of an existing FQDN connection.
     *
     * @example
     * ```ts
     * const fqdnConnection = await client.fqdnConnections.update(
     *   '1293384261075731499',
     * );
     * ```
     */
    update(id, body, options) {
        return this._client.patch((0, path_1.path) `/fqdn_connections/${id}`, { body, ...options });
    }
    /**
     * Returns a list of your FQDN connections.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const fqdnConnection of client.fqdnConnections.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/fqdn_connections', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Deletes an FQDN connection.
     *
     * @example
     * ```ts
     * const fqdnConnection = await client.fqdnConnections.delete(
     *   '1293384261075731499',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/fqdn_connections/${id}`, options);
    }
}
exports.FqdnConnections = FqdnConnections;
