// telnyx/resources/global-ip-assignments.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { DefaultFlatPagination } from "../core/pagination.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Global IPs
 */
export class GlobalIPAssignments extends APIResource {
    /**
     * Create a Global IP assignment.
     */
    create(body, options) {
        return this._client.post('/global_ip_assignments', { body, ...options });
    }
    /**
     * Retrieve a Global IP assignment.
     */
    retrieve(id, options) {
        return this._client.get(path `/global_ip_assignments/${id}`, options);
    }
    /**
     * Update a Global IP assignment.
     */
    update(globalIPAssignmentID, params, options) {
        const { globalIpAssignmentUpdateRequest } = params;
        return this._client.patch(path `/global_ip_assignments/${globalIPAssignmentID}`, {
            body: globalIpAssignmentUpdateRequest,
            ...options,
        });
    }
    /**
     * List all Global IP assignments.
     */
    list(query = {}, options) {
        return this._client.getAPIList('/global_ip_assignments', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete a Global IP assignment.
     */
    delete(id, options) {
        return this._client.delete(path `/global_ip_assignments/${id}`, options);
    }
}
