// telnyx/resources/global-ip-assignments.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.GlobalIPAssignments = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Global IPs
 */
class GlobalIPAssignments extends resource_1.APIResource {
    /**
     * Create a Global IP assignment.
     */
    create(body, options) {
        return this._client.post('/global_ip_assignments', { body, ...options });
    }
    /**
     * Retrieve a Global IP assignment.
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/global_ip_assignments/${id}`, options);
    }
    /**
     * Update a Global IP assignment.
     */
    update(globalIPAssignmentID, params, options) {
        const { globalIpAssignmentUpdateRequest } = params;
        return this._client.patch((0, path_1.path) `/global_ip_assignments/${globalIPAssignmentID}`, {
            body: globalIpAssignmentUpdateRequest,
            ...options,
        });
    }
    /**
     * List all Global IP assignments.
     */
    list(query = {}, options) {
        return this._client.getAPIList('/global_ip_assignments', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete a Global IP assignment.
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/global_ip_assignments/${id}`, options);
    }
}
exports.GlobalIPAssignments = GlobalIPAssignments;
