// telnyx/resources/ip-connections.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.IPConnections = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const path_1 = require("../internal/utils/path.js");
/**
 * IP connection operations
 */
class IPConnections extends resource_1.APIResource {
    /**
     * Creates an IP connection.
     *
     * @example
     * ```ts
     * const ipConnection = await client.ipConnections.create();
     * ```
     */
    create(body, options) {
        return this._client.post('/ip_connections', { body, ...options });
    }
    /**
     * Retrieves the details of an existing ip connection.
     *
     * @example
     * ```ts
     * const ipConnection = await client.ipConnections.retrieve(
     *   'id',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/ip_connections/${id}`, options);
    }
    /**
     * Updates settings of an existing IP connection.
     *
     * @example
     * ```ts
     * const ipConnection = await client.ipConnections.update(
     *   'id',
     * );
     * ```
     */
    update(id, body, options) {
        return this._client.patch((0, path_1.path) `/ip_connections/${id}`, { body, ...options });
    }
    /**
     * Returns a list of your IP connections.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const ipConnection of client.ipConnections.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/ip_connections', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Deletes an existing IP connection.
     *
     * @example
     * ```ts
     * const ipConnection = await client.ipConnections.delete(
     *   'id',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/ip_connections/${id}`, options);
    }
}
exports.IPConnections = IPConnections;
