// telnyx/resources/ip-connections.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { DefaultFlatPagination } from "../core/pagination.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * IP connection operations
 */
export class IPConnections extends APIResource {
    /**
     * Creates an IP connection.
     *
     * @example
     * ```ts
     * const ipConnection = await client.ipConnections.create();
     * ```
     */
    create(body, options) {
        return this._client.post('/ip_connections', { body, ...options });
    }
    /**
     * Retrieves the details of an existing ip connection.
     *
     * @example
     * ```ts
     * const ipConnection = await client.ipConnections.retrieve(
     *   'id',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/ip_connections/${id}`, options);
    }
    /**
     * Updates settings of an existing IP connection.
     *
     * @example
     * ```ts
     * const ipConnection = await client.ipConnections.update(
     *   'id',
     * );
     * ```
     */
    update(id, body, options) {
        return this._client.patch(path `/ip_connections/${id}`, { body, ...options });
    }
    /**
     * Returns a list of your IP connections.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const ipConnection of client.ipConnections.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/ip_connections', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Deletes an existing IP connection.
     *
     * @example
     * ```ts
     * const ipConnection = await client.ipConnections.delete(
     *   'id',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/ip_connections/${id}`, options);
    }
}
