// telnyx/resources/ips.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { DefaultFlatPagination } from "../core/pagination.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * IP operations
 */
export class IPs extends APIResource {
    /**
     * Create a new IP object.
     *
     * @example
     * ```ts
     * const ip = await client.ips.create({
     *   ip_address: '192.168.0.0',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/ips', { body, ...options });
    }
    /**
     * Return the details regarding a specific IP.
     *
     * @example
     * ```ts
     * const ip = await client.ips.retrieve(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/ips/${id}`, options);
    }
    /**
     * Update the details of a specific IP.
     *
     * @example
     * ```ts
     * const ip = await client.ips.update(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *   { ip_address: '192.168.0.0' },
     * );
     * ```
     */
    update(id, body, options) {
        return this._client.patch(path `/ips/${id}`, { body, ...options });
    }
    /**
     * Get all IPs belonging to the user that match the given filters.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const ip of client.ips.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/ips', (DefaultFlatPagination), { query, ...options });
    }
    /**
     * Delete an IP.
     *
     * @example
     * ```ts
     * const ip = await client.ips.delete(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/ips/${id}`, options);
    }
}
