// telnyx/resources/media.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { buildHeaders } from "../internal/headers.mjs";
import { maybeMultipartFormRequestOptions } from "../internal/uploads.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Media Storage operations
 */
export class Media extends APIResource {
    /**
     * Returns the information about a stored media file.
     *
     * @example
     * ```ts
     * const media = await client.media.retrieve('media_name');
     * ```
     */
    retrieve(mediaName, options) {
        return this._client.get(path `/media/${mediaName}`, options);
    }
    /**
     * Updates a stored media file.
     *
     * @example
     * ```ts
     * const media = await client.media.update('media_name');
     * ```
     */
    update(mediaName, body, options) {
        return this._client.put(path `/media/${mediaName}`, maybeMultipartFormRequestOptions({ body, ...options }, this._client));
    }
    /**
     * Returns a list of stored media files.
     *
     * @example
     * ```ts
     * const media = await client.media.list();
     * ```
     */
    list(query = {}, options) {
        return this._client.get('/media', { query, ...options });
    }
    /**
     * Deletes a stored media file.
     *
     * @example
     * ```ts
     * await client.media.delete('media_name');
     * ```
     */
    delete(mediaName, options) {
        return this._client.delete(path `/media/${mediaName}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Downloads a stored media file.
     *
     * @example
     * ```ts
     * const response = await client.media.download('media_name');
     *
     * const content = await response.blob();
     * console.log(content);
     * ```
     */
    download(mediaName, options) {
        return this._client.get(path `/media/${mediaName}/download`, {
            ...options,
            headers: buildHeaders([{ Accept: 'application/octet-stream' }, options?.headers]),
            __binaryResponse: true,
        });
    }
    /**
     * Upload media file to Telnyx so it can be used with other Telnyx services
     *
     * @example
     * ```ts
     * const response = await client.media.upload({
     *   media_url: 'http://www.example.com/audio.mp3',
     * });
     * ```
     */
    upload(body, options) {
        return this._client.post('/media', maybeMultipartFormRequestOptions({ body, ...options }, this._client));
    }
}
