// telnyx/resources/media.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Media = void 0;
const resource_1 = require("../core/resource.js");
const headers_1 = require("../internal/headers.js");
const uploads_1 = require("../internal/uploads.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Media Storage operations
 */
class Media extends resource_1.APIResource {
    /**
     * Returns the information about a stored media file.
     *
     * @example
     * ```ts
     * const media = await client.media.retrieve('media_name');
     * ```
     */
    retrieve(mediaName, options) {
        return this._client.get((0, path_1.path) `/media/${mediaName}`, options);
    }
    /**
     * Updates a stored media file.
     *
     * @example
     * ```ts
     * const media = await client.media.update('media_name');
     * ```
     */
    update(mediaName, body, options) {
        return this._client.put((0, path_1.path) `/media/${mediaName}`, (0, uploads_1.maybeMultipartFormRequestOptions)({ body, ...options }, this._client));
    }
    /**
     * Returns a list of stored media files.
     *
     * @example
     * ```ts
     * const media = await client.media.list();
     * ```
     */
    list(query = {}, options) {
        return this._client.get('/media', { query, ...options });
    }
    /**
     * Deletes a stored media file.
     *
     * @example
     * ```ts
     * await client.media.delete('media_name');
     * ```
     */
    delete(mediaName, options) {
        return this._client.delete((0, path_1.path) `/media/${mediaName}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Downloads a stored media file.
     *
     * @example
     * ```ts
     * const response = await client.media.download('media_name');
     *
     * const content = await response.blob();
     * console.log(content);
     * ```
     */
    download(mediaName, options) {
        return this._client.get((0, path_1.path) `/media/${mediaName}/download`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: 'application/octet-stream' }, options?.headers]),
            __binaryResponse: true,
        });
    }
    /**
     * Upload media file to Telnyx so it can be used with other Telnyx services
     *
     * @example
     * ```ts
     * const response = await client.media.upload({
     *   media_url: 'http://www.example.com/audio.mp3',
     * });
     * ```
     */
    upload(body, options) {
        return this._client.post('/media', (0, uploads_1.maybeMultipartFormRequestOptions)({ body, ...options }, this._client));
    }
}
exports.Media = Media;
