// telnyx/resources/mobile-push-credentials.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { DefaultFlatPagination } from "../core/pagination.mjs";
import { buildHeaders } from "../internal/headers.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Mobile push credential management
 */
export class MobilePushCredentials extends APIResource {
    /**
     * Creates a new mobile push credential
     */
    create(params, options) {
        const { createMobilePushCredentialRequest } = params;
        return this._client.post('/mobile_push_credentials', {
            body: createMobilePushCredentialRequest,
            ...options,
        });
    }
    /**
     * Retrieves mobile push credential based on the given `push_credential_id`
     */
    retrieve(pushCredentialID, options) {
        return this._client.get(path `/mobile_push_credentials/${pushCredentialID}`, options);
    }
    /**
     * List mobile push credentials
     */
    list(query = {}, options) {
        return this._client.getAPIList('/mobile_push_credentials', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Deletes a mobile push credential based on the given `push_credential_id`
     */
    delete(pushCredentialID, options) {
        return this._client.delete(path `/mobile_push_credentials/${pushCredentialID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
