// telnyx/resources/mobile-voice-connections.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.MobileVoiceConnections = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Mobile voice connection operations
 */
class MobileVoiceConnections extends resource_1.APIResource {
    /**
     * Create a Mobile Voice Connection
     */
    create(body, options) {
        return this._client.post('/v2/mobile_voice_connections', { body, ...options });
    }
    /**
     * Retrieve a Mobile Voice Connection
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/v2/mobile_voice_connections/${id}`, options);
    }
    /**
     * Update a Mobile Voice Connection
     */
    update(id, body, options) {
        return this._client.patch((0, path_1.path) `/v2/mobile_voice_connections/${id}`, { body, ...options });
    }
    /**
     * List Mobile Voice Connections
     */
    list(query = {}, options) {
        return this._client.getAPIList('/v2/mobile_voice_connections', (pagination_1.DefaultFlatPagination), { query, ...options });
    }
    /**
     * Delete a Mobile Voice Connection
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/v2/mobile_voice_connections/${id}`, options);
    }
}
exports.MobileVoiceConnections = MobileVoiceConnections;
