// telnyx/resources/mobile-voice-connections.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { DefaultFlatPagination } from "../core/pagination.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Mobile voice connection operations
 */
export class MobileVoiceConnections extends APIResource {
    /**
     * Create a Mobile Voice Connection
     */
    create(body, options) {
        return this._client.post('/v2/mobile_voice_connections', { body, ...options });
    }
    /**
     * Retrieve a Mobile Voice Connection
     */
    retrieve(id, options) {
        return this._client.get(path `/v2/mobile_voice_connections/${id}`, options);
    }
    /**
     * Update a Mobile Voice Connection
     */
    update(id, body, options) {
        return this._client.patch(path `/v2/mobile_voice_connections/${id}`, { body, ...options });
    }
    /**
     * List Mobile Voice Connections
     */
    list(query = {}, options) {
        return this._client.getAPIList('/v2/mobile_voice_connections', (DefaultFlatPagination), { query, ...options });
    }
    /**
     * Delete a Mobile Voice Connection
     */
    delete(id, options) {
        return this._client.delete(path `/v2/mobile_voice_connections/${id}`, options);
    }
}
