// telnyx/resources/mobile-push-credentials.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.MobilePushCredentials = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const headers_1 = require("../internal/headers.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Mobile push credential management
 */
class MobilePushCredentials extends resource_1.APIResource {
    /**
     * Creates a new mobile push credential
     */
    create(params, options) {
        const { createMobilePushCredentialRequest } = params;
        return this._client.post('/mobile_push_credentials', {
            body: createMobilePushCredentialRequest,
            ...options,
        });
    }
    /**
     * Retrieves mobile push credential based on the given `push_credential_id`
     */
    retrieve(pushCredentialID, options) {
        return this._client.get((0, path_1.path) `/mobile_push_credentials/${pushCredentialID}`, options);
    }
    /**
     * List mobile push credentials
     */
    list(query = {}, options) {
        return this._client.getAPIList('/mobile_push_credentials', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Deletes a mobile push credential based on the given `push_credential_id`
     */
    delete(pushCredentialID, options) {
        return this._client.delete((0, path_1.path) `/mobile_push_credentials/${pushCredentialID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.MobilePushCredentials = MobilePushCredentials;
