// telnyx/resources/notification-profiles.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotificationProfiles = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Notification settings operations
 */
class NotificationProfiles extends resource_1.APIResource {
    /**
     * Create a notification profile.
     */
    create(body = {}, options) {
        return this._client.post('/notification_profiles', { body, ...options });
    }
    /**
     * Get a notification profile.
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/notification_profiles/${id}`, options);
    }
    /**
     * Update a notification profile.
     */
    update(notificationProfileID, body, options) {
        return this._client.patch((0, path_1.path) `/notification_profiles/${notificationProfileID}`, { body, ...options });
    }
    /**
     * Returns a list of your notifications profiles.
     */
    list(query = {}, options) {
        return this._client.getAPIList('/notification_profiles', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete a notification profile.
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/notification_profiles/${id}`, options);
    }
}
exports.NotificationProfiles = NotificationProfiles;
