// telnyx/resources/notification-channels.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { DefaultFlatPagination } from "../core/pagination.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Notification settings operations
 */
export class NotificationChannels extends APIResource {
    /**
     * Create a notification channel.
     *
     * @example
     * ```ts
     * const notificationChannel =
     *   await client.notificationChannels.create();
     * ```
     */
    create(body = {}, options) {
        return this._client.post('/notification_channels', { body, ...options });
    }
    /**
     * Get a notification channel.
     *
     * @example
     * ```ts
     * const notificationChannel =
     *   await client.notificationChannels.retrieve(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/notification_channels/${id}`, options);
    }
    /**
     * Update a notification channel.
     *
     * @example
     * ```ts
     * const notificationChannel =
     *   await client.notificationChannels.update(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    update(notificationChannelID, body, options) {
        return this._client.patch(path `/notification_channels/${notificationChannelID}`, { body, ...options });
    }
    /**
     * List notification channels.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const notificationChannel of client.notificationChannels.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/notification_channels', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete a notification channel.
     *
     * @example
     * ```ts
     * const notificationChannel =
     *   await client.notificationChannels.delete(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/notification_channels/${id}`, options);
    }
}
