// telnyx/resources/notification-channels.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotificationChannels = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Notification settings operations
 */
class NotificationChannels extends resource_1.APIResource {
    /**
     * Create a notification channel.
     *
     * @example
     * ```ts
     * const notificationChannel =
     *   await client.notificationChannels.create();
     * ```
     */
    create(body = {}, options) {
        return this._client.post('/notification_channels', { body, ...options });
    }
    /**
     * Get a notification channel.
     *
     * @example
     * ```ts
     * const notificationChannel =
     *   await client.notificationChannels.retrieve(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/notification_channels/${id}`, options);
    }
    /**
     * Update a notification channel.
     *
     * @example
     * ```ts
     * const notificationChannel =
     *   await client.notificationChannels.update(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    update(notificationChannelID, body, options) {
        return this._client.patch((0, path_1.path) `/notification_channels/${notificationChannelID}`, { body, ...options });
    }
    /**
     * List notification channels.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const notificationChannel of client.notificationChannels.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/notification_channels', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete a notification channel.
     *
     * @example
     * ```ts
     * const notificationChannel =
     *   await client.notificationChannels.delete(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/notification_channels/${id}`, options);
    }
}
exports.NotificationChannels = NotificationChannels;
