// telnyx/resources/notification-profiles.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { DefaultFlatPagination } from "../core/pagination.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Notification settings operations
 */
export class NotificationProfiles extends APIResource {
    /**
     * Create a notification profile.
     */
    create(body = {}, options) {
        return this._client.post('/notification_profiles', { body, ...options });
    }
    /**
     * Get a notification profile.
     */
    retrieve(id, options) {
        return this._client.get(path `/notification_profiles/${id}`, options);
    }
    /**
     * Update a notification profile.
     */
    update(notificationProfileID, body, options) {
        return this._client.patch(path `/notification_profiles/${notificationProfileID}`, { body, ...options });
    }
    /**
     * Returns a list of your notifications profiles.
     */
    list(query = {}, options) {
        return this._client.getAPIList('/notification_profiles', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete a notification profile.
     */
    delete(id, options) {
        return this._client.delete(path `/notification_profiles/${id}`, options);
    }
}
