// telnyx/resources/oauth-clients.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.OAuthClients = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const headers_1 = require("../internal/headers.js");
const path_1 = require("../internal/utils/path.js");
class OAuthClients extends resource_1.APIResource {
    /**
     * Create a new OAuth client
     *
     * @example
     * ```ts
     * const oauthClient = await client.oauthClients.create({
     *   allowed_grant_types: ['client_credentials'],
     *   allowed_scopes: ['admin'],
     *   client_type: 'public',
     *   name: 'My OAuth client',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/oauth_clients', { body, ...options });
    }
    /**
     * Retrieve a single OAuth client by ID
     *
     * @example
     * ```ts
     * const oauthClient = await client.oauthClients.retrieve(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/oauth_clients/${id}`, options);
    }
    /**
     * Update an existing OAuth client
     *
     * @example
     * ```ts
     * const oauthClient = await client.oauthClients.update(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    update(id, body, options) {
        return this._client.put((0, path_1.path) `/oauth_clients/${id}`, { body, ...options });
    }
    /**
     * Retrieve a paginated list of OAuth clients for the authenticated user
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const oauthClient of client.oauthClients.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/oauth_clients', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete an OAuth client
     *
     * @example
     * ```ts
     * await client.oauthClients.delete(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/oauth_clients/${id}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.OAuthClients = OAuthClients;
