// telnyx/resources/oauth.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.OAuth = void 0;
const resource_1 = require("../core/resource.js");
const headers_1 = require("../internal/headers.js");
const path_1 = require("../internal/utils/path.js");
class OAuth extends resource_1.APIResource {
    /**
     * Retrieve details about an OAuth consent token
     *
     * @example
     * ```ts
     * const oauth = await client.oauth.retrieve('consent_token');
     * ```
     */
    retrieve(consentToken, options) {
        return this._client.get((0, path_1.path) `/oauth/consent/${consentToken}`, options);
    }
    /**
     * Create an OAuth authorization grant
     *
     * @example
     * ```ts
     * const response = await client.oauth.grants({
     *   allowed: true,
     *   consent_token: 'consent_token',
     * });
     * ```
     */
    grants(body, options) {
        return this._client.post('/oauth/grants', { body, ...options });
    }
    /**
     * Introspect an OAuth access token to check its validity and metadata
     *
     * @example
     * ```ts
     * const response = await client.oauth.introspect({
     *   token: 'token',
     * });
     * ```
     */
    introspect(body, options) {
        return this._client.post('/oauth/introspect', {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ 'Content-Type': 'application/x-www-form-urlencoded' }, options?.headers]),
        });
    }
    /**
     * Register a new OAuth client dynamically (RFC 7591)
     *
     * @example
     * ```ts
     * const response = await client.oauth.register();
     * ```
     */
    register(body, options) {
        return this._client.post('/oauth/register', { body, ...options });
    }
    /**
     * OAuth 2.0 authorization endpoint for the authorization code flow
     *
     * @example
     * ```ts
     * await client.oauth.retrieveAuthorize({
     *   client_id: 'client_id',
     *   redirect_uri: 'https://example.com',
     *   response_type: 'code',
     * });
     * ```
     */
    retrieveAuthorize(query, options) {
        return this._client.get('/oauth/authorize', {
            query,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Retrieve the JSON Web Key Set for token verification
     *
     * @example
     * ```ts
     * const response = await client.oauth.retrieveJwks();
     * ```
     */
    retrieveJwks(options) {
        return this._client.get('/oauth/jwks', options);
    }
    /**
     * Exchange authorization code, client credentials, or refresh token for access
     * token
     *
     * @example
     * ```ts
     * const response = await client.oauth.token({
     *   grant_type: 'client_credentials',
     * });
     * ```
     */
    token(body, options) {
        return this._client.post('/oauth/token', {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ 'Content-Type': 'application/x-www-form-urlencoded' }, options?.headers]),
        });
    }
}
exports.OAuth = OAuth;
