// telnyx/resources/outbound-voice-profiles.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { DefaultFlatPagination } from "../core/pagination.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Outbound voice profiles operations
 */
export class OutboundVoiceProfiles extends APIResource {
    /**
     * Create an outbound voice profile.
     *
     * @example
     * ```ts
     * const outboundVoiceProfile =
     *   await client.outboundVoiceProfiles.create({
     *     name: 'office',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/outbound_voice_profiles', { body, ...options });
    }
    /**
     * Retrieves the details of an existing outbound voice profile.
     *
     * @example
     * ```ts
     * const outboundVoiceProfile =
     *   await client.outboundVoiceProfiles.retrieve(
     *     '1293384261075731499',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/outbound_voice_profiles/${id}`, options);
    }
    /**
     * Updates an existing outbound voice profile.
     *
     * @example
     * ```ts
     * const outboundVoiceProfile =
     *   await client.outboundVoiceProfiles.update(
     *     '1293384261075731499',
     *     { name: 'office' },
     *   );
     * ```
     */
    update(id, body, options) {
        return this._client.patch(path `/outbound_voice_profiles/${id}`, { body, ...options });
    }
    /**
     * Get all outbound voice profiles belonging to the user that match the given
     * filters.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const outboundVoiceProfile of client.outboundVoiceProfiles.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/outbound_voice_profiles', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Deletes an existing outbound voice profile.
     *
     * @example
     * ```ts
     * const outboundVoiceProfile =
     *   await client.outboundVoiceProfiles.delete(
     *     '1293384261075731499',
     *   );
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/outbound_voice_profiles/${id}`, options);
    }
}
