// telnyx/resources/outbound-voice-profiles.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.OutboundVoiceProfiles = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Outbound voice profiles operations
 */
class OutboundVoiceProfiles extends resource_1.APIResource {
    /**
     * Create an outbound voice profile.
     *
     * @example
     * ```ts
     * const outboundVoiceProfile =
     *   await client.outboundVoiceProfiles.create({
     *     name: 'office',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/outbound_voice_profiles', { body, ...options });
    }
    /**
     * Retrieves the details of an existing outbound voice profile.
     *
     * @example
     * ```ts
     * const outboundVoiceProfile =
     *   await client.outboundVoiceProfiles.retrieve(
     *     '1293384261075731499',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/outbound_voice_profiles/${id}`, options);
    }
    /**
     * Updates an existing outbound voice profile.
     *
     * @example
     * ```ts
     * const outboundVoiceProfile =
     *   await client.outboundVoiceProfiles.update(
     *     '1293384261075731499',
     *     { name: 'office' },
     *   );
     * ```
     */
    update(id, body, options) {
        return this._client.patch((0, path_1.path) `/outbound_voice_profiles/${id}`, { body, ...options });
    }
    /**
     * Get all outbound voice profiles belonging to the user that match the given
     * filters.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const outboundVoiceProfile of client.outboundVoiceProfiles.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/outbound_voice_profiles', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Deletes an existing outbound voice profile.
     *
     * @example
     * ```ts
     * const outboundVoiceProfile =
     *   await client.outboundVoiceProfiles.delete(
     *     '1293384261075731499',
     *   );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/outbound_voice_profiles/${id}`, options);
    }
}
exports.OutboundVoiceProfiles = OutboundVoiceProfiles;
