// telnyx/resources/pronunciation-dicts.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.PronunciationDicts = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const headers_1 = require("../internal/headers.js");
const uploads_1 = require("../internal/uploads.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Manage pronunciation dictionaries for text-to-speech synthesis. Dictionaries contain alias items (text replacement) and phoneme items (IPA pronunciation notation) that control how specific words are spoken.
 */
class PronunciationDicts extends resource_1.APIResource {
    /**
     * Create a new pronunciation dictionary for the authenticated organization. Each
     * dictionary contains a list of items that control how specific words are spoken.
     * Items can be alias type (text replacement) or phoneme type (IPA pronunciation
     * notation).
     *
     * As an alternative to providing items directly as JSON, you can upload a
     * dictionary file (PLS/XML or plain text format, max 1MB) using
     * multipart/form-data. PLS files use the standard W3C Pronunciation Lexicon
     * Specification XML format. Text files use a line-based format: `word=alias` for
     * aliases, `word:/phoneme/` for IPA phonemes.
     *
     * Limits:
     *
     * - Maximum 50 dictionaries per organization
     * - Maximum 100 items per dictionary
     * - Text: max 200 characters
     * - Alias/phoneme value: max 500 characters
     * - File upload: max 1MB (1,048,576 bytes)
     *
     * @example
     * ```ts
     * const pronunciationDict =
     *   await client.pronunciationDicts.create({
     *     items: [
     *       {
     *         alias: 'tel-nicks',
     *         text: 'Telnyx',
     *         type: 'alias',
     *       },
     *     ],
     *     name: 'Brand Names',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/pronunciation_dicts', (0, uploads_1.maybeMultipartFormRequestOptions)({ body, ...options }, this._client));
    }
    /**
     * Retrieve a single pronunciation dictionary by ID.
     *
     * @example
     * ```ts
     * const pronunciationDict =
     *   await client.pronunciationDicts.retrieve(
     *     'c215a3e1-be41-4701-97e8-1d3c22f9a5b7',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/pronunciation_dicts/${id}`, options);
    }
    /**
     * Update the name and/or items of an existing pronunciation dictionary. Uses
     * optimistic locking — if the dictionary was modified concurrently, the request
     * returns 409 Conflict.
     *
     * @example
     * ```ts
     * const pronunciationDict =
     *   await client.pronunciationDicts.update(
     *     'c215a3e1-be41-4701-97e8-1d3c22f9a5b7',
     *   );
     * ```
     */
    update(id, body, options) {
        return this._client.patch((0, path_1.path) `/pronunciation_dicts/${id}`, { body, ...options });
    }
    /**
     * List all pronunciation dictionaries for the authenticated organization. Results
     * are paginated using offset-based pagination.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const pronunciationDictData of client.pronunciationDicts.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/pronunciation_dicts', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Permanently delete a pronunciation dictionary.
     *
     * @example
     * ```ts
     * await client.pronunciationDicts.delete(
     *   'c215a3e1-be41-4701-97e8-1d3c22f9a5b7',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/pronunciation_dicts/${id}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.PronunciationDicts = PronunciationDicts;
