// telnyx/resources/pronunciation-dicts.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { DefaultFlatPagination } from "../core/pagination.mjs";
import { buildHeaders } from "../internal/headers.mjs";
import { maybeMultipartFormRequestOptions } from "../internal/uploads.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Manage pronunciation dictionaries for text-to-speech synthesis. Dictionaries contain alias items (text replacement) and phoneme items (IPA pronunciation notation) that control how specific words are spoken.
 */
export class PronunciationDicts extends APIResource {
    /**
     * Create a new pronunciation dictionary for the authenticated organization. Each
     * dictionary contains a list of items that control how specific words are spoken.
     * Items can be alias type (text replacement) or phoneme type (IPA pronunciation
     * notation).
     *
     * As an alternative to providing items directly as JSON, you can upload a
     * dictionary file (PLS/XML or plain text format, max 1MB) using
     * multipart/form-data. PLS files use the standard W3C Pronunciation Lexicon
     * Specification XML format. Text files use a line-based format: `word=alias` for
     * aliases, `word:/phoneme/` for IPA phonemes.
     *
     * Limits:
     *
     * - Maximum 50 dictionaries per organization
     * - Maximum 100 items per dictionary
     * - Text: max 200 characters
     * - Alias/phoneme value: max 500 characters
     * - File upload: max 1MB (1,048,576 bytes)
     *
     * @example
     * ```ts
     * const pronunciationDict =
     *   await client.pronunciationDicts.create({
     *     items: [
     *       {
     *         alias: 'tel-nicks',
     *         text: 'Telnyx',
     *         type: 'alias',
     *       },
     *     ],
     *     name: 'Brand Names',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/pronunciation_dicts', maybeMultipartFormRequestOptions({ body, ...options }, this._client));
    }
    /**
     * Retrieve a single pronunciation dictionary by ID.
     *
     * @example
     * ```ts
     * const pronunciationDict =
     *   await client.pronunciationDicts.retrieve(
     *     'c215a3e1-be41-4701-97e8-1d3c22f9a5b7',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/pronunciation_dicts/${id}`, options);
    }
    /**
     * Update the name and/or items of an existing pronunciation dictionary. Uses
     * optimistic locking — if the dictionary was modified concurrently, the request
     * returns 409 Conflict.
     *
     * @example
     * ```ts
     * const pronunciationDict =
     *   await client.pronunciationDicts.update(
     *     'c215a3e1-be41-4701-97e8-1d3c22f9a5b7',
     *   );
     * ```
     */
    update(id, body, options) {
        return this._client.patch(path `/pronunciation_dicts/${id}`, { body, ...options });
    }
    /**
     * List all pronunciation dictionaries for the authenticated organization. Results
     * are paginated using offset-based pagination.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const pronunciationDictData of client.pronunciationDicts.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/pronunciation_dicts', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Permanently delete a pronunciation dictionary.
     *
     * @example
     * ```ts
     * await client.pronunciationDicts.delete(
     *   'c215a3e1-be41-4701-97e8-1d3c22f9a5b7',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/pronunciation_dicts/${id}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
