// telnyx/resources/requirement-groups.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Requirement Groups
 */
export class RequirementGroups extends APIResource {
    /**
     * Create a new requirement group
     *
     * @example
     * ```ts
     * const requirementGroup =
     *   await client.requirementGroups.create({
     *     action: 'ordering',
     *     country_code: 'US',
     *     phone_number_type: 'local',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/requirement_groups', { body, ...options });
    }
    /**
     * Get a single requirement group by ID
     *
     * @example
     * ```ts
     * const requirementGroup =
     *   await client.requirementGroups.retrieve('id');
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/requirement_groups/${id}`, options);
    }
    /**
     * Update requirement values in requirement group
     *
     * @example
     * ```ts
     * const requirementGroup =
     *   await client.requirementGroups.update('id');
     * ```
     */
    update(id, body, options) {
        return this._client.patch(path `/requirement_groups/${id}`, { body, ...options });
    }
    /**
     * List requirement groups
     *
     * @example
     * ```ts
     * const requirementGroups =
     *   await client.requirementGroups.list();
     * ```
     */
    list(query = {}, options) {
        return this._client.get('/requirement_groups', { query, ...options });
    }
    /**
     * Delete a requirement group by ID
     *
     * @example
     * ```ts
     * const requirementGroup =
     *   await client.requirementGroups.delete('id');
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/requirement_groups/${id}`, options);
    }
    /**
     * Submit a Requirement Group for Approval
     *
     * @example
     * ```ts
     * const requirementGroup =
     *   await client.requirementGroups.submitForApproval('id');
     * ```
     */
    submitForApproval(id, options) {
        return this._client.post(path `/requirement_groups/${id}/submit_for_approval`, options);
    }
}
