// telnyx/resources/requirement-groups.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.RequirementGroups = void 0;
const resource_1 = require("../core/resource.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Requirement Groups
 */
class RequirementGroups extends resource_1.APIResource {
    /**
     * Create a new requirement group
     *
     * @example
     * ```ts
     * const requirementGroup =
     *   await client.requirementGroups.create({
     *     action: 'ordering',
     *     country_code: 'US',
     *     phone_number_type: 'local',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/requirement_groups', { body, ...options });
    }
    /**
     * Get a single requirement group by ID
     *
     * @example
     * ```ts
     * const requirementGroup =
     *   await client.requirementGroups.retrieve('id');
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/requirement_groups/${id}`, options);
    }
    /**
     * Update requirement values in requirement group
     *
     * @example
     * ```ts
     * const requirementGroup =
     *   await client.requirementGroups.update('id');
     * ```
     */
    update(id, body, options) {
        return this._client.patch((0, path_1.path) `/requirement_groups/${id}`, { body, ...options });
    }
    /**
     * List requirement groups
     *
     * @example
     * ```ts
     * const requirementGroups =
     *   await client.requirementGroups.list();
     * ```
     */
    list(query = {}, options) {
        return this._client.get('/requirement_groups', { query, ...options });
    }
    /**
     * Delete a requirement group by ID
     *
     * @example
     * ```ts
     * const requirementGroup =
     *   await client.requirementGroups.delete('id');
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/requirement_groups/${id}`, options);
    }
    /**
     * Submit a Requirement Group for Approval
     *
     * @example
     * ```ts
     * const requirementGroup =
     *   await client.requirementGroups.submitForApproval('id');
     * ```
     */
    submitForApproval(id, options) {
        return this._client.post((0, path_1.path) `/requirement_groups/${id}/submit_for_approval`, options);
    }
}
exports.RequirementGroups = RequirementGroups;
