// telnyx/resources/room-recordings.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoomRecordings = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const headers_1 = require("../internal/headers.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Rooms Recordings operations.
 */
class RoomRecordings extends resource_1.APIResource {
    /**
     * View a room recording.
     */
    retrieve(roomRecordingID, options) {
        return this._client.get((0, path_1.path) `/room_recordings/${roomRecordingID}`, options);
    }
    /**
     * View a list of room recordings.
     */
    list(query = {}, options) {
        return this._client.getAPIList('/room_recordings', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Synchronously delete a Room Recording.
     */
    delete(roomRecordingID, options) {
        return this._client.delete((0, path_1.path) `/room_recordings/${roomRecordingID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Delete several room recordings in a bulk.
     */
    deleteBulk(params = {}, options) {
        const { filter, 'page[number]': pageNumber, 'page[size]': pageSize } = params ?? {};
        return this._client.delete('/room_recordings', {
            query: { filter, 'page[number]': pageNumber, 'page[size]': pageSize },
            ...options,
        });
    }
}
exports.RoomRecordings = RoomRecordings;
