// telnyx/resources/room-compositions.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { DefaultFlatPagination } from "../core/pagination.mjs";
import { buildHeaders } from "../internal/headers.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Rooms Compositions operations.
 */
export class RoomCompositions extends APIResource {
    /**
     * Asynchronously create a room composition.
     *
     * @example
     * ```ts
     * const roomComposition =
     *   await client.roomCompositions.create();
     * ```
     */
    create(body, options) {
        return this._client.post('/room_compositions', { body, ...options });
    }
    /**
     * View a room composition.
     *
     * @example
     * ```ts
     * const roomComposition =
     *   await client.roomCompositions.retrieve(
     *     '5219b3af-87c6-4c08-9b58-5a533d893e21',
     *   );
     * ```
     */
    retrieve(roomCompositionID, options) {
        return this._client.get(path `/room_compositions/${roomCompositionID}`, options);
    }
    /**
     * View a list of room compositions.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const roomComposition of client.roomCompositions.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/room_compositions', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Synchronously delete a room composition.
     *
     * @example
     * ```ts
     * await client.roomCompositions.delete(
     *   '5219b3af-87c6-4c08-9b58-5a533d893e21',
     * );
     * ```
     */
    delete(roomCompositionID, options) {
        return this._client.delete(path `/room_compositions/${roomCompositionID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
