// telnyx/resources/room-compositions.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoomCompositions = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const headers_1 = require("../internal/headers.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Rooms Compositions operations.
 */
class RoomCompositions extends resource_1.APIResource {
    /**
     * Asynchronously create a room composition.
     *
     * @example
     * ```ts
     * const roomComposition =
     *   await client.roomCompositions.create();
     * ```
     */
    create(body, options) {
        return this._client.post('/room_compositions', { body, ...options });
    }
    /**
     * View a room composition.
     *
     * @example
     * ```ts
     * const roomComposition =
     *   await client.roomCompositions.retrieve(
     *     '5219b3af-87c6-4c08-9b58-5a533d893e21',
     *   );
     * ```
     */
    retrieve(roomCompositionID, options) {
        return this._client.get((0, path_1.path) `/room_compositions/${roomCompositionID}`, options);
    }
    /**
     * View a list of room compositions.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const roomComposition of client.roomCompositions.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/room_compositions', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Synchronously delete a room composition.
     *
     * @example
     * ```ts
     * await client.roomCompositions.delete(
     *   '5219b3af-87c6-4c08-9b58-5a533d893e21',
     * );
     * ```
     */
    delete(roomCompositionID, options) {
        return this._client.delete((0, path_1.path) `/room_compositions/${roomCompositionID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.RoomCompositions = RoomCompositions;
