// telnyx/resources/room-recordings.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { DefaultFlatPagination } from "../core/pagination.mjs";
import { buildHeaders } from "../internal/headers.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Rooms Recordings operations.
 */
export class RoomRecordings extends APIResource {
    /**
     * View a room recording.
     */
    retrieve(roomRecordingID, options) {
        return this._client.get(path `/room_recordings/${roomRecordingID}`, options);
    }
    /**
     * View a list of room recordings.
     */
    list(query = {}, options) {
        return this._client.getAPIList('/room_recordings', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Synchronously delete a Room Recording.
     */
    delete(roomRecordingID, options) {
        return this._client.delete(path `/room_recordings/${roomRecordingID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Delete several room recordings in a bulk.
     */
    deleteBulk(params = {}, options) {
        const { filter, 'page[number]': pageNumber, 'page[size]': pageSize } = params ?? {};
        return this._client.delete('/room_recordings', {
            query: { filter, 'page[number]': pageNumber, 'page[size]': pageSize },
            ...options,
        });
    }
}
