// telnyx/resources/sim-card-data-usage-notifications.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.SimCardDataUsageNotifications = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const path_1 = require("../internal/utils/path.js");
/**
 * SIM Cards operations
 */
class SimCardDataUsageNotifications extends resource_1.APIResource {
    /**
     * Creates a new SIM card data usage notification.
     *
     * @example
     * ```ts
     * const simCardDataUsageNotification =
     *   await client.simCardDataUsageNotifications.create({
     *     sim_card_id: '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *     threshold: {},
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/sim_card_data_usage_notifications', { body, ...options });
    }
    /**
     * Get a single SIM Card Data Usage Notification.
     *
     * @example
     * ```ts
     * const simCardDataUsageNotification =
     *   await client.simCardDataUsageNotifications.retrieve(
     *     '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/sim_card_data_usage_notifications/${id}`, options);
    }
    /**
     * Updates information for a SIM Card Data Usage Notification.
     *
     * @example
     * ```ts
     * const simCardDataUsageNotification =
     *   await client.simCardDataUsageNotifications.update(
     *     '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *   );
     * ```
     */
    update(simCardDataUsageNotificationID, body, options) {
        return this._client.patch((0, path_1.path) `/sim_card_data_usage_notifications/${simCardDataUsageNotificationID}`, {
            body,
            ...options,
        });
    }
    /**
     * Lists a paginated collection of SIM card data usage notifications. It enables
     * exploring the collection using specific filters.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const simCardDataUsageNotification of client.simCardDataUsageNotifications.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/sim_card_data_usage_notifications', (pagination_1.DefaultFlatPagination), { query, ...options });
    }
    /**
     * Delete the SIM Card Data Usage Notification.
     *
     * @example
     * ```ts
     * const simCardDataUsageNotification =
     *   await client.simCardDataUsageNotifications.delete(
     *     '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *   );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/sim_card_data_usage_notifications/${id}`, options);
    }
}
exports.SimCardDataUsageNotifications = SimCardDataUsageNotifications;
