// telnyx/resources/siprec-connectors.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { buildHeaders } from "../internal/headers.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * SIPREC connectors configuration.
 */
export class SiprecConnectors extends APIResource {
    /**
     * Creates a new SIPREC connector configuration.
     *
     * @example
     * ```ts
     * const siprecConnector =
     *   await client.siprecConnectors.create({
     *     host: 'siprec.telnyx.com',
     *     name: 'my-siprec-connector',
     *     port: 5060,
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/siprec_connectors', { body, ...options });
    }
    /**
     * Returns details of a stored SIPREC connector.
     *
     * @example
     * ```ts
     * const siprecConnector =
     *   await client.siprecConnectors.retrieve('connector_name');
     * ```
     */
    retrieve(connectorName, options) {
        return this._client.get(path `/siprec_connectors/${connectorName}`, options);
    }
    /**
     * Updates a stored SIPREC connector configuration.
     *
     * @example
     * ```ts
     * const siprecConnector =
     *   await client.siprecConnectors.update('connector_name', {
     *     host: 'siprec.telnyx.com',
     *     name: 'my-siprec-connector',
     *     port: 5060,
     *   });
     * ```
     */
    update(connectorName, body, options) {
        return this._client.put(path `/siprec_connectors/${connectorName}`, { body, ...options });
    }
    /**
     * Deletes a stored SIPREC connector.
     *
     * @example
     * ```ts
     * await client.siprecConnectors.delete('connector_name');
     * ```
     */
    delete(connectorName, options) {
        return this._client.delete(path `/siprec_connectors/${connectorName}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
