// telnyx/resources/siprec-connectors.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.SiprecConnectors = void 0;
const resource_1 = require("../core/resource.js");
const headers_1 = require("../internal/headers.js");
const path_1 = require("../internal/utils/path.js");
/**
 * SIPREC connectors configuration.
 */
class SiprecConnectors extends resource_1.APIResource {
    /**
     * Creates a new SIPREC connector configuration.
     *
     * @example
     * ```ts
     * const siprecConnector =
     *   await client.siprecConnectors.create({
     *     host: 'siprec.telnyx.com',
     *     name: 'my-siprec-connector',
     *     port: 5060,
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/siprec_connectors', { body, ...options });
    }
    /**
     * Returns details of a stored SIPREC connector.
     *
     * @example
     * ```ts
     * const siprecConnector =
     *   await client.siprecConnectors.retrieve('connector_name');
     * ```
     */
    retrieve(connectorName, options) {
        return this._client.get((0, path_1.path) `/siprec_connectors/${connectorName}`, options);
    }
    /**
     * Updates a stored SIPREC connector configuration.
     *
     * @example
     * ```ts
     * const siprecConnector =
     *   await client.siprecConnectors.update('connector_name', {
     *     host: 'siprec.telnyx.com',
     *     name: 'my-siprec-connector',
     *     port: 5060,
     *   });
     * ```
     */
    update(connectorName, body, options) {
        return this._client.put((0, path_1.path) `/siprec_connectors/${connectorName}`, { body, ...options });
    }
    /**
     * Deletes a stored SIPREC connector.
     *
     * @example
     * ```ts
     * await client.siprecConnectors.delete('connector_name');
     * ```
     */
    delete(connectorName, options) {
        return this._client.delete((0, path_1.path) `/siprec_connectors/${connectorName}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.SiprecConnectors = SiprecConnectors;
