// telnyx/resources/sub-number-orders.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { path } from "../internal/utils/path.mjs";
export class SubNumberOrders extends APIResource {
    /**
     * Get an existing sub number order.
     *
     * @example
     * ```ts
     * const subNumberOrder =
     *   await client.subNumberOrders.retrieve(
     *     'sub_number_order_id',
     *   );
     * ```
     */
    retrieve(subNumberOrderID, query = {}, options) {
        return this._client.get(path `/sub_number_orders/${subNumberOrderID}`, { query, ...options });
    }
    /**
     * Updates a sub number order.
     *
     * @example
     * ```ts
     * const subNumberOrder = await client.subNumberOrders.update(
     *   'sub_number_order_id',
     * );
     * ```
     */
    update(subNumberOrderID, body, options) {
        return this._client.patch(path `/sub_number_orders/${subNumberOrderID}`, { body, ...options });
    }
    /**
     * Get a paginated list of sub number orders.
     *
     * @example
     * ```ts
     * const subNumberOrders = await client.subNumberOrders.list();
     * ```
     */
    list(query = {}, options) {
        return this._client.get('/sub_number_orders', { query, ...options });
    }
    /**
     * Allows you to cancel a sub number order in 'pending' status.
     *
     * @example
     * ```ts
     * const response = await client.subNumberOrders.cancel(
     *   'sub_number_order_id',
     * );
     * ```
     */
    cancel(subNumberOrderID, options) {
        return this._client.patch(path `/sub_number_orders/${subNumberOrderID}/cancel`, options);
    }
    /**
     * Update requirement group for a sub number order
     *
     * @example
     * ```ts
     * const response =
     *   await client.subNumberOrders.updateRequirementGroup(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     {
     *       requirement_group_id:
     *         'a4b201f9-8646-4e54-a7d2-b2e403eeaf8c',
     *     },
     *   );
     * ```
     */
    updateRequirementGroup(id, body, options) {
        return this._client.post(path `/sub_number_orders/${id}/requirement_group`, { body, ...options });
    }
}
