// telnyx/resources/telephony-credentials.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.TelephonyCredentials = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const headers_1 = require("../internal/headers.js");
const path_1 = require("../internal/utils/path.js");
class TelephonyCredentials extends resource_1.APIResource {
    /**
     * Create a credential.
     *
     * @example
     * ```ts
     * const telephonyCredential =
     *   await client.telephonyCredentials.create({
     *     connection_id: '1234567890',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/telephony_credentials', { body, ...options });
    }
    /**
     * Get the details of an existing On-demand Credential.
     *
     * @example
     * ```ts
     * const telephonyCredential =
     *   await client.telephonyCredentials.retrieve('id');
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/telephony_credentials/${id}`, options);
    }
    /**
     * Update an existing credential.
     *
     * @example
     * ```ts
     * const telephonyCredential =
     *   await client.telephonyCredentials.update('id');
     * ```
     */
    update(id, body, options) {
        return this._client.patch((0, path_1.path) `/telephony_credentials/${id}`, { body, ...options });
    }
    /**
     * List all On-demand Credentials.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const telephonyCredential of client.telephonyCredentials.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/telephony_credentials', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete an existing credential.
     *
     * @example
     * ```ts
     * const telephonyCredential =
     *   await client.telephonyCredentials.delete('id');
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/telephony_credentials/${id}`, options);
    }
    /**
     * Create an Access Token (JWT) for the credential.
     *
     * @example
     * ```ts
     * const response =
     *   await client.telephonyCredentials.createToken('id');
     * ```
     */
    createToken(id, options) {
        return this._client.post((0, path_1.path) `/telephony_credentials/${id}/token`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: 'text/plain' }, options?.headers]),
        });
    }
}
exports.TelephonyCredentials = TelephonyCredentials;
