// telnyx/resources/telephony-credentials.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { DefaultFlatPagination } from "../core/pagination.mjs";
import { buildHeaders } from "../internal/headers.mjs";
import { path } from "../internal/utils/path.mjs";
export class TelephonyCredentials extends APIResource {
    /**
     * Create a credential.
     *
     * @example
     * ```ts
     * const telephonyCredential =
     *   await client.telephonyCredentials.create({
     *     connection_id: '1234567890',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/telephony_credentials', { body, ...options });
    }
    /**
     * Get the details of an existing On-demand Credential.
     *
     * @example
     * ```ts
     * const telephonyCredential =
     *   await client.telephonyCredentials.retrieve('id');
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/telephony_credentials/${id}`, options);
    }
    /**
     * Update an existing credential.
     *
     * @example
     * ```ts
     * const telephonyCredential =
     *   await client.telephonyCredentials.update('id');
     * ```
     */
    update(id, body, options) {
        return this._client.patch(path `/telephony_credentials/${id}`, { body, ...options });
    }
    /**
     * List all On-demand Credentials.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const telephonyCredential of client.telephonyCredentials.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/telephony_credentials', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete an existing credential.
     *
     * @example
     * ```ts
     * const telephonyCredential =
     *   await client.telephonyCredentials.delete('id');
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/telephony_credentials/${id}`, options);
    }
    /**
     * Create an Access Token (JWT) for the credential.
     *
     * @example
     * ```ts
     * const response =
     *   await client.telephonyCredentials.createToken('id');
     * ```
     */
    createToken(id, options) {
        return this._client.post(path `/telephony_credentials/${id}/token`, {
            ...options,
            headers: buildHeaders([{ Accept: 'text/plain' }, options?.headers]),
        });
    }
}
