// telnyx/resources/texml-applications.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { DefaultFlatPagination } from "../core/pagination.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * TeXML Applications operations
 */
export class TexmlApplications extends APIResource {
    /**
     * Creates a TeXML Application.
     *
     * @example
     * ```ts
     * const texmlApplication =
     *   await client.texmlApplications.create({
     *     friendly_name: 'call-router',
     *     voice_url: 'https://example.com',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/texml_applications', { body, ...options });
    }
    /**
     * Retrieves the details of an existing TeXML Application.
     *
     * @example
     * ```ts
     * const texmlApplication =
     *   await client.texmlApplications.retrieve(
     *     '1293384261075731499',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/texml_applications/${id}`, options);
    }
    /**
     * Updates settings of an existing TeXML Application.
     *
     * @example
     * ```ts
     * const texmlApplication =
     *   await client.texmlApplications.update(
     *     '1293384261075731499',
     *     {
     *       friendly_name: 'call-router',
     *       voice_url: 'https://example.com',
     *     },
     *   );
     * ```
     */
    update(id, body, options) {
        return this._client.patch(path `/texml_applications/${id}`, { body, ...options });
    }
    /**
     * Returns a list of your TeXML Applications.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const texmlApplication of client.texmlApplications.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/texml_applications', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Deletes a TeXML Application.
     *
     * @example
     * ```ts
     * const texmlApplication =
     *   await client.texmlApplications.delete(
     *     '1293384261075731499',
     *   );
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/texml_applications/${id}`, options);
    }
}
