// telnyx/resources/texml-applications.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.TexmlApplications = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const path_1 = require("../internal/utils/path.js");
/**
 * TeXML Applications operations
 */
class TexmlApplications extends resource_1.APIResource {
    /**
     * Creates a TeXML Application.
     *
     * @example
     * ```ts
     * const texmlApplication =
     *   await client.texmlApplications.create({
     *     friendly_name: 'call-router',
     *     voice_url: 'https://example.com',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/texml_applications', { body, ...options });
    }
    /**
     * Retrieves the details of an existing TeXML Application.
     *
     * @example
     * ```ts
     * const texmlApplication =
     *   await client.texmlApplications.retrieve(
     *     '1293384261075731499',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/texml_applications/${id}`, options);
    }
    /**
     * Updates settings of an existing TeXML Application.
     *
     * @example
     * ```ts
     * const texmlApplication =
     *   await client.texmlApplications.update(
     *     '1293384261075731499',
     *     {
     *       friendly_name: 'call-router',
     *       voice_url: 'https://example.com',
     *     },
     *   );
     * ```
     */
    update(id, body, options) {
        return this._client.patch((0, path_1.path) `/texml_applications/${id}`, { body, ...options });
    }
    /**
     * Returns a list of your TeXML Applications.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const texmlApplication of client.texmlApplications.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/texml_applications', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Deletes a TeXML Application.
     *
     * @example
     * ```ts
     * const texmlApplication =
     *   await client.texmlApplications.delete(
     *     '1293384261075731499',
     *   );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/texml_applications/${id}`, options);
    }
}
exports.TexmlApplications = TexmlApplications;
