// telnyx/resources/traffic-policy-profiles.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrafficPolicyProfiles = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Traffic Policy Profiles operations
 */
class TrafficPolicyProfiles extends resource_1.APIResource {
    /**
     * Create a new traffic policy profile. At least one of `services`, `ip_ranges`, or
     * `domains` must be provided.
     *
     * @example
     * ```ts
     * const trafficPolicyProfile =
     *   await client.trafficPolicyProfiles.create({
     *     type: 'whitelist',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/traffic_policy_profiles', { body, ...options });
    }
    /**
     * Returns the details regarding a specific traffic policy profile.
     *
     * @example
     * ```ts
     * const trafficPolicyProfile =
     *   await client.trafficPolicyProfiles.retrieve(
     *     '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/traffic_policy_profiles/${id}`, options);
    }
    /**
     * Updates a traffic policy profile.
     *
     * @example
     * ```ts
     * const trafficPolicyProfile =
     *   await client.trafficPolicyProfiles.update(
     *     '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *   );
     * ```
     */
    update(id, body, options) {
        return this._client.patch((0, path_1.path) `/traffic_policy_profiles/${id}`, { body, ...options });
    }
    /**
     * Get all traffic policy profiles belonging to the user that match the given
     * filters.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const trafficPolicyProfile of client.trafficPolicyProfiles.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/traffic_policy_profiles', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Deletes the traffic policy profile.
     *
     * @example
     * ```ts
     * const trafficPolicyProfile =
     *   await client.trafficPolicyProfiles.delete(
     *     '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *   );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/traffic_policy_profiles/${id}`, options);
    }
    /**
     * Get all available PCEF services that can be used in traffic policy profiles.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const trafficPolicyProfileListServicesResponse of client.trafficPolicyProfiles.listServices()) {
     *   // ...
     * }
     * ```
     */
    listServices(query = {}, options) {
        return this._client.getAPIList('/traffic_policy_profiles/services', (pagination_1.DefaultFlatPagination), { query, ...options });
    }
}
exports.TrafficPolicyProfiles = TrafficPolicyProfiles;
