// telnyx/resources/verify-profiles.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.VerifyProfiles = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Two factor authentication API
 */
class VerifyProfiles extends resource_1.APIResource {
    /**
     * Creates a new Verify profile to associate verifications with.
     *
     * @example
     * ```ts
     * const verifyProfileData =
     *   await client.verifyProfiles.create({
     *     name: 'Test Profile',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/verify_profiles', { body, ...options });
    }
    /**
     * Gets a single Verify profile.
     *
     * @example
     * ```ts
     * const verifyProfileData =
     *   await client.verifyProfiles.retrieve(
     *     '12ade33a-21c0-473b-b055-b3c836e1c292',
     *   );
     * ```
     */
    retrieve(verifyProfileID, options) {
        return this._client.get((0, path_1.path) `/verify_profiles/${verifyProfileID}`, options);
    }
    /**
     * Update Verify profile
     *
     * @example
     * ```ts
     * const verifyProfileData =
     *   await client.verifyProfiles.update(
     *     '12ade33a-21c0-473b-b055-b3c836e1c292',
     *   );
     * ```
     */
    update(verifyProfileID, body, options) {
        return this._client.patch((0, path_1.path) `/verify_profiles/${verifyProfileID}`, { body, ...options });
    }
    /**
     * Gets a paginated list of Verify profiles.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const verifyProfile of client.verifyProfiles.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/verify_profiles', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete Verify profile
     *
     * @example
     * ```ts
     * const verifyProfileData =
     *   await client.verifyProfiles.delete(
     *     '12ade33a-21c0-473b-b055-b3c836e1c292',
     *   );
     * ```
     */
    delete(verifyProfileID, options) {
        return this._client.delete((0, path_1.path) `/verify_profiles/${verifyProfileID}`, options);
    }
    /**
     * Create a new Verify profile message template.
     *
     * @example
     * ```ts
     * const messageTemplate =
     *   await client.verifyProfiles.createTemplate({
     *     text: 'Your {{app_name}} verification code is: {{code}}.',
     *   });
     * ```
     */
    createTemplate(body, options) {
        return this._client.post('/verify_profiles/templates', { body, ...options });
    }
    /**
     * List all Verify profile message templates.
     *
     * @example
     * ```ts
     * const response =
     *   await client.verifyProfiles.retrieveTemplates();
     * ```
     */
    retrieveTemplates(options) {
        return this._client.get('/verify_profiles/templates', options);
    }
    /**
     * Update an existing Verify profile message template.
     *
     * @example
     * ```ts
     * const messageTemplate =
     *   await client.verifyProfiles.updateTemplate(
     *     '12ade33a-21c0-473b-b055-b3c836e1c292',
     *     {
     *       text: 'Your {{app_name}} verification code is: {{code}}.',
     *     },
     *   );
     * ```
     */
    updateTemplate(templateID, body, options) {
        return this._client.patch((0, path_1.path) `/verify_profiles/templates/${templateID}`, { body, ...options });
    }
}
exports.VerifyProfiles = VerifyProfiles;
