// telnyx/resources/virtual-cross-connects.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualCrossConnects = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Virtual Cross Connect operations
 */
class VirtualCrossConnects extends resource_1.APIResource {
    /**
     * Create a new Virtual Cross Connect.<br /><br />For AWS and GCE, you have the
     * option of creating the primary connection first and the secondary connection
     * later. You also have the option of disabling the primary and/or secondary
     * connections at any time and later re-enabling them. With Azure, you do not have
     * this option. Azure requires both the primary and secondary connections to be
     * created at the same time and they can not be independantly disabled.
     *
     * @example
     * ```ts
     * const virtualCrossConnect =
     *   await client.virtualCrossConnects.create({
     *     region_code: 'ashburn-va',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/virtual_cross_connects', { body, ...options });
    }
    /**
     * Retrieve a Virtual Cross Connect.
     *
     * @example
     * ```ts
     * const virtualCrossConnect =
     *   await client.virtualCrossConnects.retrieve(
     *     '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/virtual_cross_connects/${id}`, options);
    }
    /**
     * Update the Virtual Cross Connect.<br /><br />Cloud IPs can only be patched
     * during the `created` state, as GCE will only inform you of your generated IP
     * once the pending connection requested has been accepted. Once the Virtual Cross
     * Connect has moved to `provisioning`, the IPs can no longer be
     * patched.<br /><br />Once the Virtual Cross Connect has moved to `provisioned`
     * and you are ready to enable routing, you can toggle the routing announcements to
     * `true`.
     *
     * @example
     * ```ts
     * const virtualCrossConnect =
     *   await client.virtualCrossConnects.update(
     *     '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *   );
     * ```
     */
    update(id, body, options) {
        return this._client.patch((0, path_1.path) `/virtual_cross_connects/${id}`, { body, ...options });
    }
    /**
     * List all Virtual Cross Connects.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const virtualCrossConnectCombined of client.virtualCrossConnects.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/virtual_cross_connects', (pagination_1.DefaultFlatPagination), { query, ...options });
    }
    /**
     * Delete a Virtual Cross Connect.
     *
     * @example
     * ```ts
     * const virtualCrossConnect =
     *   await client.virtualCrossConnects.delete(
     *     '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *   );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/virtual_cross_connects/${id}`, options);
    }
}
exports.VirtualCrossConnects = VirtualCrossConnects;
