// telnyx/resources/voice-designs.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.VoiceDesigns = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const headers_1 = require("../internal/headers.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Create and manage AI-generated voice designs using natural language prompts.
 */
class VoiceDesigns extends resource_1.APIResource {
    /**
     * Creates a new voice design (version 1) when `voice_design_id` is omitted. When
     * `voice_design_id` is provided, adds a new version to the existing design
     * instead. A design can have at most 50 versions.
     *
     * @example
     * ```ts
     * const voiceDesign = await client.voiceDesigns.create({
     *   prompt: 'Speak in a warm, friendly tone',
     *   text: 'Hello, welcome to our service.',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/voice_designs', { body, ...options });
    }
    /**
     * Returns the latest version of a voice design, or a specific version when
     * `?version=N` is provided. The `id` parameter accepts either a UUID or the design
     * name.
     *
     * @example
     * ```ts
     * const voiceDesign = await client.voiceDesigns.retrieve(
     *   'id',
     * );
     * ```
     */
    retrieve(id, query = {}, options) {
        return this._client.get((0, path_1.path) `/voice_designs/${id}`, { query, ...options });
    }
    /**
     * Returns a paginated list of voice designs belonging to the authenticated
     * account.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const voiceDesignListResponse of client.voiceDesigns.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/voice_designs', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Permanently deletes a voice design and all of its versions. This action cannot
     * be undone.
     *
     * @example
     * ```ts
     * await client.voiceDesigns.delete('id');
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/voice_designs/${id}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Permanently deletes a specific version of a voice design. The version number
     * must be a positive integer.
     *
     * @example
     * ```ts
     * await client.voiceDesigns.deleteVersion(1, { id: 'id' });
     * ```
     */
    deleteVersion(version, params, options) {
        const { id } = params;
        return this._client.delete((0, path_1.path) `/voice_designs/${id}/versions/${version}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Downloads the WAV audio sample for the voice design. Returns the latest
     * version's sample by default, or a specific version when `?version=N` is
     * provided. The `id` parameter accepts either a UUID or the design name.
     *
     * @example
     * ```ts
     * const response = await client.voiceDesigns.downloadSample(
     *   'id',
     * );
     *
     * const content = await response.blob();
     * console.log(content);
     * ```
     */
    downloadSample(id, query = {}, options) {
        return this._client.get((0, path_1.path) `/voice_designs/${id}/sample`, {
            query,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: 'audio/wav' }, options?.headers]),
            __binaryResponse: true,
        });
    }
    /**
     * Updates the name of a voice design. All versions retain their other properties.
     *
     * @example
     * ```ts
     * const response = await client.voiceDesigns.rename('id', {
     *   name: 'updated-narrator',
     * });
     * ```
     */
    rename(id, body, options) {
        return this._client.patch((0, path_1.path) `/voice_designs/${id}`, { body, ...options });
    }
}
exports.VoiceDesigns = VoiceDesigns;
