// telnyx/resources/voice-designs.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { DefaultFlatPagination } from "../core/pagination.mjs";
import { buildHeaders } from "../internal/headers.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Create and manage AI-generated voice designs using natural language prompts.
 */
export class VoiceDesigns extends APIResource {
    /**
     * Creates a new voice design (version 1) when `voice_design_id` is omitted. When
     * `voice_design_id` is provided, adds a new version to the existing design
     * instead. A design can have at most 50 versions.
     *
     * @example
     * ```ts
     * const voiceDesign = await client.voiceDesigns.create({
     *   prompt: 'Speak in a warm, friendly tone',
     *   text: 'Hello, welcome to our service.',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/voice_designs', { body, ...options });
    }
    /**
     * Returns the latest version of a voice design, or a specific version when
     * `?version=N` is provided. The `id` parameter accepts either a UUID or the design
     * name.
     *
     * @example
     * ```ts
     * const voiceDesign = await client.voiceDesigns.retrieve(
     *   'id',
     * );
     * ```
     */
    retrieve(id, query = {}, options) {
        return this._client.get(path `/voice_designs/${id}`, { query, ...options });
    }
    /**
     * Returns a paginated list of voice designs belonging to the authenticated
     * account.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const voiceDesignListResponse of client.voiceDesigns.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/voice_designs', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Permanently deletes a voice design and all of its versions. This action cannot
     * be undone.
     *
     * @example
     * ```ts
     * await client.voiceDesigns.delete('id');
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/voice_designs/${id}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Permanently deletes a specific version of a voice design. The version number
     * must be a positive integer.
     *
     * @example
     * ```ts
     * await client.voiceDesigns.deleteVersion(1, { id: 'id' });
     * ```
     */
    deleteVersion(version, params, options) {
        const { id } = params;
        return this._client.delete(path `/voice_designs/${id}/versions/${version}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Downloads the WAV audio sample for the voice design. Returns the latest
     * version's sample by default, or a specific version when `?version=N` is
     * provided. The `id` parameter accepts either a UUID or the design name.
     *
     * @example
     * ```ts
     * const response = await client.voiceDesigns.downloadSample(
     *   'id',
     * );
     *
     * const content = await response.blob();
     * console.log(content);
     * ```
     */
    downloadSample(id, query = {}, options) {
        return this._client.get(path `/voice_designs/${id}/sample`, {
            query,
            ...options,
            headers: buildHeaders([{ Accept: 'audio/wav' }, options?.headers]),
            __binaryResponse: true,
        });
    }
    /**
     * Updates the name of a voice design. All versions retain their other properties.
     *
     * @example
     * ```ts
     * const response = await client.voiceDesigns.rename('id', {
     *   name: 'updated-narrator',
     * });
     * ```
     */
    rename(id, body, options) {
        return this._client.patch(path `/voice_designs/${id}`, { body, ...options });
    }
}
