// telnyx/resources/voice-clones.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.VoiceClones = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const headers_1 = require("../internal/headers.js");
const uploads_1 = require("../internal/uploads.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Capture and manage voice identities as clones for use in text-to-speech synthesis.
 */
class VoiceClones extends resource_1.APIResource {
    /**
     * Creates a new voice clone by capturing the voice identity of an existing voice
     * design. The clone can then be used for text-to-speech synthesis.
     *
     * @example
     * ```ts
     * const voiceClone = await client.voiceClones.create({
     *   params: {
     *     gender: 'male',
     *     language: 'en',
     *     name: 'clone-narrator',
     *     voice_design_id: '550e8400-e29b-41d4-a716-446655440000',
     *     provider: 'telnyx',
     *   },
     * });
     * ```
     */
    create(args, options) {
        const { params } = args;
        return this._client.post('/voice_clones', { body: params, ...options });
    }
    /**
     * Updates the name, language, or gender of a voice clone.
     *
     * @example
     * ```ts
     * const voiceClone = await client.voiceClones.update(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   { name: 'updated-clone' },
     * );
     * ```
     */
    update(id, body, options) {
        return this._client.patch((0, path_1.path) `/voice_clones/${id}`, { body, ...options });
    }
    /**
     * Returns a paginated list of voice clones belonging to the authenticated account.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const voiceCloneData of client.voiceClones.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/voice_clones', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Permanently deletes a voice clone. This action cannot be undone.
     *
     * @example
     * ```ts
     * await client.voiceClones.delete(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/voice_clones/${id}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Creates a new voice clone by uploading an audio file directly. Supported
     * formats: WAV, MP3, FLAC, OGG, M4A. For best results, provide 5–10 seconds of
     * clear speech. Maximum file size: 5MB for Telnyx, 20MB for Minimax.
     *
     * @example
     * ```ts
     * const response = await client.voiceClones.createFromUpload({
     *   params: {
     *     audio_file: fs.createReadStream('path/to/file'),
     *     gender: 'male',
     *     language: 'lkf-Lz1vLbBu-9uDh-9AHaOS2D-Cbf',
     *     name: 'name',
     *     provider: 'telnyx',
     *   },
     * });
     * ```
     */
    createFromUpload(args, options) {
        const { params } = args;
        return this._client.post('/voice_clones/from_upload', (0, uploads_1.multipartFormRequestOptions)({ body: params, ...options }, this._client));
    }
    /**
     * Downloads the WAV audio sample that was used to create the voice clone.
     *
     * @example
     * ```ts
     * const response = await client.voiceClones.downloadSample(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     *
     * const content = await response.blob();
     * console.log(content);
     * ```
     */
    downloadSample(id, options) {
        return this._client.get((0, path_1.path) `/voice_clones/${id}/sample`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: 'audio/wav' }, options?.headers]),
            __binaryResponse: true,
        });
    }
}
exports.VoiceClones = VoiceClones;
