// telnyx/resources/wireguard-peers.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { DefaultFlatPagination } from "../core/pagination.mjs";
import { buildHeaders } from "../internal/headers.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * WireGuard Interface operations
 */
export class WireguardPeers extends APIResource {
    /**
     * Create a new WireGuard Peer. Current limitation of 5 peers per interface can be
     * created.
     *
     * @example
     * ```ts
     * const wireguardPeer = await client.wireguardPeers.create({
     *   wireguard_interface_id:
     *     '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/wireguard_peers', { body, ...options });
    }
    /**
     * Retrieve the WireGuard peer.
     *
     * @example
     * ```ts
     * const wireguardPeer = await client.wireguardPeers.retrieve(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/wireguard_peers/${id}`, options);
    }
    /**
     * Update the WireGuard peer.
     *
     * @example
     * ```ts
     * const wireguardPeer = await client.wireguardPeers.update(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    update(id, body, options) {
        return this._client.patch(path `/wireguard_peers/${id}`, { body, ...options });
    }
    /**
     * List all WireGuard peers.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const wireguardPeerListResponse of client.wireguardPeers.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/wireguard_peers', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete the WireGuard peer.
     *
     * @example
     * ```ts
     * const wireguardPeer = await client.wireguardPeers.delete(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/wireguard_peers/${id}`, options);
    }
    /**
     * Retrieve Wireguard config template for Peer
     *
     * @example
     * ```ts
     * const response = await client.wireguardPeers.retrieveConfig(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    retrieveConfig(id, options) {
        return this._client.get(path `/wireguard_peers/${id}/config`, {
            ...options,
            headers: buildHeaders([{ Accept: 'text/plain' }, options?.headers]),
        });
    }
}
