// telnyx/resources/wireguard-peers.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.WireguardPeers = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const headers_1 = require("../internal/headers.js");
const path_1 = require("../internal/utils/path.js");
/**
 * WireGuard Interface operations
 */
class WireguardPeers extends resource_1.APIResource {
    /**
     * Create a new WireGuard Peer. Current limitation of 5 peers per interface can be
     * created.
     *
     * @example
     * ```ts
     * const wireguardPeer = await client.wireguardPeers.create({
     *   wireguard_interface_id:
     *     '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/wireguard_peers', { body, ...options });
    }
    /**
     * Retrieve the WireGuard peer.
     *
     * @example
     * ```ts
     * const wireguardPeer = await client.wireguardPeers.retrieve(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/wireguard_peers/${id}`, options);
    }
    /**
     * Update the WireGuard peer.
     *
     * @example
     * ```ts
     * const wireguardPeer = await client.wireguardPeers.update(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    update(id, body, options) {
        return this._client.patch((0, path_1.path) `/wireguard_peers/${id}`, { body, ...options });
    }
    /**
     * List all WireGuard peers.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const wireguardPeerListResponse of client.wireguardPeers.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/wireguard_peers', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete the WireGuard peer.
     *
     * @example
     * ```ts
     * const wireguardPeer = await client.wireguardPeers.delete(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/wireguard_peers/${id}`, options);
    }
    /**
     * Retrieve Wireguard config template for Peer
     *
     * @example
     * ```ts
     * const response = await client.wireguardPeers.retrieveConfig(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    retrieveConfig(id, options) {
        return this._client.get((0, path_1.path) `/wireguard_peers/${id}/config`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: 'text/plain' }, options?.headers]),
        });
    }
}
exports.WireguardPeers = WireguardPeers;
