// telnyx/resources/wireless-blocklists.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.WirelessBlocklists = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Wireless Blocklists operations
 */
class WirelessBlocklists extends resource_1.APIResource {
    /**
     * Create a Wireless Blocklist to prevent SIMs from connecting to certain networks.
     *
     * @example
     * ```ts
     * const wirelessBlocklist =
     *   await client.wirelessBlocklists.create({
     *     name: 'My Wireless Blocklist',
     *     type: 'country',
     *     values: ['CA', 'US'],
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/wireless_blocklists', { body, ...options });
    }
    /**
     * Retrieve information about a Wireless Blocklist.
     *
     * @example
     * ```ts
     * const wirelessBlocklist =
     *   await client.wirelessBlocklists.retrieve(
     *     '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/wireless_blocklists/${id}`, options);
    }
    /**
     * Update a Wireless Blocklist.
     *
     * @example
     * ```ts
     * const wirelessBlocklist =
     *   await client.wirelessBlocklists.update(
     *     '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *   );
     * ```
     */
    update(id, body, options) {
        return this._client.patch((0, path_1.path) `/wireless_blocklists/${id}`, { body, ...options });
    }
    /**
     * Get all Wireless Blocklists belonging to the user.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const wirelessBlocklist of client.wirelessBlocklists.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/wireless_blocklists', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Deletes the Wireless Blocklist.
     *
     * @example
     * ```ts
     * const wirelessBlocklist =
     *   await client.wirelessBlocklists.delete(
     *     '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *   );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/wireless_blocklists/${id}`, options);
    }
}
exports.WirelessBlocklists = WirelessBlocklists;
