// telnyx/resources/rooms/rooms.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Rooms = void 0;
const tslib_1 = require("../../internal/tslib.js");
const resource_1 = require("../../core/resource.js");
const ActionsAPI = tslib_1.__importStar(require("./actions.js"));
const actions_1 = require("./actions.js");
const SessionsAPI = tslib_1.__importStar(require("./sessions/sessions.js"));
const sessions_1 = require("./sessions/sessions.js");
const pagination_1 = require("../../core/pagination.js");
const headers_1 = require("../../internal/headers.js");
const path_1 = require("../../internal/utils/path.js");
/**
 * Rooms operations.
 */
class Rooms extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.actions = new ActionsAPI.Actions(this._client);
        this.sessions = new SessionsAPI.Sessions(this._client);
    }
    /**
     * Synchronously create a Room.
     *
     * @example
     * ```ts
     * const room = await client.rooms.create();
     * ```
     */
    create(body, options) {
        return this._client.post('/rooms', { body, ...options });
    }
    /**
     * View a room.
     *
     * @example
     * ```ts
     * const room = await client.rooms.retrieve(
     *   '0ccc7b54-4df3-4bca-a65a-3da1ecc777f0',
     * );
     * ```
     */
    retrieve(roomID, query = {}, options) {
        return this._client.get((0, path_1.path) `/rooms/${roomID}`, { query, ...options });
    }
    /**
     * Synchronously update a Room.
     *
     * @example
     * ```ts
     * const room = await client.rooms.update(
     *   '0ccc7b54-4df3-4bca-a65a-3da1ecc777f0',
     * );
     * ```
     */
    update(roomID, body, options) {
        return this._client.patch((0, path_1.path) `/rooms/${roomID}`, { body, ...options });
    }
    /**
     * View a list of rooms.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const room of client.rooms.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/rooms', (pagination_1.DefaultFlatPagination), { query, ...options });
    }
    /**
     * Synchronously delete a Room. Participants from that room will be kicked out,
     * they won't be able to join that room anymore, and you won't be charged anymore
     * for that room.
     *
     * @example
     * ```ts
     * await client.rooms.delete(
     *   '0ccc7b54-4df3-4bca-a65a-3da1ecc777f0',
     * );
     * ```
     */
    delete(roomID, options) {
        return this._client.delete((0, path_1.path) `/rooms/${roomID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.Rooms = Rooms;
Rooms.Actions = actions_1.Actions;
Rooms.Sessions = sessions_1.Sessions;
