// telnyx/resources/rooms/rooms.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as ActionsAPI from "./actions.mjs";
import { Actions, } from "./actions.mjs";
import * as SessionsAPI from "./sessions/sessions.mjs";
import { Sessions, } from "./sessions/sessions.mjs";
import { DefaultFlatPagination } from "../../core/pagination.mjs";
import { buildHeaders } from "../../internal/headers.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * Rooms operations.
 */
export class Rooms extends APIResource {
    constructor() {
        super(...arguments);
        this.actions = new ActionsAPI.Actions(this._client);
        this.sessions = new SessionsAPI.Sessions(this._client);
    }
    /**
     * Synchronously create a Room.
     *
     * @example
     * ```ts
     * const room = await client.rooms.create();
     * ```
     */
    create(body, options) {
        return this._client.post('/rooms', { body, ...options });
    }
    /**
     * View a room.
     *
     * @example
     * ```ts
     * const room = await client.rooms.retrieve(
     *   '0ccc7b54-4df3-4bca-a65a-3da1ecc777f0',
     * );
     * ```
     */
    retrieve(roomID, query = {}, options) {
        return this._client.get(path `/rooms/${roomID}`, { query, ...options });
    }
    /**
     * Synchronously update a Room.
     *
     * @example
     * ```ts
     * const room = await client.rooms.update(
     *   '0ccc7b54-4df3-4bca-a65a-3da1ecc777f0',
     * );
     * ```
     */
    update(roomID, body, options) {
        return this._client.patch(path `/rooms/${roomID}`, { body, ...options });
    }
    /**
     * View a list of rooms.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const room of client.rooms.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/rooms', (DefaultFlatPagination), { query, ...options });
    }
    /**
     * Synchronously delete a Room. Participants from that room will be kicked out,
     * they won't be able to join that room anymore, and you won't be charged anymore
     * for that room.
     *
     * @example
     * ```ts
     * await client.rooms.delete(
     *   '0ccc7b54-4df3-4bca-a65a-3da1ecc777f0',
     * );
     * ```
     */
    delete(roomID, options) {
        return this._client.delete(path `/rooms/${roomID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
Rooms.Actions = Actions;
Rooms.Sessions = Sessions;
