// telnyx/resources/legacy/reporting/usage-reports/messaging.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Messaging = void 0;
const resource_1 = require("../../../../core/resource.js");
const pagination_1 = require("../../../../core/pagination.js");
const headers_1 = require("../../../../internal/headers.js");
const path_1 = require("../../../../internal/utils/path.js");
/**
 * Messaging usage reports
 */
class Messaging extends resource_1.APIResource {
    /**
     * Creates a new legacy usage V2 MDR report request with the specified filters
     *
     * @example
     * ```ts
     * const messaging =
     *   await client.legacy.reporting.usageReports.messaging.create(
     *     { aggregation_type: 0 },
     *   );
     * ```
     */
    create(body, options) {
        return this._client.post('/legacy/reporting/usage_reports/messaging', {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ 'Content-Type': '*/*' }, options?.headers]),
        });
    }
    /**
     * Fetch single MDR usage report by id.
     *
     * @example
     * ```ts
     * const messaging =
     *   await client.legacy.reporting.usageReports.messaging.retrieve(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/legacy/reporting/usage_reports/messaging/${id}`, options);
    }
    /**
     * Fetch all previous requests for MDR usage reports.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const mdrUsageReportResponseLegacy of client.legacy.reporting.usageReports.messaging.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/legacy/reporting/usage_reports/messaging', (pagination_1.PerPagePagination), { query, ...options });
    }
    /**
     * Deletes a specific V2 legacy usage MDR report request by ID
     *
     * @example
     * ```ts
     * const messaging =
     *   await client.legacy.reporting.usageReports.messaging.delete(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/legacy/reporting/usage_reports/messaging/${id}`, options);
    }
}
exports.Messaging = Messaging;
