// telnyx/resources/legacy/reporting/usage-reports/messaging.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../../core/resource.mjs";
import { PerPagePagination } from "../../../../core/pagination.mjs";
import { buildHeaders } from "../../../../internal/headers.mjs";
import { path } from "../../../../internal/utils/path.mjs";
/**
 * Messaging usage reports
 */
export class Messaging extends APIResource {
    /**
     * Creates a new legacy usage V2 MDR report request with the specified filters
     *
     * @example
     * ```ts
     * const messaging =
     *   await client.legacy.reporting.usageReports.messaging.create(
     *     { aggregation_type: 0 },
     *   );
     * ```
     */
    create(body, options) {
        return this._client.post('/legacy/reporting/usage_reports/messaging', {
            body,
            ...options,
            headers: buildHeaders([{ 'Content-Type': '*/*' }, options?.headers]),
        });
    }
    /**
     * Fetch single MDR usage report by id.
     *
     * @example
     * ```ts
     * const messaging =
     *   await client.legacy.reporting.usageReports.messaging.retrieve(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/legacy/reporting/usage_reports/messaging/${id}`, options);
    }
    /**
     * Fetch all previous requests for MDR usage reports.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const mdrUsageReportResponseLegacy of client.legacy.reporting.usageReports.messaging.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/legacy/reporting/usage_reports/messaging', (PerPagePagination), { query, ...options });
    }
    /**
     * Deletes a specific V2 legacy usage MDR report request by ID
     *
     * @example
     * ```ts
     * const messaging =
     *   await client.legacy.reporting.usageReports.messaging.delete(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/legacy/reporting/usage_reports/messaging/${id}`, options);
    }
}
