// telnyx/resources/legacy/reporting/usage-reports/number-lookup.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.NumberLookup = void 0;
const resource_1 = require("../../../../core/resource.js");
const headers_1 = require("../../../../internal/headers.js");
const path_1 = require("../../../../internal/utils/path.js");
/**
 * Number lookup usage reports
 */
class NumberLookup extends resource_1.APIResource {
    /**
     * Submit a new telco data usage report
     *
     * @example
     * ```ts
     * const numberLookup =
     *   await client.legacy.reporting.usageReports.numberLookup.create();
     * ```
     */
    create(body, options) {
        return this._client.post('/legacy/reporting/usage_reports/number_lookup', {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ 'Content-Type': '*/*' }, options?.headers]),
        });
    }
    /**
     * Retrieve a specific telco data usage report by its ID
     *
     * @example
     * ```ts
     * const numberLookup =
     *   await client.legacy.reporting.usageReports.numberLookup.retrieve(
     *     'id',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/legacy/reporting/usage_reports/number_lookup/${id}`, options);
    }
    /**
     * Retrieve a paginated list of telco data usage reports
     *
     * @example
     * ```ts
     * const numberLookups =
     *   await client.legacy.reporting.usageReports.numberLookup.list();
     * ```
     */
    list(options) {
        return this._client.get('/legacy/reporting/usage_reports/number_lookup', options);
    }
    /**
     * Delete a specific telco data usage report by its ID
     *
     * @example
     * ```ts
     * await client.legacy.reporting.usageReports.numberLookup.delete(
     *   'id',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/legacy/reporting/usage_reports/number_lookup/${id}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.NumberLookup = NumberLookup;
