// telnyx/resources/legacy/reporting/usage-reports/number-lookup.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../../core/resource.mjs";
import { buildHeaders } from "../../../../internal/headers.mjs";
import { path } from "../../../../internal/utils/path.mjs";
/**
 * Number lookup usage reports
 */
export class NumberLookup extends APIResource {
    /**
     * Submit a new telco data usage report
     *
     * @example
     * ```ts
     * const numberLookup =
     *   await client.legacy.reporting.usageReports.numberLookup.create();
     * ```
     */
    create(body, options) {
        return this._client.post('/legacy/reporting/usage_reports/number_lookup', {
            body,
            ...options,
            headers: buildHeaders([{ 'Content-Type': '*/*' }, options?.headers]),
        });
    }
    /**
     * Retrieve a specific telco data usage report by its ID
     *
     * @example
     * ```ts
     * const numberLookup =
     *   await client.legacy.reporting.usageReports.numberLookup.retrieve(
     *     'id',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/legacy/reporting/usage_reports/number_lookup/${id}`, options);
    }
    /**
     * Retrieve a paginated list of telco data usage reports
     *
     * @example
     * ```ts
     * const numberLookups =
     *   await client.legacy.reporting.usageReports.numberLookup.list();
     * ```
     */
    list(options) {
        return this._client.get('/legacy/reporting/usage_reports/number_lookup', options);
    }
    /**
     * Delete a specific telco data usage report by its ID
     *
     * @example
     * ```ts
     * await client.legacy.reporting.usageReports.numberLookup.delete(
     *   'id',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/legacy/reporting/usage_reports/number_lookup/${id}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
