// telnyx/resources/legacy/reporting/usage-reports/voice.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../../core/resource.mjs";
import { PerPagePagination } from "../../../../core/pagination.mjs";
import { buildHeaders } from "../../../../internal/headers.mjs";
import { path } from "../../../../internal/utils/path.mjs";
/**
 * Voice usage reports
 */
export class Voice extends APIResource {
    /**
     * Creates a new legacy usage V2 CDR report request with the specified filters
     *
     * @example
     * ```ts
     * const voice =
     *   await client.legacy.reporting.usageReports.voice.create({
     *     end_time: '2024-02-01T00:00:00Z',
     *     start_time: '2024-02-01T00:00:00Z',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/legacy/reporting/usage_reports/voice', {
            body,
            ...options,
            headers: buildHeaders([{ 'Content-Type': '*/*' }, options?.headers]),
        });
    }
    /**
     * Fetch single cdr usage report by id.
     *
     * @example
     * ```ts
     * const voice =
     *   await client.legacy.reporting.usageReports.voice.retrieve(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/legacy/reporting/usage_reports/voice/${id}`, options);
    }
    /**
     * Fetch all previous requests for cdr usage reports.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const cdrUsageReportResponseLegacy of client.legacy.reporting.usageReports.voice.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/legacy/reporting/usage_reports/voice', (PerPagePagination), { query, ...options });
    }
    /**
     * Deletes a specific V2 legacy usage CDR report request by ID
     *
     * @example
     * ```ts
     * const voice =
     *   await client.legacy.reporting.usageReports.voice.delete(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/legacy/reporting/usage_reports/voice/${id}`, options);
    }
}
