// telnyx/resources/legacy/reporting/usage-reports/voice.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Voice = void 0;
const resource_1 = require("../../../../core/resource.js");
const pagination_1 = require("../../../../core/pagination.js");
const headers_1 = require("../../../../internal/headers.js");
const path_1 = require("../../../../internal/utils/path.js");
/**
 * Voice usage reports
 */
class Voice extends resource_1.APIResource {
    /**
     * Creates a new legacy usage V2 CDR report request with the specified filters
     *
     * @example
     * ```ts
     * const voice =
     *   await client.legacy.reporting.usageReports.voice.create({
     *     end_time: '2024-02-01T00:00:00Z',
     *     start_time: '2024-02-01T00:00:00Z',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/legacy/reporting/usage_reports/voice', {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ 'Content-Type': '*/*' }, options?.headers]),
        });
    }
    /**
     * Fetch single cdr usage report by id.
     *
     * @example
     * ```ts
     * const voice =
     *   await client.legacy.reporting.usageReports.voice.retrieve(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/legacy/reporting/usage_reports/voice/${id}`, options);
    }
    /**
     * Fetch all previous requests for cdr usage reports.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const cdrUsageReportResponseLegacy of client.legacy.reporting.usageReports.voice.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/legacy/reporting/usage_reports/voice', (pagination_1.PerPagePagination), { query, ...options });
    }
    /**
     * Deletes a specific V2 legacy usage CDR report request by ID
     *
     * @example
     * ```ts
     * const voice =
     *   await client.legacy.reporting.usageReports.voice.delete(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/legacy/reporting/usage_reports/voice/${id}`, options);
    }
}
exports.Voice = Voice;
