// telnyx/resources/texml/accounts/conferences/conferences.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Conferences = void 0;
const tslib_1 = require("../../../../internal/tslib.js");
const resource_1 = require("../../../../core/resource.js");
const ParticipantsAPI = tslib_1.__importStar(require("./participants.js"));
const participants_1 = require("./participants.js");
const headers_1 = require("../../../../internal/headers.js");
const path_1 = require("../../../../internal/utils/path.js");
/**
 * TeXML REST Commands
 */
class Conferences extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.participants = new ParticipantsAPI.Participants(this._client);
    }
    /**
     * Returns a conference resource.
     *
     * @example
     * ```ts
     * const conference =
     *   await client.texml.accounts.conferences.retrieve(
     *     'conference_sid',
     *     { account_sid: 'account_sid' },
     *   );
     * ```
     */
    retrieve(conferenceSid, params, options) {
        const { account_sid } = params;
        return this._client.get((0, path_1.path) `/texml/Accounts/${account_sid}/Conferences/${conferenceSid}`, options);
    }
    /**
     * Updates a conference resource.
     *
     * @example
     * ```ts
     * const conference =
     *   await client.texml.accounts.conferences.update(
     *     'conference_sid',
     *     { account_sid: 'account_sid' },
     *   );
     * ```
     */
    update(conferenceSid, params, options) {
        const { account_sid, ...body } = params;
        return this._client.post((0, path_1.path) `/texml/Accounts/${account_sid}/Conferences/${conferenceSid}`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ 'Content-Type': 'application/x-www-form-urlencoded' }, options?.headers]),
        });
    }
    /**
     * Lists conference resources.
     *
     * @example
     * ```ts
     * const response =
     *   await client.texml.accounts.conferences.retrieveConferences(
     *     'account_sid',
     *   );
     * ```
     */
    retrieveConferences(accountSid, query = {}, options) {
        return this._client.get((0, path_1.path) `/texml/Accounts/${accountSid}/Conferences`, { query, ...options });
    }
    /**
     * Lists conference recordings
     *
     * @example
     * ```ts
     * const response =
     *   await client.texml.accounts.conferences.retrieveRecordings(
     *     'conference_sid',
     *     { account_sid: 'account_sid' },
     *   );
     * ```
     */
    retrieveRecordings(conferenceSid, params, options) {
        const { account_sid } = params;
        return this._client.get((0, path_1.path) `/texml/Accounts/${account_sid}/Conferences/${conferenceSid}/Recordings`, options);
    }
    /**
     * Returns recordings for a conference identified by conference_sid.
     *
     * @example
     * ```ts
     * const response =
     *   await client.texml.accounts.conferences.retrieveRecordingsJson(
     *     'conference_sid',
     *     { account_sid: 'account_sid' },
     *   );
     * ```
     */
    retrieveRecordingsJson(conferenceSid, params, options) {
        const { account_sid } = params;
        return this._client.get((0, path_1.path) `/texml/Accounts/${account_sid}/Conferences/${conferenceSid}/Recordings.json`, options);
    }
}
exports.Conferences = Conferences;
Conferences.Participants = participants_1.Participants;
