// telnyx/resources/texml/accounts/conferences/conferences.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../../core/resource.mjs";
import * as ParticipantsAPI from "./participants.mjs";
import { Participants, } from "./participants.mjs";
import { buildHeaders } from "../../../../internal/headers.mjs";
import { path } from "../../../../internal/utils/path.mjs";
/**
 * TeXML REST Commands
 */
export class Conferences extends APIResource {
    constructor() {
        super(...arguments);
        this.participants = new ParticipantsAPI.Participants(this._client);
    }
    /**
     * Returns a conference resource.
     *
     * @example
     * ```ts
     * const conference =
     *   await client.texml.accounts.conferences.retrieve(
     *     'conference_sid',
     *     { account_sid: 'account_sid' },
     *   );
     * ```
     */
    retrieve(conferenceSid, params, options) {
        const { account_sid } = params;
        return this._client.get(path `/texml/Accounts/${account_sid}/Conferences/${conferenceSid}`, options);
    }
    /**
     * Updates a conference resource.
     *
     * @example
     * ```ts
     * const conference =
     *   await client.texml.accounts.conferences.update(
     *     'conference_sid',
     *     { account_sid: 'account_sid' },
     *   );
     * ```
     */
    update(conferenceSid, params, options) {
        const { account_sid, ...body } = params;
        return this._client.post(path `/texml/Accounts/${account_sid}/Conferences/${conferenceSid}`, {
            body,
            ...options,
            headers: buildHeaders([{ 'Content-Type': 'application/x-www-form-urlencoded' }, options?.headers]),
        });
    }
    /**
     * Lists conference resources.
     *
     * @example
     * ```ts
     * const response =
     *   await client.texml.accounts.conferences.retrieveConferences(
     *     'account_sid',
     *   );
     * ```
     */
    retrieveConferences(accountSid, query = {}, options) {
        return this._client.get(path `/texml/Accounts/${accountSid}/Conferences`, { query, ...options });
    }
    /**
     * Lists conference recordings
     *
     * @example
     * ```ts
     * const response =
     *   await client.texml.accounts.conferences.retrieveRecordings(
     *     'conference_sid',
     *     { account_sid: 'account_sid' },
     *   );
     * ```
     */
    retrieveRecordings(conferenceSid, params, options) {
        const { account_sid } = params;
        return this._client.get(path `/texml/Accounts/${account_sid}/Conferences/${conferenceSid}/Recordings`, options);
    }
    /**
     * Returns recordings for a conference identified by conference_sid.
     *
     * @example
     * ```ts
     * const response =
     *   await client.texml.accounts.conferences.retrieveRecordingsJson(
     *     'conference_sid',
     *     { account_sid: 'account_sid' },
     *   );
     * ```
     */
    retrieveRecordingsJson(conferenceSid, params, options) {
        const { account_sid } = params;
        return this._client.get(path `/texml/Accounts/${account_sid}/Conferences/${conferenceSid}/Recordings.json`, options);
    }
}
Conferences.Participants = Participants;
