// telnyx/resources/texml/accounts/conferences/participants.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Participants = void 0;
const resource_1 = require("../../../../core/resource.js");
const headers_1 = require("../../../../internal/headers.js");
const path_1 = require("../../../../internal/utils/path.js");
/**
 * TeXML REST Commands
 */
class Participants extends resource_1.APIResource {
    /**
     * Gets conference participant resource
     *
     * @example
     * ```ts
     * const participant =
     *   await client.texml.accounts.conferences.participants.retrieve(
     *     'call_sid_or_participant_label',
     *     {
     *       account_sid: 'account_sid',
     *       conference_sid: 'conference_sid',
     *     },
     *   );
     * ```
     */
    retrieve(callSidOrParticipantLabel, params, options) {
        const { account_sid, conference_sid } = params;
        return this._client.get((0, path_1.path) `/texml/Accounts/${account_sid}/Conferences/${conference_sid}/Participants/${callSidOrParticipantLabel}`, options);
    }
    /**
     * Updates a conference participant
     *
     * @example
     * ```ts
     * const participant =
     *   await client.texml.accounts.conferences.participants.update(
     *     'call_sid_or_participant_label',
     *     {
     *       account_sid: 'account_sid',
     *       conference_sid: 'conference_sid',
     *     },
     *   );
     * ```
     */
    update(callSidOrParticipantLabel, params, options) {
        const { account_sid, conference_sid, ...body } = params;
        return this._client.post((0, path_1.path) `/texml/Accounts/${account_sid}/Conferences/${conference_sid}/Participants/${callSidOrParticipantLabel}`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ 'Content-Type': 'application/x-www-form-urlencoded' }, options?.headers]),
        });
    }
    /**
     * Deletes a conference participant
     *
     * @example
     * ```ts
     * await client.texml.accounts.conferences.participants.delete(
     *   'call_sid_or_participant_label',
     *   {
     *     account_sid: 'account_sid',
     *     conference_sid: 'conference_sid',
     *   },
     * );
     * ```
     */
    delete(callSidOrParticipantLabel, params, options) {
        const { account_sid, conference_sid } = params;
        return this._client.delete((0, path_1.path) `/texml/Accounts/${account_sid}/Conferences/${conference_sid}/Participants/${callSidOrParticipantLabel}`, { ...options, headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]) });
    }
    /**
     * Dials a new conference participant
     *
     * @example
     * ```ts
     * const response =
     *   await client.texml.accounts.conferences.participants.participants(
     *     'conference_sid',
     *     { account_sid: 'account_sid' },
     *   );
     * ```
     */
    participants(conferenceSid, params, options) {
        const { account_sid, timeout_seconds, ...body } = params;
        return this._client.post((0, path_1.path) `/texml/Accounts/${account_sid}/Conferences/${conferenceSid}/Participants`, {
            body: { Timeout: timeout_seconds, ...body },
            ...options,
            headers: (0, headers_1.buildHeaders)([{ 'Content-Type': 'application/x-www-form-urlencoded' }, options?.headers]),
        });
    }
    /**
     * Lists conference participants
     *
     * @example
     * ```ts
     * const response =
     *   await client.texml.accounts.conferences.participants.retrieveParticipants(
     *     'conference_sid',
     *     { account_sid: 'account_sid' },
     *   );
     * ```
     */
    retrieveParticipants(conferenceSid, params, options) {
        const { account_sid } = params;
        return this._client.get((0, path_1.path) `/texml/Accounts/${account_sid}/Conferences/${conferenceSid}/Participants`, options);
    }
}
exports.Participants = Participants;
