// telnyx/resources/texml/accounts/conferences/participants.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../../core/resource.mjs";
import { buildHeaders } from "../../../../internal/headers.mjs";
import { path } from "../../../../internal/utils/path.mjs";
/**
 * TeXML REST Commands
 */
export class Participants extends APIResource {
    /**
     * Gets conference participant resource
     *
     * @example
     * ```ts
     * const participant =
     *   await client.texml.accounts.conferences.participants.retrieve(
     *     'call_sid_or_participant_label',
     *     {
     *       account_sid: 'account_sid',
     *       conference_sid: 'conference_sid',
     *     },
     *   );
     * ```
     */
    retrieve(callSidOrParticipantLabel, params, options) {
        const { account_sid, conference_sid } = params;
        return this._client.get(path `/texml/Accounts/${account_sid}/Conferences/${conference_sid}/Participants/${callSidOrParticipantLabel}`, options);
    }
    /**
     * Updates a conference participant
     *
     * @example
     * ```ts
     * const participant =
     *   await client.texml.accounts.conferences.participants.update(
     *     'call_sid_or_participant_label',
     *     {
     *       account_sid: 'account_sid',
     *       conference_sid: 'conference_sid',
     *     },
     *   );
     * ```
     */
    update(callSidOrParticipantLabel, params, options) {
        const { account_sid, conference_sid, ...body } = params;
        return this._client.post(path `/texml/Accounts/${account_sid}/Conferences/${conference_sid}/Participants/${callSidOrParticipantLabel}`, {
            body,
            ...options,
            headers: buildHeaders([{ 'Content-Type': 'application/x-www-form-urlencoded' }, options?.headers]),
        });
    }
    /**
     * Deletes a conference participant
     *
     * @example
     * ```ts
     * await client.texml.accounts.conferences.participants.delete(
     *   'call_sid_or_participant_label',
     *   {
     *     account_sid: 'account_sid',
     *     conference_sid: 'conference_sid',
     *   },
     * );
     * ```
     */
    delete(callSidOrParticipantLabel, params, options) {
        const { account_sid, conference_sid } = params;
        return this._client.delete(path `/texml/Accounts/${account_sid}/Conferences/${conference_sid}/Participants/${callSidOrParticipantLabel}`, { ...options, headers: buildHeaders([{ Accept: '*/*' }, options?.headers]) });
    }
    /**
     * Dials a new conference participant
     *
     * @example
     * ```ts
     * const response =
     *   await client.texml.accounts.conferences.participants.participants(
     *     'conference_sid',
     *     { account_sid: 'account_sid' },
     *   );
     * ```
     */
    participants(conferenceSid, params, options) {
        const { account_sid, timeout_seconds, ...body } = params;
        return this._client.post(path `/texml/Accounts/${account_sid}/Conferences/${conferenceSid}/Participants`, {
            body: { Timeout: timeout_seconds, ...body },
            ...options,
            headers: buildHeaders([{ 'Content-Type': 'application/x-www-form-urlencoded' }, options?.headers]),
        });
    }
    /**
     * Lists conference participants
     *
     * @example
     * ```ts
     * const response =
     *   await client.texml.accounts.conferences.participants.retrieveParticipants(
     *     'conference_sid',
     *     { account_sid: 'account_sid' },
     *   );
     * ```
     */
    retrieveParticipants(conferenceSid, params, options) {
        const { account_sid } = params;
        return this._client.get(path `/texml/Accounts/${account_sid}/Conferences/${conferenceSid}/Participants`, options);
    }
}
