// telnyx/resources/queues/calls.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import { DefaultFlatPagination } from "../../core/pagination.mjs";
import { buildHeaders } from "../../internal/headers.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * Queue commands operations
 */
export class Calls extends APIResource {
    /**
     * Retrieve an existing call from an existing queue
     *
     * @example
     * ```ts
     * const call = await client.queues.calls.retrieve(
     *   'call_control_id',
     *   { queue_name: 'queue_name' },
     * );
     * ```
     */
    retrieve(callControlID, params, options) {
        const { queue_name } = params;
        return this._client.get(path `/queues/${queue_name}/calls/${callControlID}`, options);
    }
    /**
     * Update queued call's keep_after_hangup flag
     *
     * @example
     * ```ts
     * await client.queues.calls.update('call_control_id', {
     *   queue_name: 'queue_name',
     * });
     * ```
     */
    update(callControlID, params, options) {
        const { queue_name, ...body } = params;
        return this._client.patch(path `/queues/${queue_name}/calls/${callControlID}`, {
            body,
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Retrieve the list of calls in an existing queue
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const callListResponse of client.queues.calls.list(
     *   'queue_name',
     * )) {
     *   // ...
     * }
     * ```
     */
    list(queueName, query = {}, options) {
        return this._client.getAPIList(path `/queues/${queueName}/calls`, (DefaultFlatPagination), { query, ...options });
    }
    /**
     * Removes an inactive call from a queue. If the call is no longer active, use this
     * command to remove it from the queue.
     *
     * @example
     * ```ts
     * await client.queues.calls.remove('call_control_id', {
     *   queue_name: 'queue_name',
     * });
     * ```
     */
    remove(callControlID, params, options) {
        const { queue_name } = params;
        return this._client.delete(path `/queues/${queue_name}/calls/${callControlID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
