// telnyx/resources/queues/calls.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Calls = void 0;
const resource_1 = require("../../core/resource.js");
const pagination_1 = require("../../core/pagination.js");
const headers_1 = require("../../internal/headers.js");
const path_1 = require("../../internal/utils/path.js");
/**
 * Queue commands operations
 */
class Calls extends resource_1.APIResource {
    /**
     * Retrieve an existing call from an existing queue
     *
     * @example
     * ```ts
     * const call = await client.queues.calls.retrieve(
     *   'call_control_id',
     *   { queue_name: 'queue_name' },
     * );
     * ```
     */
    retrieve(callControlID, params, options) {
        const { queue_name } = params;
        return this._client.get((0, path_1.path) `/queues/${queue_name}/calls/${callControlID}`, options);
    }
    /**
     * Update queued call's keep_after_hangup flag
     *
     * @example
     * ```ts
     * await client.queues.calls.update('call_control_id', {
     *   queue_name: 'queue_name',
     * });
     * ```
     */
    update(callControlID, params, options) {
        const { queue_name, ...body } = params;
        return this._client.patch((0, path_1.path) `/queues/${queue_name}/calls/${callControlID}`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Retrieve the list of calls in an existing queue
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const callListResponse of client.queues.calls.list(
     *   'queue_name',
     * )) {
     *   // ...
     * }
     * ```
     */
    list(queueName, query = {}, options) {
        return this._client.getAPIList((0, path_1.path) `/queues/${queueName}/calls`, (pagination_1.DefaultFlatPagination), { query, ...options });
    }
    /**
     * Removes an inactive call from a queue. If the call is no longer active, use this
     * command to remove it from the queue.
     *
     * @example
     * ```ts
     * await client.queues.calls.remove('call_control_id', {
     *   queue_name: 'queue_name',
     * });
     * ```
     */
    remove(callControlID, params, options) {
        const { queue_name } = params;
        return this._client.delete((0, path_1.path) `/queues/${queue_name}/calls/${callControlID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.Calls = Calls;
