// telnyx/resources/queues/queues.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Queues = void 0;
const tslib_1 = require("../../internal/tslib.js");
const resource_1 = require("../../core/resource.js");
const CallsAPI = tslib_1.__importStar(require("./calls.js"));
const calls_1 = require("./calls.js");
const pagination_1 = require("../../core/pagination.js");
const headers_1 = require("../../internal/headers.js");
const path_1 = require("../../internal/utils/path.js");
/**
 * Queue commands operations
 */
class Queues extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.calls = new CallsAPI.Calls(this._client);
    }
    /**
     * Create a new call queue.
     *
     * @example
     * ```ts
     * const queue = await client.queues.create({
     *   queue_name: 'tier_1_support',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/queues', { body, ...options });
    }
    /**
     * Retrieve an existing call queue
     *
     * @example
     * ```ts
     * const queue = await client.queues.retrieve('queue_name');
     * ```
     */
    retrieve(queueName, options) {
        return this._client.get((0, path_1.path) `/queues/${queueName}`, options);
    }
    /**
     * Update properties of an existing call queue.
     *
     * @example
     * ```ts
     * const queue = await client.queues.update('queue_name', {
     *   max_size: 200,
     * });
     * ```
     */
    update(queueName, body, options) {
        return this._client.post((0, path_1.path) `/queues/${queueName}`, { body, ...options });
    }
    /**
     * List all queues for the authenticated user.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const queue of client.queues.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/queues', (pagination_1.DefaultFlatPagination), { query, ...options });
    }
    /**
     * Delete an existing call queue.
     *
     * @example
     * ```ts
     * await client.queues.delete('queue_name');
     * ```
     */
    delete(queueName, options) {
        return this._client.delete((0, path_1.path) `/queues/${queueName}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.Queues = Queues;
Queues.Calls = calls_1.Calls;
