// telnyx/resources/queues/queues.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as CallsAPI from "./calls.mjs";
import { Calls, } from "./calls.mjs";
import { DefaultFlatPagination } from "../../core/pagination.mjs";
import { buildHeaders } from "../../internal/headers.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * Queue commands operations
 */
export class Queues extends APIResource {
    constructor() {
        super(...arguments);
        this.calls = new CallsAPI.Calls(this._client);
    }
    /**
     * Create a new call queue.
     *
     * @example
     * ```ts
     * const queue = await client.queues.create({
     *   queue_name: 'tier_1_support',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/queues', { body, ...options });
    }
    /**
     * Retrieve an existing call queue
     *
     * @example
     * ```ts
     * const queue = await client.queues.retrieve('queue_name');
     * ```
     */
    retrieve(queueName, options) {
        return this._client.get(path `/queues/${queueName}`, options);
    }
    /**
     * Update properties of an existing call queue.
     *
     * @example
     * ```ts
     * const queue = await client.queues.update('queue_name', {
     *   max_size: 200,
     * });
     * ```
     */
    update(queueName, body, options) {
        return this._client.post(path `/queues/${queueName}`, { body, ...options });
    }
    /**
     * List all queues for the authenticated user.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const queue of client.queues.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/queues', (DefaultFlatPagination), { query, ...options });
    }
    /**
     * Delete an existing call queue.
     *
     * @example
     * ```ts
     * await client.queues.delete('queue_name');
     * ```
     */
    delete(queueName, options) {
        return this._client.delete(path `/queues/${queueName}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
Queues.Calls = Calls;
