// telnyx/resources/porting-orders/porting-orders.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as ActionRequirementsAPI from "./action-requirements.mjs";
import { ActionRequirements, } from "./action-requirements.mjs";
import * as ActionsAPI from "./actions.mjs";
import { Actions, } from "./actions.mjs";
import * as ActivationJobsAPI from "./activation-jobs.mjs";
import { ActivationJobs, } from "./activation-jobs.mjs";
import * as AdditionalDocumentsAPI from "./additional-documents.mjs";
import { AdditionalDocuments, } from "./additional-documents.mjs";
import * as AssociatedPhoneNumbersAPI from "./associated-phone-numbers.mjs";
import { AssociatedPhoneNumbers, } from "./associated-phone-numbers.mjs";
import * as CommentsAPI from "./comments.mjs";
import { Comments, } from "./comments.mjs";
import * as PhoneNumberBlocksAPI from "./phone-number-blocks.mjs";
import { PhoneNumberBlocks, } from "./phone-number-blocks.mjs";
import * as PhoneNumberConfigurationsAPI from "./phone-number-configurations.mjs";
import { PhoneNumberConfigurations, } from "./phone-number-configurations.mjs";
import * as PhoneNumberExtensionsAPI from "./phone-number-extensions.mjs";
import { PhoneNumberExtensions, } from "./phone-number-extensions.mjs";
import * as VerificationCodesAPI from "./verification-codes.mjs";
import { VerificationCodes, } from "./verification-codes.mjs";
import { DefaultFlatPagination } from "../../core/pagination.mjs";
import { buildHeaders } from "../../internal/headers.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * Endpoints related to porting orders management.
 */
export class PortingOrders extends APIResource {
    constructor() {
        super(...arguments);
        this.phoneNumberConfigurations = new PhoneNumberConfigurationsAPI.PhoneNumberConfigurations(this._client);
        this.actions = new ActionsAPI.Actions(this._client);
        this.activationJobs = new ActivationJobsAPI.ActivationJobs(this._client);
        this.additionalDocuments = new AdditionalDocumentsAPI.AdditionalDocuments(this._client);
        this.comments = new CommentsAPI.Comments(this._client);
        this.verificationCodes = new VerificationCodesAPI.VerificationCodes(this._client);
        this.actionRequirements = new ActionRequirementsAPI.ActionRequirements(this._client);
        this.associatedPhoneNumbers = new AssociatedPhoneNumbersAPI.AssociatedPhoneNumbers(this._client);
        this.phoneNumberBlocks = new PhoneNumberBlocksAPI.PhoneNumberBlocks(this._client);
        this.phoneNumberExtensions = new PhoneNumberExtensionsAPI.PhoneNumberExtensions(this._client);
    }
    /**
     * Creates a new porting order object.
     *
     * @example
     * ```ts
     * const portingOrder = await client.portingOrders.create({
     *   phone_numbers: [
     *     '+13035550000',
     *     '+13035550001',
     *     '+13035550002',
     *   ],
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/porting_orders', { body, ...options });
    }
    /**
     * Retrieves the details of an existing porting order.
     *
     * @example
     * ```ts
     * const portingOrder = await client.portingOrders.retrieve(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    retrieve(id, query = {}, options) {
        return this._client.get(path `/porting_orders/${id}`, { query, ...options });
    }
    /**
     * Edits the details of an existing porting order.
     *
     * Any or all of a porting orders attributes may be included in the resource object
     * included in a PATCH request.
     *
     * If a request does not include all of the attributes for a resource, the system
     * will interpret the missing attributes as if they were included with their
     * current values. To explicitly set something to null, it must be included in the
     * request with a null value.
     *
     * @example
     * ```ts
     * const portingOrder = await client.portingOrders.update(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    update(id, body, options) {
        return this._client.patch(path `/porting_orders/${id}`, { body, ...options });
    }
    /**
     * Returns a list of your porting order.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const portingOrder of client.portingOrders.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/porting_orders', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Deletes an existing porting order. This operation is restrict to porting orders
     * in draft state.
     *
     * @example
     * ```ts
     * await client.portingOrders.delete(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/porting_orders/${id}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Returns a list of allowed FOC dates for a porting order.
     *
     * @example
     * ```ts
     * const response =
     *   await client.portingOrders.retrieveAllowedFocWindows(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    retrieveAllowedFocWindows(id, options) {
        return this._client.get(path `/porting_orders/${id}/allowed_foc_windows`, options);
    }
    /**
     * Returns a list of all possible exception types for a porting order.
     *
     * @example
     * ```ts
     * const response =
     *   await client.portingOrders.retrieveExceptionTypes();
     * ```
     */
    retrieveExceptionTypes(options) {
        return this._client.get('/porting_orders/exception_types', options);
    }
    /**
     * Download a porting order loa template
     *
     * @example
     * ```ts
     * const response =
     *   await client.portingOrders.retrieveLoaTemplate(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     *
     * const content = await response.blob();
     * console.log(content);
     * ```
     */
    retrieveLoaTemplate(id, query = {}, options) {
        return this._client.get(path `/porting_orders/${id}/loa_template`, {
            query,
            ...options,
            headers: buildHeaders([{ Accept: 'application/pdf' }, options?.headers]),
            __binaryResponse: true,
        });
    }
    /**
     * Returns a list of all requirements based on country/number type for this porting
     * order.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const portingOrderRetrieveRequirementsResponse of client.portingOrders.retrieveRequirements(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * )) {
     *   // ...
     * }
     * ```
     */
    retrieveRequirements(id, query = {}, options) {
        return this._client.getAPIList(path `/porting_orders/${id}/requirements`, (DefaultFlatPagination), { query, ...options });
    }
    /**
     * Retrieve the associated V1 sub_request_id and port_request_id
     *
     * @example
     * ```ts
     * const response =
     *   await client.portingOrders.retrieveSubRequest(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    retrieveSubRequest(id, options) {
        return this._client.get(path `/porting_orders/${id}/sub_request`, options);
    }
}
PortingOrders.PhoneNumberConfigurations = PhoneNumberConfigurations;
PortingOrders.Actions = Actions;
PortingOrders.ActivationJobs = ActivationJobs;
PortingOrders.AdditionalDocuments = AdditionalDocuments;
PortingOrders.Comments = Comments;
PortingOrders.VerificationCodes = VerificationCodes;
PortingOrders.ActionRequirements = ActionRequirements;
PortingOrders.AssociatedPhoneNumbers = AssociatedPhoneNumbers;
PortingOrders.PhoneNumberBlocks = PhoneNumberBlocks;
PortingOrders.PhoneNumberExtensions = PhoneNumberExtensions;
