// telnyx/resources/porting-orders/porting-orders.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.PortingOrders = void 0;
const tslib_1 = require("../../internal/tslib.js");
const resource_1 = require("../../core/resource.js");
const ActionRequirementsAPI = tslib_1.__importStar(require("./action-requirements.js"));
const action_requirements_1 = require("./action-requirements.js");
const ActionsAPI = tslib_1.__importStar(require("./actions.js"));
const actions_1 = require("./actions.js");
const ActivationJobsAPI = tslib_1.__importStar(require("./activation-jobs.js"));
const activation_jobs_1 = require("./activation-jobs.js");
const AdditionalDocumentsAPI = tslib_1.__importStar(require("./additional-documents.js"));
const additional_documents_1 = require("./additional-documents.js");
const AssociatedPhoneNumbersAPI = tslib_1.__importStar(require("./associated-phone-numbers.js"));
const associated_phone_numbers_1 = require("./associated-phone-numbers.js");
const CommentsAPI = tslib_1.__importStar(require("./comments.js"));
const comments_1 = require("./comments.js");
const PhoneNumberBlocksAPI = tslib_1.__importStar(require("./phone-number-blocks.js"));
const phone_number_blocks_1 = require("./phone-number-blocks.js");
const PhoneNumberConfigurationsAPI = tslib_1.__importStar(require("./phone-number-configurations.js"));
const phone_number_configurations_1 = require("./phone-number-configurations.js");
const PhoneNumberExtensionsAPI = tslib_1.__importStar(require("./phone-number-extensions.js"));
const phone_number_extensions_1 = require("./phone-number-extensions.js");
const VerificationCodesAPI = tslib_1.__importStar(require("./verification-codes.js"));
const verification_codes_1 = require("./verification-codes.js");
const pagination_1 = require("../../core/pagination.js");
const headers_1 = require("../../internal/headers.js");
const path_1 = require("../../internal/utils/path.js");
/**
 * Endpoints related to porting orders management.
 */
class PortingOrders extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.phoneNumberConfigurations = new PhoneNumberConfigurationsAPI.PhoneNumberConfigurations(this._client);
        this.actions = new ActionsAPI.Actions(this._client);
        this.activationJobs = new ActivationJobsAPI.ActivationJobs(this._client);
        this.additionalDocuments = new AdditionalDocumentsAPI.AdditionalDocuments(this._client);
        this.comments = new CommentsAPI.Comments(this._client);
        this.verificationCodes = new VerificationCodesAPI.VerificationCodes(this._client);
        this.actionRequirements = new ActionRequirementsAPI.ActionRequirements(this._client);
        this.associatedPhoneNumbers = new AssociatedPhoneNumbersAPI.AssociatedPhoneNumbers(this._client);
        this.phoneNumberBlocks = new PhoneNumberBlocksAPI.PhoneNumberBlocks(this._client);
        this.phoneNumberExtensions = new PhoneNumberExtensionsAPI.PhoneNumberExtensions(this._client);
    }
    /**
     * Creates a new porting order object.
     *
     * @example
     * ```ts
     * const portingOrder = await client.portingOrders.create({
     *   phone_numbers: [
     *     '+13035550000',
     *     '+13035550001',
     *     '+13035550002',
     *   ],
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/porting_orders', { body, ...options });
    }
    /**
     * Retrieves the details of an existing porting order.
     *
     * @example
     * ```ts
     * const portingOrder = await client.portingOrders.retrieve(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    retrieve(id, query = {}, options) {
        return this._client.get((0, path_1.path) `/porting_orders/${id}`, { query, ...options });
    }
    /**
     * Edits the details of an existing porting order.
     *
     * Any or all of a porting orders attributes may be included in the resource object
     * included in a PATCH request.
     *
     * If a request does not include all of the attributes for a resource, the system
     * will interpret the missing attributes as if they were included with their
     * current values. To explicitly set something to null, it must be included in the
     * request with a null value.
     *
     * @example
     * ```ts
     * const portingOrder = await client.portingOrders.update(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    update(id, body, options) {
        return this._client.patch((0, path_1.path) `/porting_orders/${id}`, { body, ...options });
    }
    /**
     * Returns a list of your porting order.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const portingOrder of client.portingOrders.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/porting_orders', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Deletes an existing porting order. This operation is restrict to porting orders
     * in draft state.
     *
     * @example
     * ```ts
     * await client.portingOrders.delete(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/porting_orders/${id}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Returns a list of allowed FOC dates for a porting order.
     *
     * @example
     * ```ts
     * const response =
     *   await client.portingOrders.retrieveAllowedFocWindows(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    retrieveAllowedFocWindows(id, options) {
        return this._client.get((0, path_1.path) `/porting_orders/${id}/allowed_foc_windows`, options);
    }
    /**
     * Returns a list of all possible exception types for a porting order.
     *
     * @example
     * ```ts
     * const response =
     *   await client.portingOrders.retrieveExceptionTypes();
     * ```
     */
    retrieveExceptionTypes(options) {
        return this._client.get('/porting_orders/exception_types', options);
    }
    /**
     * Download a porting order loa template
     *
     * @example
     * ```ts
     * const response =
     *   await client.portingOrders.retrieveLoaTemplate(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     *
     * const content = await response.blob();
     * console.log(content);
     * ```
     */
    retrieveLoaTemplate(id, query = {}, options) {
        return this._client.get((0, path_1.path) `/porting_orders/${id}/loa_template`, {
            query,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: 'application/pdf' }, options?.headers]),
            __binaryResponse: true,
        });
    }
    /**
     * Returns a list of all requirements based on country/number type for this porting
     * order.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const portingOrderRetrieveRequirementsResponse of client.portingOrders.retrieveRequirements(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * )) {
     *   // ...
     * }
     * ```
     */
    retrieveRequirements(id, query = {}, options) {
        return this._client.getAPIList((0, path_1.path) `/porting_orders/${id}/requirements`, (pagination_1.DefaultFlatPagination), { query, ...options });
    }
    /**
     * Retrieve the associated V1 sub_request_id and port_request_id
     *
     * @example
     * ```ts
     * const response =
     *   await client.portingOrders.retrieveSubRequest(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    retrieveSubRequest(id, options) {
        return this._client.get((0, path_1.path) `/porting_orders/${id}/sub_request`, options);
    }
}
exports.PortingOrders = PortingOrders;
PortingOrders.PhoneNumberConfigurations = phone_number_configurations_1.PhoneNumberConfigurations;
PortingOrders.Actions = actions_1.Actions;
PortingOrders.ActivationJobs = activation_jobs_1.ActivationJobs;
PortingOrders.AdditionalDocuments = additional_documents_1.AdditionalDocuments;
PortingOrders.Comments = comments_1.Comments;
PortingOrders.VerificationCodes = verification_codes_1.VerificationCodes;
PortingOrders.ActionRequirements = action_requirements_1.ActionRequirements;
PortingOrders.AssociatedPhoneNumbers = associated_phone_numbers_1.AssociatedPhoneNumbers;
PortingOrders.PhoneNumberBlocks = phone_number_blocks_1.PhoneNumberBlocks;
PortingOrders.PhoneNumberExtensions = phone_number_extensions_1.PhoneNumberExtensions;
