// telnyx/resources/enterprises/enterprises.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as ReputationAPI from "./reputation/reputation.mjs";
import { Reputation, } from "./reputation/reputation.mjs";
import { DefaultFlatPagination } from "../../core/pagination.mjs";
import { buildHeaders } from "../../internal/headers.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * Enterprise management for Branded Calling and Number Reputation services
 */
export class Enterprises extends APIResource {
    constructor() {
        super(...arguments);
        this.reputation = new ReputationAPI.Reputation(this._client);
    }
    /**
     * Create a new enterprise for Branded Calling / Number Reputation services.
     *
     * Registers the enterprise in the Branded Calling / Number Reputation services,
     * enabling it to create Display Identity Records (DIRs) or enroll in Number
     * Reputation monitoring.
     *
     * **Required Fields:** `legal_name`, `doing_business_as`, `organization_type`,
     * `country_code`, `website`, `fein`, `industry`, `number_of_employees`,
     * `organization_legal_type`, `organization_contact`, `billing_contact`,
     * `organization_physical_address`, `billing_address`
     *
     * @example
     * ```ts
     * const enterprise = await client.enterprises.create({
     *   billing_address: {
     *     administrative_area: 'Illinois',
     *     city: 'Chicago',
     *     country: 'United States',
     *     postal_code: '60601',
     *     street_address: '123 Main St',
     *   },
     *   billing_contact: {
     *     email: 'billing@acme.com',
     *     first_name: 'John',
     *     last_name: 'Doe',
     *     phone_number: '15551234568',
     *   },
     *   country_code: 'US',
     *   doing_business_as: 'Acme',
     *   fein: '12-3456789',
     *   industry: 'technology',
     *   legal_name: 'Acme Corp Inc.',
     *   number_of_employees: '51-200',
     *   organization_contact: {
     *     email: 'jane.smith@acme.com',
     *     first_name: 'Jane',
     *     job_title: 'VP of Engineering',
     *     last_name: 'Smith',
     *     phone: '+16035551234',
     *   },
     *   organization_legal_type: 'corporation',
     *   organization_physical_address: {
     *     administrative_area: 'Illinois',
     *     city: 'Chicago',
     *     country: 'United States',
     *     postal_code: '60601',
     *     street_address: '123 Main St',
     *   },
     *   organization_type: 'commercial',
     *   website: 'https://acme.com',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/enterprises', { body, ...options });
    }
    /**
     * Retrieve details of a specific enterprise by ID.
     *
     * @example
     * ```ts
     * const enterprise = await client.enterprises.retrieve(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    retrieve(enterpriseID, options) {
        return this._client.get(path `/enterprises/${enterpriseID}`, options);
    }
    /**
     * Update enterprise information. All fields are optional — only the provided
     * fields will be updated.
     *
     * @example
     * ```ts
     * const enterprise = await client.enterprises.update(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    update(enterpriseID, body, options) {
        return this._client.put(path `/enterprises/${enterpriseID}`, { body, ...options });
    }
    /**
     * Retrieve a paginated list of enterprises associated with your account.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const enterprisePublic of client.enterprises.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/enterprises', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete an enterprise and all associated resources. This action is irreversible.
     *
     * @example
     * ```ts
     * await client.enterprises.delete(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    delete(enterpriseID, options) {
        return this._client.delete(path `/enterprises/${enterpriseID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
Enterprises.Reputation = Reputation;
