// telnyx/resources/conferences/conferences.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as ActionsAPI from "./actions.mjs";
import { Actions, } from "./actions.mjs";
import { DefaultFlatPagination } from "../../core/pagination.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * Conference command operations
 */
export class Conferences extends APIResource {
    constructor() {
        super(...arguments);
        this.actions = new ActionsAPI.Actions(this._client);
    }
    /**
     * Create a conference from an existing call leg using a `call_control_id` and a
     * conference name. Upon creating the conference, the call will be automatically
     * bridged to the conference. Conferences will expire after all participants have
     * left the conference or after 4 hours regardless of the number of active
     * participants.
     *
     * **Expected Webhooks:**
     *
     * - `conference.created`
     * - `conference.participant.joined`
     * - `conference.participant.left`
     * - `conference.ended`
     * - `conference.recording.saved`
     * - `conference.floor.changed`
     *
     * @example
     * ```ts
     * const conference = await client.conferences.create({
     *   call_control_id:
     *     'v3:MdI91X4lWFEs7IgbBEOT9M4AigoY08M0WWZFISt1Yw2axZ_IiE4pqg',
     *   name: 'Business',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/conferences', { body, ...options });
    }
    /**
     * Retrieve an existing conference
     *
     * @example
     * ```ts
     * const conference = await client.conferences.retrieve('id');
     * ```
     */
    retrieve(id, query = {}, options) {
        return this._client.get(path `/conferences/${id}`, { query, ...options });
    }
    /**
     * Lists conferences. Conferences are created on demand, and will expire after all
     * participants have left the conference or after 4 hours regardless of the number
     * of active participants. Conferences are listed in descending order by
     * `expires_at`.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const conference of client.conferences.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/conferences', (DefaultFlatPagination), { query, ...options });
    }
    /**
     * Lists conference participants
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const conferenceListParticipantsResponse of client.conferences.listParticipants(
     *   'conference_id',
     * )) {
     *   // ...
     * }
     * ```
     */
    listParticipants(conferenceID, query = {}, options) {
        return this._client.getAPIList(path `/conferences/${conferenceID}/participants`, (DefaultFlatPagination), { query, ...options });
    }
    /**
     * Retrieve details of a specific conference participant by their ID or label.
     *
     * @example
     * ```ts
     * const response =
     *   await client.conferences.retrieveParticipant(
     *     'participant_id',
     *     { id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e' },
     *   );
     * ```
     */
    retrieveParticipant(participantID, params, options) {
        const { id } = params;
        return this._client.get(path `/conferences/${id}/participants/${participantID}`, options);
    }
    /**
     * Update properties of a conference participant.
     *
     * @example
     * ```ts
     * const response = await client.conferences.updateParticipant(
     *   'participant_id',
     *   { id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e' },
     * );
     * ```
     */
    updateParticipant(participantID, params, options) {
        const { id, ...body } = params;
        return this._client.patch(path `/conferences/${id}/participants/${participantID}`, { body, ...options });
    }
}
Conferences.Actions = Actions;
