// telnyx/resources/ai/missions/knowledge-bases.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../core/resource.mjs";
import { buildHeaders } from "../../../internal/headers.mjs";
import { path } from "../../../internal/utils/path.mjs";
export class KnowledgeBases extends APIResource {
    /**
     * Create a new knowledge base for a mission
     *
     * @example
     * ```ts
     * const response =
     *   await client.ai.missions.knowledgeBases.createKnowledgeBase(
     *     'mission_id',
     *   );
     * ```
     */
    createKnowledgeBase(missionID, options) {
        return this._client.post(path `/ai/missions/${missionID}/knowledge-bases`, options);
    }
    /**
     * Delete a knowledge base from a mission
     *
     * @example
     * ```ts
     * await client.ai.missions.knowledgeBases.deleteKnowledgeBase(
     *   'knowledge_base_id',
     *   { mission_id: 'mission_id' },
     * );
     * ```
     */
    deleteKnowledgeBase(knowledgeBaseID, params, options) {
        const { mission_id } = params;
        return this._client.delete(path `/ai/missions/${mission_id}/knowledge-bases/${knowledgeBaseID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Get a specific knowledge base by ID
     *
     * @example
     * ```ts
     * const response =
     *   await client.ai.missions.knowledgeBases.getKnowledgeBase(
     *     'knowledge_base_id',
     *     { mission_id: 'mission_id' },
     *   );
     * ```
     */
    getKnowledgeBase(knowledgeBaseID, params, options) {
        const { mission_id } = params;
        return this._client.get(path `/ai/missions/${mission_id}/knowledge-bases/${knowledgeBaseID}`, options);
    }
    /**
     * List all knowledge bases for a mission
     *
     * @example
     * ```ts
     * const response =
     *   await client.ai.missions.knowledgeBases.listKnowledgeBases(
     *     'mission_id',
     *   );
     * ```
     */
    listKnowledgeBases(missionID, options) {
        return this._client.get(path `/ai/missions/${missionID}/knowledge-bases`, options);
    }
    /**
     * Update a knowledge base definition
     *
     * @example
     * ```ts
     * const response =
     *   await client.ai.missions.knowledgeBases.updateKnowledgeBase(
     *     'knowledge_base_id',
     *     { mission_id: 'mission_id' },
     *   );
     * ```
     */
    updateKnowledgeBase(knowledgeBaseID, params, options) {
        const { mission_id } = params;
        return this._client.put(path `/ai/missions/${mission_id}/knowledge-bases/${knowledgeBaseID}`, options);
    }
}
