// telnyx/resources/ai/missions/knowledge-bases.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.KnowledgeBases = void 0;
const resource_1 = require("../../../core/resource.js");
const headers_1 = require("../../../internal/headers.js");
const path_1 = require("../../../internal/utils/path.js");
class KnowledgeBases extends resource_1.APIResource {
    /**
     * Create a new knowledge base for a mission
     *
     * @example
     * ```ts
     * const response =
     *   await client.ai.missions.knowledgeBases.createKnowledgeBase(
     *     'mission_id',
     *   );
     * ```
     */
    createKnowledgeBase(missionID, options) {
        return this._client.post((0, path_1.path) `/ai/missions/${missionID}/knowledge-bases`, options);
    }
    /**
     * Delete a knowledge base from a mission
     *
     * @example
     * ```ts
     * await client.ai.missions.knowledgeBases.deleteKnowledgeBase(
     *   'knowledge_base_id',
     *   { mission_id: 'mission_id' },
     * );
     * ```
     */
    deleteKnowledgeBase(knowledgeBaseID, params, options) {
        const { mission_id } = params;
        return this._client.delete((0, path_1.path) `/ai/missions/${mission_id}/knowledge-bases/${knowledgeBaseID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Get a specific knowledge base by ID
     *
     * @example
     * ```ts
     * const response =
     *   await client.ai.missions.knowledgeBases.getKnowledgeBase(
     *     'knowledge_base_id',
     *     { mission_id: 'mission_id' },
     *   );
     * ```
     */
    getKnowledgeBase(knowledgeBaseID, params, options) {
        const { mission_id } = params;
        return this._client.get((0, path_1.path) `/ai/missions/${mission_id}/knowledge-bases/${knowledgeBaseID}`, options);
    }
    /**
     * List all knowledge bases for a mission
     *
     * @example
     * ```ts
     * const response =
     *   await client.ai.missions.knowledgeBases.listKnowledgeBases(
     *     'mission_id',
     *   );
     * ```
     */
    listKnowledgeBases(missionID, options) {
        return this._client.get((0, path_1.path) `/ai/missions/${missionID}/knowledge-bases`, options);
    }
    /**
     * Update a knowledge base definition
     *
     * @example
     * ```ts
     * const response =
     *   await client.ai.missions.knowledgeBases.updateKnowledgeBase(
     *     'knowledge_base_id',
     *     { mission_id: 'mission_id' },
     *   );
     * ```
     */
    updateKnowledgeBase(knowledgeBaseID, params, options) {
        const { mission_id } = params;
        return this._client.put((0, path_1.path) `/ai/missions/${mission_id}/knowledge-bases/${knowledgeBaseID}`, options);
    }
}
exports.KnowledgeBases = KnowledgeBases;
