// telnyx/resources/ai/missions/mcp-servers.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../core/resource.mjs";
import { buildHeaders } from "../../../internal/headers.mjs";
import { path } from "../../../internal/utils/path.mjs";
export class McpServers extends APIResource {
    /**
     * Create a new MCP server for a mission
     *
     * @example
     * ```ts
     * const response =
     *   await client.ai.missions.mcpServers.createMcpServer(
     *     'mission_id',
     *   );
     * ```
     */
    createMcpServer(missionID, options) {
        return this._client.post(path `/ai/missions/${missionID}/mcp-servers`, options);
    }
    /**
     * Delete an MCP server from a mission
     *
     * @example
     * ```ts
     * await client.ai.missions.mcpServers.deleteMcpServer(
     *   'mcp_server_id',
     *   { mission_id: 'mission_id' },
     * );
     * ```
     */
    deleteMcpServer(mcpServerID, params, options) {
        const { mission_id } = params;
        return this._client.delete(path `/ai/missions/${mission_id}/mcp-servers/${mcpServerID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Get a specific MCP server by ID
     *
     * @example
     * ```ts
     * const response =
     *   await client.ai.missions.mcpServers.getMcpServer(
     *     'mcp_server_id',
     *     { mission_id: 'mission_id' },
     *   );
     * ```
     */
    getMcpServer(mcpServerID, params, options) {
        const { mission_id } = params;
        return this._client.get(path `/ai/missions/${mission_id}/mcp-servers/${mcpServerID}`, options);
    }
    /**
     * List all MCP servers for a mission
     *
     * @example
     * ```ts
     * const response =
     *   await client.ai.missions.mcpServers.listMcpServers(
     *     'mission_id',
     *   );
     * ```
     */
    listMcpServers(missionID, options) {
        return this._client.get(path `/ai/missions/${missionID}/mcp-servers`, options);
    }
    /**
     * Update an MCP server definition
     *
     * @example
     * ```ts
     * const response =
     *   await client.ai.missions.mcpServers.updateMcpServer(
     *     'mcp_server_id',
     *     { mission_id: 'mission_id' },
     *   );
     * ```
     */
    updateMcpServer(mcpServerID, params, options) {
        const { mission_id } = params;
        return this._client.put(path `/ai/missions/${mission_id}/mcp-servers/${mcpServerID}`, options);
    }
}
