// telnyx/resources/ai/missions/mcp-servers.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.McpServers = void 0;
const resource_1 = require("../../../core/resource.js");
const headers_1 = require("../../../internal/headers.js");
const path_1 = require("../../../internal/utils/path.js");
class McpServers extends resource_1.APIResource {
    /**
     * Create a new MCP server for a mission
     *
     * @example
     * ```ts
     * const response =
     *   await client.ai.missions.mcpServers.createMcpServer(
     *     'mission_id',
     *   );
     * ```
     */
    createMcpServer(missionID, options) {
        return this._client.post((0, path_1.path) `/ai/missions/${missionID}/mcp-servers`, options);
    }
    /**
     * Delete an MCP server from a mission
     *
     * @example
     * ```ts
     * await client.ai.missions.mcpServers.deleteMcpServer(
     *   'mcp_server_id',
     *   { mission_id: 'mission_id' },
     * );
     * ```
     */
    deleteMcpServer(mcpServerID, params, options) {
        const { mission_id } = params;
        return this._client.delete((0, path_1.path) `/ai/missions/${mission_id}/mcp-servers/${mcpServerID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Get a specific MCP server by ID
     *
     * @example
     * ```ts
     * const response =
     *   await client.ai.missions.mcpServers.getMcpServer(
     *     'mcp_server_id',
     *     { mission_id: 'mission_id' },
     *   );
     * ```
     */
    getMcpServer(mcpServerID, params, options) {
        const { mission_id } = params;
        return this._client.get((0, path_1.path) `/ai/missions/${mission_id}/mcp-servers/${mcpServerID}`, options);
    }
    /**
     * List all MCP servers for a mission
     *
     * @example
     * ```ts
     * const response =
     *   await client.ai.missions.mcpServers.listMcpServers(
     *     'mission_id',
     *   );
     * ```
     */
    listMcpServers(missionID, options) {
        return this._client.get((0, path_1.path) `/ai/missions/${missionID}/mcp-servers`, options);
    }
    /**
     * Update an MCP server definition
     *
     * @example
     * ```ts
     * const response =
     *   await client.ai.missions.mcpServers.updateMcpServer(
     *     'mcp_server_id',
     *     { mission_id: 'mission_id' },
     *   );
     * ```
     */
    updateMcpServer(mcpServerID, params, options) {
        const { mission_id } = params;
        return this._client.put((0, path_1.path) `/ai/missions/${mission_id}/mcp-servers/${mcpServerID}`, options);
    }
}
exports.McpServers = McpServers;
