// telnyx/resources/ai/missions/missions.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Missions = void 0;
const tslib_1 = require("../../../internal/tslib.js");
const resource_1 = require("../../../core/resource.js");
const KnowledgeBasesAPI = tslib_1.__importStar(require("./knowledge-bases.js"));
const knowledge_bases_1 = require("./knowledge-bases.js");
const McpServersAPI = tslib_1.__importStar(require("./mcp-servers.js"));
const mcp_servers_1 = require("./mcp-servers.js");
const ToolsAPI = tslib_1.__importStar(require("./tools.js"));
const tools_1 = require("./tools.js");
const RunsAPI = tslib_1.__importStar(require("./runs/runs.js"));
const runs_1 = require("./runs/runs.js");
const pagination_1 = require("../../../core/pagination.js");
const headers_1 = require("../../../internal/headers.js");
const path_1 = require("../../../internal/utils/path.js");
class Missions extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.runs = new RunsAPI.Runs(this._client);
        this.knowledgeBases = new KnowledgeBasesAPI.KnowledgeBases(this._client);
        this.mcpServers = new McpServersAPI.McpServers(this._client);
        this.tools = new ToolsAPI.Tools(this._client);
    }
    /**
     * Create a new mission definition
     *
     * @example
     * ```ts
     * const mission = await client.ai.missions.create({
     *   name: 'name',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/ai/missions', { body, ...options });
    }
    /**
     * Get a mission by ID (includes tools, knowledge_bases, mcp_servers)
     *
     * @example
     * ```ts
     * const mission = await client.ai.missions.retrieve(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    retrieve(missionID, options) {
        return this._client.get((0, path_1.path) `/ai/missions/${missionID}`, options);
    }
    /**
     * List all missions for the organization
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const missionData of client.ai.missions.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/ai/missions', (pagination_1.DefaultFlatPagination), { query, ...options });
    }
    /**
     * Clone an existing mission
     *
     * @example
     * ```ts
     * const response = await client.ai.missions.cloneMission(
     *   'mission_id',
     * );
     * ```
     */
    cloneMission(missionID, options) {
        return this._client.post((0, path_1.path) `/ai/missions/${missionID}/clone`, options);
    }
    /**
     * Delete a mission
     *
     * @example
     * ```ts
     * await client.ai.missions.deleteMission(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    deleteMission(missionID, options) {
        return this._client.delete((0, path_1.path) `/ai/missions/${missionID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * List recent events across all missions
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const eventData of client.ai.missions.listEvents()) {
     *   // ...
     * }
     * ```
     */
    listEvents(query = {}, options) {
        return this._client.getAPIList('/ai/missions/events', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Update a mission definition
     *
     * @example
     * ```ts
     * const response = await client.ai.missions.updateMission(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    updateMission(missionID, body, options) {
        return this._client.put((0, path_1.path) `/ai/missions/${missionID}`, { body, ...options });
    }
}
exports.Missions = Missions;
Missions.Runs = runs_1.Runs;
Missions.KnowledgeBases = knowledge_bases_1.KnowledgeBases;
Missions.McpServers = mcp_servers_1.McpServers;
Missions.Tools = tools_1.Tools;
