// telnyx/resources/ai/missions/missions.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../core/resource.mjs";
import * as KnowledgeBasesAPI from "./knowledge-bases.mjs";
import { KnowledgeBases, } from "./knowledge-bases.mjs";
import * as McpServersAPI from "./mcp-servers.mjs";
import { McpServers, } from "./mcp-servers.mjs";
import * as ToolsAPI from "./tools.mjs";
import { Tools, } from "./tools.mjs";
import * as RunsAPI from "./runs/runs.mjs";
import { Runs, } from "./runs/runs.mjs";
import { DefaultFlatPagination, } from "../../../core/pagination.mjs";
import { buildHeaders } from "../../../internal/headers.mjs";
import { path } from "../../../internal/utils/path.mjs";
export class Missions extends APIResource {
    constructor() {
        super(...arguments);
        this.runs = new RunsAPI.Runs(this._client);
        this.knowledgeBases = new KnowledgeBasesAPI.KnowledgeBases(this._client);
        this.mcpServers = new McpServersAPI.McpServers(this._client);
        this.tools = new ToolsAPI.Tools(this._client);
    }
    /**
     * Create a new mission definition
     *
     * @example
     * ```ts
     * const mission = await client.ai.missions.create({
     *   name: 'name',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/ai/missions', { body, ...options });
    }
    /**
     * Get a mission by ID (includes tools, knowledge_bases, mcp_servers)
     *
     * @example
     * ```ts
     * const mission = await client.ai.missions.retrieve(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    retrieve(missionID, options) {
        return this._client.get(path `/ai/missions/${missionID}`, options);
    }
    /**
     * List all missions for the organization
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const missionData of client.ai.missions.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/ai/missions', (DefaultFlatPagination), { query, ...options });
    }
    /**
     * Clone an existing mission
     *
     * @example
     * ```ts
     * const response = await client.ai.missions.cloneMission(
     *   'mission_id',
     * );
     * ```
     */
    cloneMission(missionID, options) {
        return this._client.post(path `/ai/missions/${missionID}/clone`, options);
    }
    /**
     * Delete a mission
     *
     * @example
     * ```ts
     * await client.ai.missions.deleteMission(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    deleteMission(missionID, options) {
        return this._client.delete(path `/ai/missions/${missionID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * List recent events across all missions
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const eventData of client.ai.missions.listEvents()) {
     *   // ...
     * }
     * ```
     */
    listEvents(query = {}, options) {
        return this._client.getAPIList('/ai/missions/events', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Update a mission definition
     *
     * @example
     * ```ts
     * const response = await client.ai.missions.updateMission(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    updateMission(missionID, body, options) {
        return this._client.put(path `/ai/missions/${missionID}`, { body, ...options });
    }
}
Missions.Runs = Runs;
Missions.KnowledgeBases = KnowledgeBases;
Missions.McpServers = McpServers;
Missions.Tools = Tools;
