// telnyx/resources/ai/missions/tools.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../core/resource.mjs";
import { buildHeaders } from "../../../internal/headers.mjs";
import { path } from "../../../internal/utils/path.mjs";
export class Tools extends APIResource {
    /**
     * Create a new tool for a mission
     *
     * @example
     * ```ts
     * const response = await client.ai.missions.tools.createTool(
     *   'mission_id',
     * );
     * ```
     */
    createTool(missionID, options) {
        return this._client.post(path `/ai/missions/${missionID}/tools`, options);
    }
    /**
     * Delete a tool from a mission
     *
     * @example
     * ```ts
     * await client.ai.missions.tools.deleteTool('tool_id', {
     *   mission_id: 'mission_id',
     * });
     * ```
     */
    deleteTool(toolID, params, options) {
        const { mission_id } = params;
        return this._client.delete(path `/ai/missions/${mission_id}/tools/${toolID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Get a specific tool by ID
     *
     * @example
     * ```ts
     * const response = await client.ai.missions.tools.getTool(
     *   'tool_id',
     *   { mission_id: 'mission_id' },
     * );
     * ```
     */
    getTool(toolID, params, options) {
        const { mission_id } = params;
        return this._client.get(path `/ai/missions/${mission_id}/tools/${toolID}`, options);
    }
    /**
     * List all tools for a mission
     *
     * @example
     * ```ts
     * const response = await client.ai.missions.tools.listTools(
     *   'mission_id',
     * );
     * ```
     */
    listTools(missionID, options) {
        return this._client.get(path `/ai/missions/${missionID}/tools`, options);
    }
    /**
     * Update a tool definition
     *
     * @example
     * ```ts
     * const response = await client.ai.missions.tools.updateTool(
     *   'tool_id',
     *   { mission_id: 'mission_id' },
     * );
     * ```
     */
    updateTool(toolID, params, options) {
        const { mission_id } = params;
        return this._client.put(path `/ai/missions/${mission_id}/tools/${toolID}`, options);
    }
}
