// telnyx/resources/ai/missions/tools.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tools = void 0;
const resource_1 = require("../../../core/resource.js");
const headers_1 = require("../../../internal/headers.js");
const path_1 = require("../../../internal/utils/path.js");
class Tools extends resource_1.APIResource {
    /**
     * Create a new tool for a mission
     *
     * @example
     * ```ts
     * const response = await client.ai.missions.tools.createTool(
     *   'mission_id',
     * );
     * ```
     */
    createTool(missionID, options) {
        return this._client.post((0, path_1.path) `/ai/missions/${missionID}/tools`, options);
    }
    /**
     * Delete a tool from a mission
     *
     * @example
     * ```ts
     * await client.ai.missions.tools.deleteTool('tool_id', {
     *   mission_id: 'mission_id',
     * });
     * ```
     */
    deleteTool(toolID, params, options) {
        const { mission_id } = params;
        return this._client.delete((0, path_1.path) `/ai/missions/${mission_id}/tools/${toolID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Get a specific tool by ID
     *
     * @example
     * ```ts
     * const response = await client.ai.missions.tools.getTool(
     *   'tool_id',
     *   { mission_id: 'mission_id' },
     * );
     * ```
     */
    getTool(toolID, params, options) {
        const { mission_id } = params;
        return this._client.get((0, path_1.path) `/ai/missions/${mission_id}/tools/${toolID}`, options);
    }
    /**
     * List all tools for a mission
     *
     * @example
     * ```ts
     * const response = await client.ai.missions.tools.listTools(
     *   'mission_id',
     * );
     * ```
     */
    listTools(missionID, options) {
        return this._client.get((0, path_1.path) `/ai/missions/${missionID}/tools`, options);
    }
    /**
     * Update a tool definition
     *
     * @example
     * ```ts
     * const response = await client.ai.missions.tools.updateTool(
     *   'tool_id',
     *   { mission_id: 'mission_id' },
     * );
     * ```
     */
    updateTool(toolID, params, options) {
        const { mission_id } = params;
        return this._client.put((0, path_1.path) `/ai/missions/${mission_id}/tools/${toolID}`, options);
    }
}
exports.Tools = Tools;
