// telnyx/resources/sim-cards/actions.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import { DefaultFlatPagination } from "../../core/pagination.mjs";
import { path } from "../../internal/utils/path.mjs";
export class Actions extends APIResource {
    /**
     * This API fetches detailed information about a SIM card action to follow-up on an
     * existing asynchronous operation.
     *
     * @example
     * ```ts
     * const action = await client.simCards.actions.retrieve(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/sim_card_actions/${id}`, options);
    }
    /**
     * This API lists a paginated collection of SIM card actions. It enables exploring
     * a collection of existing asynchronous operations using specific filters.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const simCardAction of client.simCards.actions.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/sim_card_actions', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * This API triggers an asynchronous operation to disable voice on SIM cards
     * belonging to a specified SIM Card Group.<br/> For each SIM Card a SIM Card
     * Action will be generated. The status of the SIM Card Actions can be followed
     * through the
     * [List SIM Card Action](https://developers.telnyx.com/api-reference/sim-card-actions/list-sim-card-actions)
     * API.
     *
     * The overall status of the Bulk SIM Card Action can be followed through the
     * [List Bulk SIM Card Action](https://developers.telnyx.com/api-reference/sim-card-actions/list-bulk-sim-card-actions)
     * API.
     *
     * @example
     * ```ts
     * const response =
     *   await client.simCards.actions.bulkDisableVoice({
     *     sim_card_group_id:
     *       '6b14e151-8493-4fa1-8664-1cc4e6d14158',
     *   });
     * ```
     */
    bulkDisableVoice(body, options) {
        return this._client.post('/sim_cards/actions/bulk_disable_voice', { body, ...options });
    }
    /**
     * This API triggers an asynchronous operation to enable voice on SIM cards
     * belonging to a specified SIM Card Group.<br/> For each SIM Card a SIM Card
     * Action will be generated. The status of the SIM Card Actions can be followed
     * through the
     * [List SIM Card Action](https://developers.telnyx.com/api-reference/sim-card-actions/list-sim-card-actions)
     * API.
     *
     * The overall status of the Bulk SIM Card Action can be followed through the
     * [List Bulk SIM Card Action](https://developers.telnyx.com/api-reference/sim-card-actions/list-bulk-sim-card-actions)
     * API.
     *
     * @example
     * ```ts
     * const response =
     *   await client.simCards.actions.bulkEnableVoice({
     *     sim_card_group_id:
     *       '6b14e151-8493-4fa1-8664-1cc4e6d14158',
     *   });
     * ```
     */
    bulkEnableVoice(body, options) {
        return this._client.post('/sim_cards/actions/bulk_enable_voice', { body, ...options });
    }
    /**
     * This API triggers an asynchronous operation to set a public IP for each of the
     * specified SIM cards.<br/> For each SIM Card a SIM Card Action will be generated.
     * The status of the SIM Card Action can be followed through the
     * [List SIM Card Action](https://developers.telnyx.com/api-reference/sim-card-actions/list-sim-card-actions)
     * API.
     *
     * @example
     * ```ts
     * const response =
     *   await client.simCards.actions.bulkSetPublicIPs({
     *     sim_card_ids: ['6b14e151-8493-4fa1-8664-1cc4e6d14158'],
     *   });
     * ```
     */
    bulkSetPublicIPs(body, options) {
        return this._client.post('/sim_cards/actions/bulk_set_public_ips', { body, ...options });
    }
    /**
     * This API disables a SIM card, disconnecting it from the network and making it
     * impossible to consume data.<br/> The API will trigger an asynchronous operation
     * called a SIM Card Action. Transitioning to the disabled state may take a period
     * of time. The status of the SIM Card Action can be followed through the
     * [List SIM Card Action](https://developers.telnyx.com/api-reference/sim-card-actions/list-sim-card-actions)
     * API.
     *
     * @example
     * ```ts
     * const response = await client.simCards.actions.disable(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    disable(id, options) {
        return this._client.post(path `/sim_cards/${id}/actions/disable`, options);
    }
    /**
     * This API enables a SIM card, connecting it to the network and making it possible
     * to consume data.<br/> To enable a SIM card, it must be associated with a SIM
     * card group.<br/> The API will trigger an asynchronous operation called a SIM
     * Card Action. Transitioning to the enabled state may take a period of time. The
     * status of the SIM Card Action can be followed through the
     * [List SIM Card Action](https://developers.telnyx.com/api-reference/sim-card-actions/list-sim-card-actions)
     * API.
     *
     * @example
     * ```ts
     * const response = await client.simCards.actions.enable(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    enable(id, options) {
        return this._client.post(path `/sim_cards/${id}/actions/enable`, options);
    }
    /**
     * This API removes an existing public IP from a SIM card. <br/><br/> The API will
     * trigger an asynchronous operation called a SIM Card Action. The status of the
     * SIM Card Action can be followed through the
     * [List SIM Card Action](https://developers.telnyx.com/api-reference/sim-card-actions/list-sim-card-actions)
     * API.
     *
     * @example
     * ```ts
     * const response =
     *   await client.simCards.actions.removePublicIP(
     *     '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *   );
     * ```
     */
    removePublicIP(id, options) {
        return this._client.post(path `/sim_cards/${id}/actions/remove_public_ip`, options);
    }
    /**
     * This API makes a SIM card reachable on the public internet by mapping a random
     * public IP to the SIM card. <br/><br/> The API will trigger an asynchronous
     * operation called a SIM Card Action. The status of the SIM Card Action can be
     * followed through the
     * [List SIM Card Action](https://developers.telnyx.com/api-reference/sim-card-actions/list-sim-card-actions)
     * API. <br/><br/> Setting a Public IP to a SIM Card incurs a charge and will only
     * succeed if the account has sufficient funds.
     *
     * @example
     * ```ts
     * const response = await client.simCards.actions.setPublicIP(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    setPublicIP(id, params = {}, options) {
        const { region_code } = params ?? {};
        return this._client.post(path `/sim_cards/${id}/actions/set_public_ip`, {
            query: { region_code },
            ...options,
        });
    }
    /**
     * The SIM card will be able to connect to the network once the process to set it
     * to standby has been completed, thus making it possible to consume data.<br/> To
     * set a SIM card to standby, it must be associated with SIM card group.<br/> The
     * API will trigger an asynchronous operation called a SIM Card Action.
     * Transitioning to the standby state may take a period of time. The status of the
     * SIM Card Action can be followed through the
     * [List SIM Card Action](https://developers.telnyx.com/api-reference/sim-card-actions/list-sim-card-actions)
     * API.
     *
     * @example
     * ```ts
     * const response = await client.simCards.actions.setStandby(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    setStandby(id, options) {
        return this._client.post(path `/sim_cards/${id}/actions/set_standby`, options);
    }
    /**
     * It validates whether SIM card registration codes are valid or not.
     *
     * @example
     * ```ts
     * const response =
     *   await client.simCards.actions.validateRegistrationCodes();
     * ```
     */
    validateRegistrationCodes(body, options) {
        return this._client.post('/sim_cards/actions/validate_registration_codes', { body, ...options });
    }
}
