// telnyx/resources/sim-cards/sim-cards.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.SimCards = void 0;
const tslib_1 = require("../../internal/tslib.js");
const resource_1 = require("../../core/resource.js");
const ActionsAPI = tslib_1.__importStar(require("./actions.js"));
const actions_1 = require("./actions.js");
const pagination_1 = require("../../core/pagination.js");
const path_1 = require("../../internal/utils/path.js");
/**
 * SIM Cards operations
 */
class SimCards extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.actions = new ActionsAPI.Actions(this._client);
    }
    /**
     * Returns the details regarding a specific SIM card.
     *
     * @example
     * ```ts
     * const simCard = await client.simCards.retrieve(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    retrieve(id, query = {}, options) {
        return this._client.get((0, path_1.path) `/sim_cards/${id}`, { query, ...options });
    }
    /**
     * Updates SIM card data
     *
     * @example
     * ```ts
     * const simCard = await client.simCards.update(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    update(simCardID, body, options) {
        return this._client.patch((0, path_1.path) `/sim_cards/${simCardID}`, { body, ...options });
    }
    /**
     * Get all SIM cards belonging to the user that match the given filters.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const simpleSimCard of client.simCards.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/sim_cards', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * The SIM card will be decommissioned, removed from your account and you will stop
     * being charged.<br />The SIM card won't be able to connect to the network after
     * the deletion is completed, thus making it impossible to consume data.<br/>
     * Transitioning to the disabled state may take a period of time. Until the
     * transition is completed, the SIM card status will be disabling
     * <code>disabling</code>.<br />In order to re-enable the SIM card, you will need
     * to re-register it.
     *
     * @example
     * ```ts
     * const simCard = await client.simCards.delete(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    delete(id, params = {}, options) {
        const { report_lost } = params ?? {};
        return this._client.delete((0, path_1.path) `/sim_cards/${id}`, { query: { report_lost }, ...options });
    }
    /**
     * It returns the activation code for an eSIM.<br/><br/> This API is only available
     * for eSIMs. If the given SIM is a physical SIM card, or has already been
     * installed, an error will be returned.
     *
     * @example
     * ```ts
     * const response = await client.simCards.getActivationCode(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    getActivationCode(id, options) {
        return this._client.get((0, path_1.path) `/sim_cards/${id}/activation_code`, options);
    }
    /**
     * It returns the device details where a SIM card is currently being used.
     *
     * @example
     * ```ts
     * const response = await client.simCards.getDeviceDetails(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    getDeviceDetails(id, options) {
        return this._client.get((0, path_1.path) `/sim_cards/${id}/device_details`, options);
    }
    /**
     * It returns the public IP requested for a SIM card.
     *
     * @example
     * ```ts
     * const response = await client.simCards.getPublicIP(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    getPublicIP(id, options) {
        return this._client.get((0, path_1.path) `/sim_cards/${id}/public_ip`, options);
    }
    /**
     * This API allows listing a paginated collection of Wireless Connectivity Logs
     * associated with a SIM Card, for troubleshooting purposes.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const simCardListWirelessConnectivityLogsResponse of client.simCards.listWirelessConnectivityLogs(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * )) {
     *   // ...
     * }
     * ```
     */
    listWirelessConnectivityLogs(id, query = {}, options) {
        return this._client.getAPIList((0, path_1.path) `/sim_cards/${id}/wireless_connectivity_logs`, (pagination_1.DefaultFlatPagination), { query, ...options });
    }
}
exports.SimCards = SimCards;
SimCards.Actions = actions_1.Actions;
