// telnyx/resources/messaging-profiles/autoresp-configs.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.AutorespConfigs = void 0;
const resource_1 = require("../../core/resource.js");
const path_1 = require("../../internal/utils/path.js");
/**
 * Opt-Out Management
 */
class AutorespConfigs extends resource_1.APIResource {
    /**
     * Create auto-response setting
     *
     * @example
     * ```ts
     * const autoRespConfigResponse =
     *   await client.messagingProfiles.autorespConfigs.create(
     *     'profile_id',
     *     {
     *       country_code: 'US',
     *       keywords: ['keyword1', 'keyword2'],
     *       op: 'start',
     *     },
     *   );
     * ```
     */
    create(profileID, body, options) {
        return this._client.post((0, path_1.path) `/messaging_profiles/${profileID}/autoresp_configs`, { body, ...options });
    }
    /**
     * Get Auto-Response Setting
     *
     * @example
     * ```ts
     * const autoRespConfigResponse =
     *   await client.messagingProfiles.autorespConfigs.retrieve(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { profile_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e' },
     *   );
     * ```
     */
    retrieve(autorespCfgID, params, options) {
        const { profile_id } = params;
        return this._client.get((0, path_1.path) `/messaging_profiles/${profile_id}/autoresp_configs/${autorespCfgID}`, options);
    }
    /**
     * Update Auto-Response Setting
     *
     * @example
     * ```ts
     * const autoRespConfigResponse =
     *   await client.messagingProfiles.autorespConfigs.update(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     {
     *       profile_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *       country_code: 'US',
     *       keywords: ['keyword1', 'keyword2'],
     *       op: 'start',
     *     },
     *   );
     * ```
     */
    update(autorespCfgID, params, options) {
        const { profile_id, ...body } = params;
        return this._client.put((0, path_1.path) `/messaging_profiles/${profile_id}/autoresp_configs/${autorespCfgID}`, {
            body,
            ...options,
        });
    }
    /**
     * List Auto-Response Settings
     *
     * @example
     * ```ts
     * const autorespConfigs =
     *   await client.messagingProfiles.autorespConfigs.list(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    list(profileID, query = {}, options) {
        return this._client.get((0, path_1.path) `/messaging_profiles/${profileID}/autoresp_configs`, { query, ...options });
    }
    /**
     * Delete Auto-Response Setting
     *
     * @example
     * ```ts
     * const autorespConfig =
     *   await client.messagingProfiles.autorespConfigs.delete(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { profile_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e' },
     *   );
     * ```
     */
    delete(autorespCfgID, params, options) {
        const { profile_id } = params;
        return this._client.delete((0, path_1.path) `/messaging_profiles/${profile_id}/autoresp_configs/${autorespCfgID}`, options);
    }
}
exports.AutorespConfigs = AutorespConfigs;
