// telnyx/resources/messaging-profiles/autoresp-configs.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * Opt-Out Management
 */
export class AutorespConfigs extends APIResource {
    /**
     * Create auto-response setting
     *
     * @example
     * ```ts
     * const autoRespConfigResponse =
     *   await client.messagingProfiles.autorespConfigs.create(
     *     'profile_id',
     *     {
     *       country_code: 'US',
     *       keywords: ['keyword1', 'keyword2'],
     *       op: 'start',
     *     },
     *   );
     * ```
     */
    create(profileID, body, options) {
        return this._client.post(path `/messaging_profiles/${profileID}/autoresp_configs`, { body, ...options });
    }
    /**
     * Get Auto-Response Setting
     *
     * @example
     * ```ts
     * const autoRespConfigResponse =
     *   await client.messagingProfiles.autorespConfigs.retrieve(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { profile_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e' },
     *   );
     * ```
     */
    retrieve(autorespCfgID, params, options) {
        const { profile_id } = params;
        return this._client.get(path `/messaging_profiles/${profile_id}/autoresp_configs/${autorespCfgID}`, options);
    }
    /**
     * Update Auto-Response Setting
     *
     * @example
     * ```ts
     * const autoRespConfigResponse =
     *   await client.messagingProfiles.autorespConfigs.update(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     {
     *       profile_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *       country_code: 'US',
     *       keywords: ['keyword1', 'keyword2'],
     *       op: 'start',
     *     },
     *   );
     * ```
     */
    update(autorespCfgID, params, options) {
        const { profile_id, ...body } = params;
        return this._client.put(path `/messaging_profiles/${profile_id}/autoresp_configs/${autorespCfgID}`, {
            body,
            ...options,
        });
    }
    /**
     * List Auto-Response Settings
     *
     * @example
     * ```ts
     * const autorespConfigs =
     *   await client.messagingProfiles.autorespConfigs.list(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    list(profileID, query = {}, options) {
        return this._client.get(path `/messaging_profiles/${profileID}/autoresp_configs`, { query, ...options });
    }
    /**
     * Delete Auto-Response Setting
     *
     * @example
     * ```ts
     * const autorespConfig =
     *   await client.messagingProfiles.autorespConfigs.delete(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { profile_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e' },
     *   );
     * ```
     */
    delete(autorespCfgID, params, options) {
        const { profile_id } = params;
        return this._client.delete(path `/messaging_profiles/${profile_id}/autoresp_configs/${autorespCfgID}`, options);
    }
}
