// telnyx/resources/messaging-profiles/messaging-profiles.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessagingProfiles = void 0;
const tslib_1 = require("../../internal/tslib.js");
const resource_1 = require("../../core/resource.js");
const ActionsAPI = tslib_1.__importStar(require("./actions.js"));
const actions_1 = require("./actions.js");
const AutorespConfigsAPI = tslib_1.__importStar(require("./autoresp-configs.js"));
const autoresp_configs_1 = require("./autoresp-configs.js");
const pagination_1 = require("../../core/pagination.js");
const path_1 = require("../../internal/utils/path.js");
class MessagingProfiles extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.autorespConfigs = new AutorespConfigsAPI.AutorespConfigs(this._client);
        this.actions = new ActionsAPI.Actions(this._client);
    }
    /**
     * Create a messaging profile
     *
     * @example
     * ```ts
     * const messagingProfile =
     *   await client.messagingProfiles.create({
     *     name: 'My name',
     *     whitelisted_destinations: ['US'],
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/messaging_profiles', { body, ...options });
    }
    /**
     * Retrieve a messaging profile
     *
     * @example
     * ```ts
     * const messagingProfile =
     *   await client.messagingProfiles.retrieve(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    retrieve(messagingProfileID, options) {
        return this._client.get((0, path_1.path) `/messaging_profiles/${messagingProfileID}`, options);
    }
    /**
     * Update a messaging profile
     *
     * @example
     * ```ts
     * const messagingProfile =
     *   await client.messagingProfiles.update(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    update(messagingProfileID, body, options) {
        return this._client.patch((0, path_1.path) `/messaging_profiles/${messagingProfileID}`, { body, ...options });
    }
    /**
     * List messaging profiles
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const messagingProfile of client.messagingProfiles.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/messaging_profiles', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete a messaging profile
     *
     * @example
     * ```ts
     * const messagingProfile =
     *   await client.messagingProfiles.delete(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    delete(messagingProfileID, options) {
        return this._client.delete((0, path_1.path) `/messaging_profiles/${messagingProfileID}`, options);
    }
    /**
     * List all alphanumeric sender IDs associated with a specific messaging profile.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const alphanumericSenderID of client.messagingProfiles.listAlphanumericSenderIDs(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * )) {
     *   // ...
     * }
     * ```
     */
    listAlphanumericSenderIDs(id, query = {}, options) {
        return this._client.getAPIList((0, path_1.path) `/messaging_profiles/${id}/alphanumeric_sender_ids`, (pagination_1.DefaultFlatPagination), { query, ...options });
    }
    /**
     * List phone numbers associated with a messaging profile
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const phoneNumberWithMessagingSettings of client.messagingProfiles.listPhoneNumbers(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * )) {
     *   // ...
     * }
     * ```
     */
    listPhoneNumbers(messagingProfileID, query = {}, options) {
        return this._client.getAPIList((0, path_1.path) `/messaging_profiles/${messagingProfileID}/phone_numbers`, (pagination_1.DefaultFlatPagination), { query, ...options });
    }
    /**
     * List short codes associated with a messaging profile
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const shortCode of client.messagingProfiles.listShortCodes(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * )) {
     *   // ...
     * }
     * ```
     */
    listShortCodes(messagingProfileID, query = {}, options) {
        return this._client.getAPIList((0, path_1.path) `/messaging_profiles/${messagingProfileID}/short_codes`, (pagination_1.DefaultFlatPagination), { query, ...options });
    }
    /**
     * Get detailed metrics for a specific messaging profile, broken down by time
     * interval.
     *
     * @example
     * ```ts
     * const response =
     *   await client.messagingProfiles.retrieveMetrics(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    retrieveMetrics(id, query = {}, options) {
        return this._client.get((0, path_1.path) `/messaging_profiles/${id}/metrics`, { query, ...options });
    }
}
exports.MessagingProfiles = MessagingProfiles;
MessagingProfiles.AutorespConfigs = autoresp_configs_1.AutorespConfigs;
MessagingProfiles.Actions = actions_1.Actions;
