// telnyx/resources/messaging-profiles/messaging-profiles.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as ActionsAPI from "./actions.mjs";
import { Actions } from "./actions.mjs";
import * as AutorespConfigsAPI from "./autoresp-configs.mjs";
import { AutorespConfigs, } from "./autoresp-configs.mjs";
import { DefaultFlatPagination } from "../../core/pagination.mjs";
import { path } from "../../internal/utils/path.mjs";
export class MessagingProfiles extends APIResource {
    constructor() {
        super(...arguments);
        this.autorespConfigs = new AutorespConfigsAPI.AutorespConfigs(this._client);
        this.actions = new ActionsAPI.Actions(this._client);
    }
    /**
     * Create a messaging profile
     *
     * @example
     * ```ts
     * const messagingProfile =
     *   await client.messagingProfiles.create({
     *     name: 'My name',
     *     whitelisted_destinations: ['US'],
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/messaging_profiles', { body, ...options });
    }
    /**
     * Retrieve a messaging profile
     *
     * @example
     * ```ts
     * const messagingProfile =
     *   await client.messagingProfiles.retrieve(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    retrieve(messagingProfileID, options) {
        return this._client.get(path `/messaging_profiles/${messagingProfileID}`, options);
    }
    /**
     * Update a messaging profile
     *
     * @example
     * ```ts
     * const messagingProfile =
     *   await client.messagingProfiles.update(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    update(messagingProfileID, body, options) {
        return this._client.patch(path `/messaging_profiles/${messagingProfileID}`, { body, ...options });
    }
    /**
     * List messaging profiles
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const messagingProfile of client.messagingProfiles.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/messaging_profiles', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete a messaging profile
     *
     * @example
     * ```ts
     * const messagingProfile =
     *   await client.messagingProfiles.delete(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    delete(messagingProfileID, options) {
        return this._client.delete(path `/messaging_profiles/${messagingProfileID}`, options);
    }
    /**
     * List all alphanumeric sender IDs associated with a specific messaging profile.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const alphanumericSenderID of client.messagingProfiles.listAlphanumericSenderIDs(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * )) {
     *   // ...
     * }
     * ```
     */
    listAlphanumericSenderIDs(id, query = {}, options) {
        return this._client.getAPIList(path `/messaging_profiles/${id}/alphanumeric_sender_ids`, (DefaultFlatPagination), { query, ...options });
    }
    /**
     * List phone numbers associated with a messaging profile
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const phoneNumberWithMessagingSettings of client.messagingProfiles.listPhoneNumbers(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * )) {
     *   // ...
     * }
     * ```
     */
    listPhoneNumbers(messagingProfileID, query = {}, options) {
        return this._client.getAPIList(path `/messaging_profiles/${messagingProfileID}/phone_numbers`, (DefaultFlatPagination), { query, ...options });
    }
    /**
     * List short codes associated with a messaging profile
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const shortCode of client.messagingProfiles.listShortCodes(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * )) {
     *   // ...
     * }
     * ```
     */
    listShortCodes(messagingProfileID, query = {}, options) {
        return this._client.getAPIList(path `/messaging_profiles/${messagingProfileID}/short_codes`, (DefaultFlatPagination), { query, ...options });
    }
    /**
     * Get detailed metrics for a specific messaging profile, broken down by time
     * interval.
     *
     * @example
     * ```ts
     * const response =
     *   await client.messagingProfiles.retrieveMetrics(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    retrieveMetrics(id, query = {}, options) {
        return this._client.get(path `/messaging_profiles/${id}/metrics`, { query, ...options });
    }
}
MessagingProfiles.AutorespConfigs = AutorespConfigs;
MessagingProfiles.Actions = Actions;
