// telnyx/resources/ai/embeddings/embeddings.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../core/resource.mjs";
import * as BucketsAPI from "./buckets.mjs";
import { Buckets } from "./buckets.mjs";
import { path } from "../../../internal/utils/path.mjs";
/**
 * Embed documents and perform text searches
 */
export class Embeddings extends APIResource {
    constructor() {
        super(...arguments);
        this.buckets = new BucketsAPI.Buckets(this._client);
    }
    /**
     * Perform embedding on a Telnyx Storage Bucket using the a embedding model. The
     * current supported file types are:
     *
     * - PDF
     * - HTML
     * - txt/unstructured text files
     * - json
     * - csv
     * - audio / video (mp3, mp4, mpeg, mpga, m4a, wav, or webm ) - Max of 100mb file
     *   size.
     *
     * Any files not matching the above types will be attempted to be embedded as
     * unstructured text.
     *
     * This process can be slow, so it runs in the background and the user can check
     * the status of the task using the endpoint `/ai/embeddings/{task_id}`.
     *
     * **Important Note**: When you update documents in a Telnyx Storage bucket, their
     * associated embeddings are automatically kept up to date. If you add or update a
     * file, it is automatically embedded. If you delete a file, the embeddings are
     * deleted for that particular file.
     *
     * You can also specify a custom `loader` param. Currently the only supported
     * loader value is `intercom` which loads Intercom article jsons as specified by
     * [the Intercom article API](https://developers.intercom.com/docs/references/rest-api/api.intercom.io/Articles/article/)
     * This loader will split each article into paragraphs and save additional
     * parameters relevant to Intercom docs, such as `article_url` and `heading`. These
     * values will be returned by the `/v2/ai/embeddings/similarity-search` endpoint in
     * the `loader_metadata` field.
     *
     * @example
     * ```ts
     * const embeddingResponse = await client.ai.embeddings.create(
     *   { bucket_name: 'bucket_name' },
     * );
     * ```
     */
    create(body, options) {
        return this._client.post('/ai/embeddings', { body, ...options });
    }
    /**
     * Check the status of a current embedding task. Will be one of the following:
     *
     * - `queued` - Task is waiting to be picked up by a worker
     * - `processing` - The embedding task is running
     * - `success` - Task completed successfully and the bucket is embedded
     * - `failure` - Task failed and no files were embedded successfully
     * - `partial_success` - Some files were embedded successfully, but at least one
     *   failed
     *
     * @example
     * ```ts
     * const embedding = await client.ai.embeddings.retrieve(
     *   'task_id',
     * );
     * ```
     */
    retrieve(taskID, options) {
        return this._client.get(path `/ai/embeddings/${taskID}`, options);
    }
    /**
     * Retrieve tasks for the user that are either `queued`, `processing`, `failed`,
     * `success` or `partial_success` based on the query string. Defaults to `queued`
     * and `processing`.
     *
     * @example
     * ```ts
     * const embeddings = await client.ai.embeddings.list();
     * ```
     */
    list(query = {}, options) {
        return this._client.get('/ai/embeddings', { query, ...options });
    }
    /**
     * Perform a similarity search on a Telnyx Storage Bucket, returning the most
     * similar `num_docs` document chunks to the query.
     *
     * Currently the only available distance metric is cosine similarity which will
     * return a `distance` between 0 and 1. The lower the distance, the more similar
     * the returned document chunks are to the query. A `certainty` will also be
     * returned, which is a value between 0 and 1 where the higher the certainty, the
     * more similar the document. You can read more about Weaviate distance metrics
     * here:
     * [Weaviate Docs](https://weaviate.io/developers/weaviate/config-refs/distances)
     *
     * If a bucket was embedded using a custom loader, such as `intercom`, the
     * additional metadata will be returned in the `loader_metadata` field.
     *
     * @example
     * ```ts
     * const response =
     *   await client.ai.embeddings.similaritySearch({
     *     bucket_name: 'bucket_name',
     *     query: 'query',
     *   });
     * ```
     */
    similaritySearch(body, options) {
        return this._client.post('/ai/embeddings/similarity-search', { body, ...options });
    }
    /**
     * Embed website content from a specified URL, including child pages up to 5 levels
     * deep within the same domain. The process crawls and loads content from the main
     * URL and its linked pages into a Telnyx Cloud Storage bucket. As soon as each
     * webpage is added to the bucket, its content is immediately processed for
     * embeddings, that can be used for
     * [similarity search](https://developers.telnyx.com/api-reference/embeddings/search-for-documents)
     * and [clustering](https://developers.telnyx.com/docs/inference/clusters).
     *
     * @example
     * ```ts
     * const embeddingResponse = await client.ai.embeddings.url({
     *   bucket_name: 'bucket_name',
     *   url: 'url',
     * });
     * ```
     */
    url(body, options) {
        return this._client.post('/ai/embeddings/url', { body, ...options });
    }
}
Embeddings.Buckets = Buckets;
