// telnyx/resources/messaging-10dlc/index.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.PhoneNumberCampaigns = exports.PhoneNumberAssignmentByProfile = exports.PartnerCampaigns = exports.Messaging10dlc = exports.CampaignBuilder = exports.Campaign = exports.Brand = void 0;
var index_1 = require("./brand/index.js");
Object.defineProperty(exports, "Brand", { enumerable: true, get: function () { return index_1.Brand; } });
var index_2 = require("./campaign/index.js");
Object.defineProperty(exports, "Campaign", { enumerable: true, get: function () { return index_2.Campaign; } });
var index_3 = require("./campaign-builder/index.js");
Object.defineProperty(exports, "CampaignBuilder", { enumerable: true, get: function () { return index_3.CampaignBuilder; } });
var messaging_10dlc_1 = require("./messaging-10dlc.js");
Object.defineProperty(exports, "Messaging10dlc", { enumerable: true, get: function () { return messaging_10dlc_1.Messaging10dlc; } });
var partner_campaigns_1 = require("./partner-campaigns.js");
Object.defineProperty(exports, "PartnerCampaigns", { enumerable: true, get: function () { return partner_campaigns_1.PartnerCampaigns; } });
var phone_number_assignment_by_profile_1 = require("./phone-number-assignment-by-profile.js");
Object.defineProperty(exports, "PhoneNumberAssignmentByProfile", { enumerable: true, get: function () { return phone_number_assignment_by_profile_1.PhoneNumberAssignmentByProfile; } });
var phone_number_campaigns_1 = require("./phone-number-campaigns.js");
Object.defineProperty(exports, "PhoneNumberCampaigns", { enumerable: true, get: function () { return phone_number_campaigns_1.PhoneNumberCampaigns; } });
