// telnyx/resources/messaging-10dlc/messaging-10dlc.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as PartnerCampaignsAPI from "./partner-campaigns.mjs";
import { PartnerCampaigns, } from "./partner-campaigns.mjs";
import * as PhoneNumberAssignmentByProfileAPI from "./phone-number-assignment-by-profile.mjs";
import { PhoneNumberAssignmentByProfile, } from "./phone-number-assignment-by-profile.mjs";
import * as PhoneNumberCampaignsAPI from "./phone-number-campaigns.mjs";
import { PhoneNumberCampaigns, } from "./phone-number-campaigns.mjs";
import * as BrandAPI from "./brand/brand.mjs";
import { Brand, } from "./brand/brand.mjs";
import * as CampaignBuilderAPI from "./campaign-builder/campaign-builder.mjs";
import { CampaignBuilder } from "./campaign-builder/campaign-builder.mjs";
import * as CampaignAPI from "./campaign/campaign.mjs";
import { Campaign, } from "./campaign/campaign.mjs";
import { path } from "../../internal/utils/path.mjs";
export class Messaging10dlc extends APIResource {
    constructor() {
        super(...arguments);
        this.brand = new BrandAPI.Brand(this._client);
        this.campaign = new CampaignAPI.Campaign(this._client);
        this.campaignBuilder = new CampaignBuilderAPI.CampaignBuilder(this._client);
        this.partnerCampaigns = new PartnerCampaignsAPI.PartnerCampaigns(this._client);
        this.phoneNumberCampaigns = new PhoneNumberCampaignsAPI.PhoneNumberCampaigns(this._client);
        this.phoneNumberAssignmentByProfile = new PhoneNumberAssignmentByProfileAPI.PhoneNumberAssignmentByProfile(this._client);
    }
    /**
     * Get Enum
     *
     * @example
     * ```ts
     * const response = await client.messaging10dlc.getEnum('mno');
     * ```
     */
    getEnum(endpoint, options) {
        return this._client.get(path `/10dlc/enum/${endpoint}`, options);
    }
}
Messaging10dlc.Brand = Brand;
Messaging10dlc.Campaign = Campaign;
Messaging10dlc.CampaignBuilder = CampaignBuilder;
Messaging10dlc.PartnerCampaigns = PartnerCampaigns;
Messaging10dlc.PhoneNumberCampaigns = PhoneNumberCampaigns;
Messaging10dlc.PhoneNumberAssignmentByProfile = PhoneNumberAssignmentByProfile;
