// telnyx/resources/messaging-10dlc/messaging-10dlc.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Messaging10dlc = void 0;
const tslib_1 = require("../../internal/tslib.js");
const resource_1 = require("../../core/resource.js");
const PartnerCampaignsAPI = tslib_1.__importStar(require("./partner-campaigns.js"));
const partner_campaigns_1 = require("./partner-campaigns.js");
const PhoneNumberAssignmentByProfileAPI = tslib_1.__importStar(require("./phone-number-assignment-by-profile.js"));
const phone_number_assignment_by_profile_1 = require("./phone-number-assignment-by-profile.js");
const PhoneNumberCampaignsAPI = tslib_1.__importStar(require("./phone-number-campaigns.js"));
const phone_number_campaigns_1 = require("./phone-number-campaigns.js");
const BrandAPI = tslib_1.__importStar(require("./brand/brand.js"));
const brand_1 = require("./brand/brand.js");
const CampaignBuilderAPI = tslib_1.__importStar(require("./campaign-builder/campaign-builder.js"));
const campaign_builder_1 = require("./campaign-builder/campaign-builder.js");
const CampaignAPI = tslib_1.__importStar(require("./campaign/campaign.js"));
const campaign_1 = require("./campaign/campaign.js");
const path_1 = require("../../internal/utils/path.js");
class Messaging10dlc extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.brand = new BrandAPI.Brand(this._client);
        this.campaign = new CampaignAPI.Campaign(this._client);
        this.campaignBuilder = new CampaignBuilderAPI.CampaignBuilder(this._client);
        this.partnerCampaigns = new PartnerCampaignsAPI.PartnerCampaigns(this._client);
        this.phoneNumberCampaigns = new PhoneNumberCampaignsAPI.PhoneNumberCampaigns(this._client);
        this.phoneNumberAssignmentByProfile = new PhoneNumberAssignmentByProfileAPI.PhoneNumberAssignmentByProfile(this._client);
    }
    /**
     * Get Enum
     *
     * @example
     * ```ts
     * const response = await client.messaging10dlc.getEnum('mno');
     * ```
     */
    getEnum(endpoint, options) {
        return this._client.get((0, path_1.path) `/10dlc/enum/${endpoint}`, options);
    }
}
exports.Messaging10dlc = Messaging10dlc;
Messaging10dlc.Brand = brand_1.Brand;
Messaging10dlc.Campaign = campaign_1.Campaign;
Messaging10dlc.CampaignBuilder = campaign_builder_1.CampaignBuilder;
Messaging10dlc.PartnerCampaigns = partner_campaigns_1.PartnerCampaigns;
Messaging10dlc.PhoneNumberCampaigns = phone_number_campaigns_1.PhoneNumberCampaigns;
Messaging10dlc.PhoneNumberAssignmentByProfile = phone_number_assignment_by_profile_1.PhoneNumberAssignmentByProfile;
