// telnyx/resources/messaging-10dlc/partner-campaigns.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.PartnerCampaigns = void 0;
const resource_1 = require("../../core/resource.js");
const pagination_1 = require("../../core/pagination.js");
const path_1 = require("../../internal/utils/path.js");
class PartnerCampaigns extends resource_1.APIResource {
    /**
     * Retrieve campaign details by `campaignId`.
     *
     * @example
     * ```ts
     * const telnyxDownstreamCampaign =
     *   await client.messaging10dlc.partnerCampaigns.retrieve(
     *     'campaignId',
     *   );
     * ```
     */
    retrieve(campaignID, options) {
        return this._client.get((0, path_1.path) `/10dlc/partner_campaigns/${campaignID}`, options);
    }
    /**
     * Update campaign details by `campaignId`. **Please note:** Only webhook urls are
     * editable.
     *
     * @example
     * ```ts
     * const telnyxDownstreamCampaign =
     *   await client.messaging10dlc.partnerCampaigns.update(
     *     'campaignId',
     *   );
     * ```
     */
    update(campaignID, body, options) {
        return this._client.patch((0, path_1.path) `/10dlc/partner_campaigns/${campaignID}`, { body, ...options });
    }
    /**
     * Retrieve all partner campaigns you have shared to Telnyx in a paginated fashion.
     *
     * This endpoint is currently limited to only returning shared campaigns that
     * Telnyx has accepted. In other words, shared but pending campaigns are currently
     * omitted from the response from this endpoint.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const telnyxDownstreamCampaign of client.messaging10dlc.partnerCampaigns.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/10dlc/partner_campaigns', (pagination_1.PerPagePaginationV2), { query, ...options });
    }
    /**
     * Get all partner campaigns you have shared to Telnyx in a paginated fashion
     *
     * This endpoint is currently limited to only returning shared campaigns that
     * Telnyx has accepted. In other words, shared but pending campaigns are currently
     * omitted from the response from this endpoint.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const partnerCampaignListSharedByMeResponse of client.messaging10dlc.partnerCampaigns.listSharedByMe()) {
     *   // ...
     * }
     * ```
     */
    listSharedByMe(query = {}, options) {
        return this._client.getAPIList('/10dlc/partnerCampaign/sharedByMe', (pagination_1.PerPagePaginationV2), { query, ...options });
    }
    /**
     * Get Sharing Status
     *
     * @example
     * ```ts
     * const response =
     *   await client.messaging10dlc.partnerCampaigns.retrieveSharingStatus(
     *     'campaignId',
     *   );
     * ```
     */
    retrieveSharingStatus(campaignID, options) {
        return this._client.get((0, path_1.path) `/10dlc/partnerCampaign/${campaignID}/sharing`, options);
    }
}
exports.PartnerCampaigns = PartnerCampaigns;
