// telnyx/resources/messaging-10dlc/phone-number-campaigns.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.PhoneNumberCampaigns = void 0;
const resource_1 = require("../../core/resource.js");
const pagination_1 = require("../../core/pagination.js");
const path_1 = require("../../internal/utils/path.js");
/**
 * Phone number campaign assignment
 */
class PhoneNumberCampaigns extends resource_1.APIResource {
    /**
     * Create New Phone Number Campaign
     *
     * @example
     * ```ts
     * const phoneNumberCampaign =
     *   await client.messaging10dlc.phoneNumberCampaigns.create({
     *     campaignId: '4b300178-131c-d902-d54e-72d90ba1620j',
     *     phoneNumber: '+18005550199',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/10dlc/phone_number_campaigns', { body, ...options });
    }
    /**
     * Retrieve an individual phone number/campaign assignment by `phoneNumber`.
     *
     * @example
     * ```ts
     * const phoneNumberCampaign =
     *   await client.messaging10dlc.phoneNumberCampaigns.retrieve(
     *     'phoneNumber',
     *   );
     * ```
     */
    retrieve(phoneNumber, options) {
        return this._client.get((0, path_1.path) `/10dlc/phone_number_campaigns/${phoneNumber}`, options);
    }
    /**
     * Create New Phone Number Campaign
     *
     * @example
     * ```ts
     * const phoneNumberCampaign =
     *   await client.messaging10dlc.phoneNumberCampaigns.update(
     *     'phoneNumber',
     *     {
     *       campaignId: '4b300178-131c-d902-d54e-72d90ba1620j',
     *       phoneNumber: '+18005550199',
     *     },
     *   );
     * ```
     */
    update(campaignPhoneNumber, body, options) {
        return this._client.put((0, path_1.path) `/10dlc/phone_number_campaigns/${campaignPhoneNumber}`, { body, ...options });
    }
    /**
     * List phone number campaigns
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const phoneNumberCampaign of client.messaging10dlc.phoneNumberCampaigns.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/10dlc/phone_number_campaigns', (pagination_1.PerPagePaginationV2), { query, ...options });
    }
    /**
     * This endpoint allows you to remove a campaign assignment from the supplied
     * `phoneNumber`.
     *
     * @example
     * ```ts
     * const phoneNumberCampaign =
     *   await client.messaging10dlc.phoneNumberCampaigns.delete(
     *     'phoneNumber',
     *   );
     * ```
     */
    delete(phoneNumber, options) {
        return this._client.delete((0, path_1.path) `/10dlc/phone_number_campaigns/${phoneNumber}`, options);
    }
}
exports.PhoneNumberCampaigns = PhoneNumberCampaigns;
