// telnyx/resources/ai/assistants/tests/tests.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../../core/resource.mjs";
import * as RunsAPI from "./runs.mjs";
import { Runs, } from "./runs.mjs";
import * as TestSuitesAPI from "./test-suites/test-suites.mjs";
import { TestSuites } from "./test-suites/test-suites.mjs";
import { DefaultFlatPagination, } from "../../../../core/pagination.mjs";
import { buildHeaders } from "../../../../internal/headers.mjs";
import { path } from "../../../../internal/utils/path.mjs";
/**
 * Configure AI assistant specifications
 */
export class Tests extends APIResource {
    constructor() {
        super(...arguments);
        this.testSuites = new TestSuitesAPI.TestSuites(this._client);
        this.runs = new RunsAPI.Runs(this._client);
    }
    /**
     * Creates a comprehensive test configuration for evaluating AI assistant
     * performance
     *
     * @example
     * ```ts
     * const assistantTest =
     *   await client.ai.assistants.tests.create({
     *     destination: '+15551234567',
     *     instructions:
     *       'Act as a frustrated customer who received a damaged product. Ask for a refund and escalate if not satisfied with the initial response.',
     *     name: 'Customer Support Bot Test',
     *     rubric: [
     *       {
     *         criteria: 'Assistant responds within 30 seconds',
     *         name: 'Response Time',
     *       },
     *       {
     *         criteria: 'Provides correct product information',
     *         name: 'Accuracy',
     *       },
     *     ],
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/ai/assistants/tests', { body, ...options });
    }
    /**
     * Retrieves detailed information about a specific assistant test
     *
     * @example
     * ```ts
     * const assistantTest =
     *   await client.ai.assistants.tests.retrieve('test_id');
     * ```
     */
    retrieve(testID, options) {
        return this._client.get(path `/ai/assistants/tests/${testID}`, options);
    }
    /**
     * Updates an existing assistant test configuration with new settings
     *
     * @example
     * ```ts
     * const assistantTest =
     *   await client.ai.assistants.tests.update('test_id');
     * ```
     */
    update(testID, body, options) {
        return this._client.put(path `/ai/assistants/tests/${testID}`, { body, ...options });
    }
    /**
     * Retrieves a paginated list of assistant tests with optional filtering
     * capabilities
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const assistantTest of client.ai.assistants.tests.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/ai/assistants/tests', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Permanently removes an assistant test and all associated data
     *
     * @example
     * ```ts
     * await client.ai.assistants.tests.delete('test_id');
     * ```
     */
    delete(testID, options) {
        return this._client.delete(path `/ai/assistants/tests/${testID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
Tests.TestSuites = TestSuites;
Tests.Runs = Runs;
