// telnyx/resources/ai/assistants/tests/tests.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tests = void 0;
const tslib_1 = require("../../../../internal/tslib.js");
const resource_1 = require("../../../../core/resource.js");
const RunsAPI = tslib_1.__importStar(require("./runs.js"));
const runs_1 = require("./runs.js");
const TestSuitesAPI = tslib_1.__importStar(require("./test-suites/test-suites.js"));
const test_suites_1 = require("./test-suites/test-suites.js");
const pagination_1 = require("../../../../core/pagination.js");
const headers_1 = require("../../../../internal/headers.js");
const path_1 = require("../../../../internal/utils/path.js");
/**
 * Configure AI assistant specifications
 */
class Tests extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.testSuites = new TestSuitesAPI.TestSuites(this._client);
        this.runs = new RunsAPI.Runs(this._client);
    }
    /**
     * Creates a comprehensive test configuration for evaluating AI assistant
     * performance
     *
     * @example
     * ```ts
     * const assistantTest =
     *   await client.ai.assistants.tests.create({
     *     destination: '+15551234567',
     *     instructions:
     *       'Act as a frustrated customer who received a damaged product. Ask for a refund and escalate if not satisfied with the initial response.',
     *     name: 'Customer Support Bot Test',
     *     rubric: [
     *       {
     *         criteria: 'Assistant responds within 30 seconds',
     *         name: 'Response Time',
     *       },
     *       {
     *         criteria: 'Provides correct product information',
     *         name: 'Accuracy',
     *       },
     *     ],
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/ai/assistants/tests', { body, ...options });
    }
    /**
     * Retrieves detailed information about a specific assistant test
     *
     * @example
     * ```ts
     * const assistantTest =
     *   await client.ai.assistants.tests.retrieve('test_id');
     * ```
     */
    retrieve(testID, options) {
        return this._client.get((0, path_1.path) `/ai/assistants/tests/${testID}`, options);
    }
    /**
     * Updates an existing assistant test configuration with new settings
     *
     * @example
     * ```ts
     * const assistantTest =
     *   await client.ai.assistants.tests.update('test_id');
     * ```
     */
    update(testID, body, options) {
        return this._client.put((0, path_1.path) `/ai/assistants/tests/${testID}`, { body, ...options });
    }
    /**
     * Retrieves a paginated list of assistant tests with optional filtering
     * capabilities
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const assistantTest of client.ai.assistants.tests.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/ai/assistants/tests', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Permanently removes an assistant test and all associated data
     *
     * @example
     * ```ts
     * await client.ai.assistants.tests.delete('test_id');
     * ```
     */
    delete(testID, options) {
        return this._client.delete((0, path_1.path) `/ai/assistants/tests/${testID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.Tests = Tests;
Tests.TestSuites = test_suites_1.TestSuites;
Tests.Runs = runs_1.Runs;
