// telnyx/resources/sim-card-groups/actions.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import { DefaultFlatPagination } from "../../core/pagination.mjs";
import { path } from "../../internal/utils/path.mjs";
export class Actions extends APIResource {
    /**
     * This API allows fetching detailed information about a SIM card group action
     * resource to make follow-ups in an existing asynchronous operation.
     *
     * @example
     * ```ts
     * const action = await client.simCardGroups.actions.retrieve(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/sim_card_group_actions/${id}`, options);
    }
    /**
     * This API allows listing a paginated collection a SIM card group actions. It
     * allows to explore a collection of existing asynchronous operation using specific
     * filters.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const simCardGroupAction of client.simCardGroups.actions.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/sim_card_group_actions', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * This action will asynchronously remove an existing Private Wireless Gateway
     * definition from a SIM card group. Completing this operation defines that all SIM
     * cards in the SIM card group will get their traffic handled by Telnyx's default
     * mobile network configuration.
     *
     * @example
     * ```ts
     * const response =
     *   await client.simCardGroups.actions.removePrivateWirelessGateway(
     *     '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *   );
     * ```
     */
    removePrivateWirelessGateway(id, options) {
        return this._client.post(path `/sim_card_groups/${id}/actions/remove_private_wireless_gateway`, options);
    }
    /**
     * This action will asynchronously remove an existing Wireless Blocklist to all the
     * SIMs in the SIM card group.
     *
     * @example
     * ```ts
     * const response =
     *   await client.simCardGroups.actions.removeWirelessBlocklist(
     *     '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *   );
     * ```
     */
    removeWirelessBlocklist(id, options) {
        return this._client.post(path `/sim_card_groups/${id}/actions/remove_wireless_blocklist`, options);
    }
    /**
     * This action will asynchronously assign a provisioned Private Wireless Gateway to
     * the SIM card group. Completing this operation defines that all SIM cards in the
     * SIM card group will get their traffic controlled by the associated Private
     * Wireless Gateway. This operation will also imply that new SIM cards assigned to
     * a group will inherit its network definitions. If it's moved to a different group
     * that doesn't have a Private Wireless Gateway, it'll use Telnyx's default mobile
     * network configuration.
     *
     * @example
     * ```ts
     * const response =
     *   await client.simCardGroups.actions.setPrivateWirelessGateway(
     *     '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *     {
     *       private_wireless_gateway_id:
     *         '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *     },
     *   );
     * ```
     */
    setPrivateWirelessGateway(id, body, options) {
        return this._client.post(path `/sim_card_groups/${id}/actions/set_private_wireless_gateway`, {
            body,
            ...options,
        });
    }
    /**
     * This action will asynchronously assign a Wireless Blocklist to all the SIMs in
     * the SIM card group.
     *
     * @example
     * ```ts
     * const response =
     *   await client.simCardGroups.actions.setWirelessBlocklist(
     *     '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *     {
     *       wireless_blocklist_id:
     *         '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     *     },
     *   );
     * ```
     */
    setWirelessBlocklist(id, body, options) {
        return this._client.post(path `/sim_card_groups/${id}/actions/set_wireless_blocklist`, {
            body,
            ...options,
        });
    }
}
