// telnyx/resources/sim-card-groups/sim-card-groups.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as ActionsAPI from "./actions.mjs";
import { Actions, } from "./actions.mjs";
import { DefaultFlatPagination } from "../../core/pagination.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * SIM Card Groups operations
 */
export class SimCardGroups extends APIResource {
    constructor() {
        super(...arguments);
        this.actions = new ActionsAPI.Actions(this._client);
    }
    /**
     * Creates a new SIM card group object
     *
     * @example
     * ```ts
     * const simCardGroup = await client.simCardGroups.create({
     *   name: 'My Test Group',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/sim_card_groups', { body, ...options });
    }
    /**
     * Returns the details regarding a specific SIM card group
     *
     * @example
     * ```ts
     * const simCardGroup = await client.simCardGroups.retrieve(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    retrieve(id, query = {}, options) {
        return this._client.get(path `/sim_card_groups/${id}`, { query, ...options });
    }
    /**
     * Updates a SIM card group
     *
     * @example
     * ```ts
     * const simCardGroup = await client.simCardGroups.update(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    update(id, body, options) {
        return this._client.patch(path `/sim_card_groups/${id}`, { body, ...options });
    }
    /**
     * Get all SIM card groups belonging to the user that match the given filters.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const simCardGroupListResponse of client.simCardGroups.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/sim_card_groups', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Permanently deletes a SIM card group
     *
     * @example
     * ```ts
     * const simCardGroup = await client.simCardGroups.delete(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/sim_card_groups/${id}`, options);
    }
}
SimCardGroups.Actions = Actions;
