// telnyx/resources/sim-card-groups/sim-card-groups.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.SimCardGroups = void 0;
const tslib_1 = require("../../internal/tslib.js");
const resource_1 = require("../../core/resource.js");
const ActionsAPI = tslib_1.__importStar(require("./actions.js"));
const actions_1 = require("./actions.js");
const pagination_1 = require("../../core/pagination.js");
const path_1 = require("../../internal/utils/path.js");
/**
 * SIM Card Groups operations
 */
class SimCardGroups extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.actions = new ActionsAPI.Actions(this._client);
    }
    /**
     * Creates a new SIM card group object
     *
     * @example
     * ```ts
     * const simCardGroup = await client.simCardGroups.create({
     *   name: 'My Test Group',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/sim_card_groups', { body, ...options });
    }
    /**
     * Returns the details regarding a specific SIM card group
     *
     * @example
     * ```ts
     * const simCardGroup = await client.simCardGroups.retrieve(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    retrieve(id, query = {}, options) {
        return this._client.get((0, path_1.path) `/sim_card_groups/${id}`, { query, ...options });
    }
    /**
     * Updates a SIM card group
     *
     * @example
     * ```ts
     * const simCardGroup = await client.simCardGroups.update(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    update(id, body, options) {
        return this._client.patch((0, path_1.path) `/sim_card_groups/${id}`, { body, ...options });
    }
    /**
     * Get all SIM card groups belonging to the user that match the given filters.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const simCardGroupListResponse of client.simCardGroups.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/sim_card_groups', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Permanently deletes a SIM card group
     *
     * @example
     * ```ts
     * const simCardGroup = await client.simCardGroups.delete(
     *   '6a09cdc3-8948-47f0-aa62-74ac943d6c58',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/sim_card_groups/${id}`, options);
    }
}
exports.SimCardGroups = SimCardGroups;
SimCardGroups.Actions = actions_1.Actions;
