// telnyx/resources/verifications/verifications.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as ActionsAPI from "./actions.mjs";
import { Actions } from "./actions.mjs";
import * as ByPhoneNumberAPI from "./by-phone-number/by-phone-number.mjs";
import { ByPhoneNumber } from "./by-phone-number/by-phone-number.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * Two factor authentication API
 */
export class Verifications extends APIResource {
    constructor() {
        super(...arguments);
        this.byPhoneNumber = new ByPhoneNumberAPI.ByPhoneNumber(this._client);
        this.actions = new ActionsAPI.Actions(this._client);
    }
    /**
     * Retrieve verification
     *
     * @example
     * ```ts
     * const verification = await client.verifications.retrieve(
     *   '12ade33a-21c0-473b-b055-b3c836e1c292',
     * );
     * ```
     */
    retrieve(verificationID, options) {
        return this._client.get(path `/verifications/${verificationID}`, options);
    }
    /**
     * Trigger Call verification
     *
     * @example
     * ```ts
     * const createVerificationResponse =
     *   await client.verifications.triggerCall({
     *     phone_number: '+13035551234',
     *     verify_profile_id:
     *       '12ade33a-21c0-473b-b055-b3c836e1c292',
     *   });
     * ```
     */
    triggerCall(body, options) {
        return this._client.post('/verifications/call', { body, ...options });
    }
    /**
     * Trigger Flash call verification
     *
     * @example
     * ```ts
     * const createVerificationResponse =
     *   await client.verifications.triggerFlashcall({
     *     phone_number: '+13035551234',
     *     verify_profile_id:
     *       '12ade33a-21c0-473b-b055-b3c836e1c292',
     *   });
     * ```
     */
    triggerFlashcall(body, options) {
        return this._client.post('/verifications/flashcall', { body, ...options });
    }
    /**
     * Trigger SMS verification
     *
     * @example
     * ```ts
     * const createVerificationResponse =
     *   await client.verifications.triggerSMS({
     *     phone_number: '+13035551234',
     *     verify_profile_id:
     *       '12ade33a-21c0-473b-b055-b3c836e1c292',
     *   });
     * ```
     */
    triggerSMS(body, options) {
        return this._client.post('/verifications/sms', { body, ...options });
    }
    /**
     * Trigger WhatsApp verification
     *
     * @example
     * ```ts
     * const createVerificationResponse =
     *   await client.verifications.triggerWhatsappVerification({
     *     phone_number: '+13035551234',
     *     verify_profile_id:
     *       '12ade33a-21c0-473b-b055-b3c836e1c292',
     *   });
     * ```
     */
    triggerWhatsappVerification(body, options) {
        return this._client.post('/verifications/whatsapp', { body, ...options });
    }
}
Verifications.ByPhoneNumber = ByPhoneNumber;
Verifications.Actions = Actions;
