// telnyx/resources/verifications/verifications.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Verifications = void 0;
const tslib_1 = require("../../internal/tslib.js");
const resource_1 = require("../../core/resource.js");
const ActionsAPI = tslib_1.__importStar(require("./actions.js"));
const actions_1 = require("./actions.js");
const ByPhoneNumberAPI = tslib_1.__importStar(require("./by-phone-number/by-phone-number.js"));
const by_phone_number_1 = require("./by-phone-number/by-phone-number.js");
const path_1 = require("../../internal/utils/path.js");
/**
 * Two factor authentication API
 */
class Verifications extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.byPhoneNumber = new ByPhoneNumberAPI.ByPhoneNumber(this._client);
        this.actions = new ActionsAPI.Actions(this._client);
    }
    /**
     * Retrieve verification
     *
     * @example
     * ```ts
     * const verification = await client.verifications.retrieve(
     *   '12ade33a-21c0-473b-b055-b3c836e1c292',
     * );
     * ```
     */
    retrieve(verificationID, options) {
        return this._client.get((0, path_1.path) `/verifications/${verificationID}`, options);
    }
    /**
     * Trigger Call verification
     *
     * @example
     * ```ts
     * const createVerificationResponse =
     *   await client.verifications.triggerCall({
     *     phone_number: '+13035551234',
     *     verify_profile_id:
     *       '12ade33a-21c0-473b-b055-b3c836e1c292',
     *   });
     * ```
     */
    triggerCall(body, options) {
        return this._client.post('/verifications/call', { body, ...options });
    }
    /**
     * Trigger Flash call verification
     *
     * @example
     * ```ts
     * const createVerificationResponse =
     *   await client.verifications.triggerFlashcall({
     *     phone_number: '+13035551234',
     *     verify_profile_id:
     *       '12ade33a-21c0-473b-b055-b3c836e1c292',
     *   });
     * ```
     */
    triggerFlashcall(body, options) {
        return this._client.post('/verifications/flashcall', { body, ...options });
    }
    /**
     * Trigger SMS verification
     *
     * @example
     * ```ts
     * const createVerificationResponse =
     *   await client.verifications.triggerSMS({
     *     phone_number: '+13035551234',
     *     verify_profile_id:
     *       '12ade33a-21c0-473b-b055-b3c836e1c292',
     *   });
     * ```
     */
    triggerSMS(body, options) {
        return this._client.post('/verifications/sms', { body, ...options });
    }
    /**
     * Trigger WhatsApp verification
     *
     * @example
     * ```ts
     * const createVerificationResponse =
     *   await client.verifications.triggerWhatsappVerification({
     *     phone_number: '+13035551234',
     *     verify_profile_id:
     *       '12ade33a-21c0-473b-b055-b3c836e1c292',
     *   });
     * ```
     */
    triggerWhatsappVerification(body, options) {
        return this._client.post('/verifications/whatsapp', { body, ...options });
    }
}
exports.Verifications = Verifications;
Verifications.ByPhoneNumber = by_phone_number_1.ByPhoneNumber;
Verifications.Actions = actions_1.Actions;
