// telnyx/resources/texml/accounts/calls/calls.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Calls = void 0;
const tslib_1 = require("../../../../internal/tslib.js");
const resource_1 = require("../../../../core/resource.js");
const RecordingsAPI = tslib_1.__importStar(require("./recordings.js"));
const recordings_1 = require("./recordings.js");
const RecordingsJsonAPI = tslib_1.__importStar(require("./recordings-json.js"));
const recordings_json_1 = require("./recordings-json.js");
const SiprecAPI = tslib_1.__importStar(require("./siprec.js"));
const siprec_1 = require("./siprec.js");
const StreamsAPI = tslib_1.__importStar(require("./streams.js"));
const streams_1 = require("./streams.js");
const headers_1 = require("../../../../internal/headers.js");
const path_1 = require("../../../../internal/utils/path.js");
/**
 * TeXML REST Commands
 */
class Calls extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.recordingsJson = new RecordingsJsonAPI.RecordingsJson(this._client);
        this.recordings = new RecordingsAPI.Recordings(this._client);
        this.siprec = new SiprecAPI.Siprec(this._client);
        this.streams = new StreamsAPI.Streams(this._client);
    }
    /**
     * Returns an individual call identified by its CallSid. This endpoint is
     * eventually consistent.
     *
     * @example
     * ```ts
     * const call = await client.texml.accounts.calls.retrieve(
     *   'call_sid',
     *   { account_sid: 'account_sid' },
     * );
     * ```
     */
    retrieve(callSid, params, options) {
        const { account_sid } = params;
        return this._client.get((0, path_1.path) `/texml/Accounts/${account_sid}/Calls/${callSid}`, options);
    }
    /**
     * Update TeXML call. Please note that the keys present in the payload MUST BE
     * formatted in CamelCase as specified in the example.
     *
     * @example
     * ```ts
     * const call = await client.texml.accounts.calls.update(
     *   'call_sid',
     *   { account_sid: 'account_sid' },
     * );
     * ```
     */
    update(callSid, params, options) {
        const { account_sid, ...body } = params;
        return this._client.post((0, path_1.path) `/texml/Accounts/${account_sid}/Calls/${callSid}`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ 'Content-Type': 'application/x-www-form-urlencoded' }, options?.headers]),
        });
    }
    /**
     * Initiate an outbound TeXML call. Telnyx will request TeXML from the XML Request
     * URL configured for the connection in the Mission Control Portal.
     *
     * @example
     * ```ts
     * const response = await client.texml.accounts.calls.calls(
     *   'account_sid',
     *   {
     *     params: {
     *       ApplicationSid: 'example-app-sid',
     *       From: '+13120001234',
     *       To: '+13121230000',
     *     },
     *   },
     * );
     * ```
     */
    calls(accountSid, args, options) {
        const { params } = args;
        return this._client.post((0, path_1.path) `/texml/Accounts/${accountSid}/Calls`, { body: params, ...options });
    }
    /**
     * Returns multiple call resouces for an account. This endpoint is eventually
     * consistent.
     *
     * @example
     * ```ts
     * const response =
     *   await client.texml.accounts.calls.retrieveCalls(
     *     'account_sid',
     *   );
     * ```
     */
    retrieveCalls(accountSid, query = {}, options) {
        return this._client.get((0, path_1.path) `/texml/Accounts/${accountSid}/Calls`, { query, ...options });
    }
    /**
     * Starts siprec session with specified parameters for call idientified by
     * call_sid.
     *
     * @example
     * ```ts
     * const response =
     *   await client.texml.accounts.calls.siprecJson('call_sid', {
     *     account_sid: 'account_sid',
     *   });
     * ```
     */
    siprecJson(callSid, params, options) {
        const { account_sid, ...body } = params;
        return this._client.post((0, path_1.path) `/texml/Accounts/${account_sid}/Calls/${callSid}/Siprec.json`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ 'Content-Type': 'application/x-www-form-urlencoded' }, options?.headers]),
        });
    }
    /**
     * Starts streaming media from a call to a specific WebSocket address.
     *
     * @example
     * ```ts
     * const response =
     *   await client.texml.accounts.calls.streamsJson(
     *     'call_sid',
     *     { account_sid: 'account_sid' },
     *   );
     * ```
     */
    streamsJson(callSid, params, options) {
        const { account_sid, ...body } = params;
        return this._client.post((0, path_1.path) `/texml/Accounts/${account_sid}/Calls/${callSid}/Streams.json`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ 'Content-Type': 'application/x-www-form-urlencoded' }, options?.headers]),
        });
    }
}
exports.Calls = Calls;
Calls.RecordingsJson = recordings_json_1.RecordingsJson;
Calls.Recordings = recordings_1.Recordings;
Calls.Siprec = siprec_1.Siprec;
Calls.Streams = streams_1.Streams;
