// telnyx/resources/texml/accounts/calls/calls.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../../core/resource.mjs";
import * as RecordingsAPI from "./recordings.mjs";
import { Recordings } from "./recordings.mjs";
import * as RecordingsJsonAPI from "./recordings-json.mjs";
import { RecordingsJson, } from "./recordings-json.mjs";
import * as SiprecAPI from "./siprec.mjs";
import { Siprec } from "./siprec.mjs";
import * as StreamsAPI from "./streams.mjs";
import { Streams } from "./streams.mjs";
import { buildHeaders } from "../../../../internal/headers.mjs";
import { path } from "../../../../internal/utils/path.mjs";
/**
 * TeXML REST Commands
 */
export class Calls extends APIResource {
    constructor() {
        super(...arguments);
        this.recordingsJson = new RecordingsJsonAPI.RecordingsJson(this._client);
        this.recordings = new RecordingsAPI.Recordings(this._client);
        this.siprec = new SiprecAPI.Siprec(this._client);
        this.streams = new StreamsAPI.Streams(this._client);
    }
    /**
     * Returns an individual call identified by its CallSid. This endpoint is
     * eventually consistent.
     *
     * @example
     * ```ts
     * const call = await client.texml.accounts.calls.retrieve(
     *   'call_sid',
     *   { account_sid: 'account_sid' },
     * );
     * ```
     */
    retrieve(callSid, params, options) {
        const { account_sid } = params;
        return this._client.get(path `/texml/Accounts/${account_sid}/Calls/${callSid}`, options);
    }
    /**
     * Update TeXML call. Please note that the keys present in the payload MUST BE
     * formatted in CamelCase as specified in the example.
     *
     * @example
     * ```ts
     * const call = await client.texml.accounts.calls.update(
     *   'call_sid',
     *   { account_sid: 'account_sid' },
     * );
     * ```
     */
    update(callSid, params, options) {
        const { account_sid, ...body } = params;
        return this._client.post(path `/texml/Accounts/${account_sid}/Calls/${callSid}`, {
            body,
            ...options,
            headers: buildHeaders([{ 'Content-Type': 'application/x-www-form-urlencoded' }, options?.headers]),
        });
    }
    /**
     * Initiate an outbound TeXML call. Telnyx will request TeXML from the XML Request
     * URL configured for the connection in the Mission Control Portal.
     *
     * @example
     * ```ts
     * const response = await client.texml.accounts.calls.calls(
     *   'account_sid',
     *   {
     *     params: {
     *       ApplicationSid: 'example-app-sid',
     *       From: '+13120001234',
     *       To: '+13121230000',
     *     },
     *   },
     * );
     * ```
     */
    calls(accountSid, args, options) {
        const { params } = args;
        return this._client.post(path `/texml/Accounts/${accountSid}/Calls`, { body: params, ...options });
    }
    /**
     * Returns multiple call resouces for an account. This endpoint is eventually
     * consistent.
     *
     * @example
     * ```ts
     * const response =
     *   await client.texml.accounts.calls.retrieveCalls(
     *     'account_sid',
     *   );
     * ```
     */
    retrieveCalls(accountSid, query = {}, options) {
        return this._client.get(path `/texml/Accounts/${accountSid}/Calls`, { query, ...options });
    }
    /**
     * Starts siprec session with specified parameters for call idientified by
     * call_sid.
     *
     * @example
     * ```ts
     * const response =
     *   await client.texml.accounts.calls.siprecJson('call_sid', {
     *     account_sid: 'account_sid',
     *   });
     * ```
     */
    siprecJson(callSid, params, options) {
        const { account_sid, ...body } = params;
        return this._client.post(path `/texml/Accounts/${account_sid}/Calls/${callSid}/Siprec.json`, {
            body,
            ...options,
            headers: buildHeaders([{ 'Content-Type': 'application/x-www-form-urlencoded' }, options?.headers]),
        });
    }
    /**
     * Starts streaming media from a call to a specific WebSocket address.
     *
     * @example
     * ```ts
     * const response =
     *   await client.texml.accounts.calls.streamsJson(
     *     'call_sid',
     *     { account_sid: 'account_sid' },
     *   );
     * ```
     */
    streamsJson(callSid, params, options) {
        const { account_sid, ...body } = params;
        return this._client.post(path `/texml/Accounts/${account_sid}/Calls/${callSid}/Streams.json`, {
            body,
            ...options,
            headers: buildHeaders([{ 'Content-Type': 'application/x-www-form-urlencoded' }, options?.headers]),
        });
    }
}
Calls.RecordingsJson = RecordingsJson;
Calls.Recordings = Recordings;
Calls.Siprec = Siprec;
Calls.Streams = Streams;
