// telnyx/resources/legacy/reporting/batch-detail-records/voice.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../../core/resource.mjs";
import { path } from "../../../../internal/utils/path.mjs";
/**
 * Voice batch detail records
 */
export class Voice extends APIResource {
    /**
     * Creates a new CDR report request with the specified filters
     *
     * @example
     * ```ts
     * const voice =
     *   await client.legacy.reporting.batchDetailRecords.voice.create(
     *     {
     *       end_time: '2024-02-12T23:59:59Z',
     *       start_time: '2024-02-01T00:00:00Z',
     *     },
     *   );
     * ```
     */
    create(body, options) {
        return this._client.post('/legacy/reporting/batch_detail_records/voice', { body, ...options });
    }
    /**
     * Retrieves a specific CDR report request by ID
     *
     * @example
     * ```ts
     * const voice =
     *   await client.legacy.reporting.batchDetailRecords.voice.retrieve(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/legacy/reporting/batch_detail_records/voice/${id}`, options);
    }
    /**
     * Retrieves all CDR report requests for the authenticated user
     *
     * @example
     * ```ts
     * const voices =
     *   await client.legacy.reporting.batchDetailRecords.voice.list();
     * ```
     */
    list(options) {
        return this._client.get('/legacy/reporting/batch_detail_records/voice', options);
    }
    /**
     * Deletes a specific CDR report request by ID
     *
     * @example
     * ```ts
     * const voice =
     *   await client.legacy.reporting.batchDetailRecords.voice.delete(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/legacy/reporting/batch_detail_records/voice/${id}`, options);
    }
    /**
     * Retrieves all available fields that can be used in CDR reports
     *
     * @example
     * ```ts
     * const response =
     *   await client.legacy.reporting.batchDetailRecords.voice.retrieveFields();
     * ```
     */
    retrieveFields(options) {
        return this._client.get('/legacy/reporting/batch_detail_records/voice/fields', options);
    }
}
