// telnyx/resources/portouts/portouts.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Portouts = void 0;
const tslib_1 = require("../../internal/tslib.js");
const resource_1 = require("../../core/resource.js");
const CommentsAPI = tslib_1.__importStar(require("./comments.js"));
const comments_1 = require("./comments.js");
const EventsAPI = tslib_1.__importStar(require("./events.js"));
const events_1 = require("./events.js");
const ReportsAPI = tslib_1.__importStar(require("./reports.js"));
const reports_1 = require("./reports.js");
const SupportingDocumentsAPI = tslib_1.__importStar(require("./supporting-documents.js"));
const supporting_documents_1 = require("./supporting-documents.js");
const pagination_1 = require("../../core/pagination.js");
const path_1 = require("../../internal/utils/path.js");
/**
 * Number portout operations
 */
class Portouts extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.events = new EventsAPI.Events(this._client);
        this.reports = new ReportsAPI.Reports(this._client);
        this.comments = new CommentsAPI.Comments(this._client);
        this.supportingDocuments = new SupportingDocumentsAPI.SupportingDocuments(this._client);
    }
    /**
     * Returns the portout request based on the ID provided
     *
     * @example
     * ```ts
     * const portout = await client.portouts.retrieve(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/portouts/${id}`, options);
    }
    /**
     * Returns the portout requests according to filters
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const portoutDetails of client.portouts.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/portouts', (pagination_1.DefaultFlatPagination), { query, ...options });
    }
    /**
     * Given a port-out ID, list rejection codes that are eligible for that port-out
     *
     * @example
     * ```ts
     * const response = await client.portouts.listRejectionCodes(
     *   '329d6658-8f93-405d-862f-648776e8afd7',
     * );
     * ```
     */
    listRejectionCodes(portoutID, query = {}, options) {
        return this._client.get((0, path_1.path) `/portouts/rejections/${portoutID}`, { query, ...options });
    }
    /**
     * Authorize or reject portout request
     *
     * @example
     * ```ts
     * const response = await client.portouts.updateStatus(
     *   'authorized',
     *   {
     *     id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     reason: 'I do not recognize this transaction',
     *   },
     * );
     * ```
     */
    updateStatus(status, params, options) {
        const { id, ...body } = params;
        return this._client.patch((0, path_1.path) `/portouts/${id}/${status}`, { body, ...options });
    }
}
exports.Portouts = Portouts;
Portouts.Events = events_1.Events;
Portouts.Reports = reports_1.Reports;
Portouts.Comments = comments_1.Comments;
Portouts.SupportingDocuments = supporting_documents_1.SupportingDocuments;
