// telnyx/resources/portouts/portouts.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as CommentsAPI from "./comments.mjs";
import { Comments, } from "./comments.mjs";
import * as EventsAPI from "./events.mjs";
import { Events, } from "./events.mjs";
import * as ReportsAPI from "./reports.mjs";
import { Reports, } from "./reports.mjs";
import * as SupportingDocumentsAPI from "./supporting-documents.mjs";
import { SupportingDocuments, } from "./supporting-documents.mjs";
import { DefaultFlatPagination } from "../../core/pagination.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * Number portout operations
 */
export class Portouts extends APIResource {
    constructor() {
        super(...arguments);
        this.events = new EventsAPI.Events(this._client);
        this.reports = new ReportsAPI.Reports(this._client);
        this.comments = new CommentsAPI.Comments(this._client);
        this.supportingDocuments = new SupportingDocumentsAPI.SupportingDocuments(this._client);
    }
    /**
     * Returns the portout request based on the ID provided
     *
     * @example
     * ```ts
     * const portout = await client.portouts.retrieve(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/portouts/${id}`, options);
    }
    /**
     * Returns the portout requests according to filters
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const portoutDetails of client.portouts.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/portouts', (DefaultFlatPagination), { query, ...options });
    }
    /**
     * Given a port-out ID, list rejection codes that are eligible for that port-out
     *
     * @example
     * ```ts
     * const response = await client.portouts.listRejectionCodes(
     *   '329d6658-8f93-405d-862f-648776e8afd7',
     * );
     * ```
     */
    listRejectionCodes(portoutID, query = {}, options) {
        return this._client.get(path `/portouts/rejections/${portoutID}`, { query, ...options });
    }
    /**
     * Authorize or reject portout request
     *
     * @example
     * ```ts
     * const response = await client.portouts.updateStatus(
     *   'authorized',
     *   {
     *     id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     reason: 'I do not recognize this transaction',
     *   },
     * );
     * ```
     */
    updateStatus(status, params, options) {
        const { id, ...body } = params;
        return this._client.patch(path `/portouts/${id}/${status}`, { body, ...options });
    }
}
Portouts.Events = Events;
Portouts.Reports = Reports;
Portouts.Comments = Comments;
Portouts.SupportingDocuments = SupportingDocuments;
