// telnyx/resources/organizations/users/users.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../core/resource.mjs";
import * as ActionsAPI from "./actions.mjs";
import { Actions } from "./actions.mjs";
import { DefaultFlatPagination, } from "../../../core/pagination.mjs";
import { buildHeaders } from "../../../internal/headers.mjs";
import { path } from "../../../internal/utils/path.mjs";
/**
 * Operations related to users in your organization
 */
export class Users extends APIResource {
    constructor() {
        super(...arguments);
        this.actions = new ActionsAPI.Actions(this._client);
    }
    /**
     * Returns a user in your organization.
     */
    retrieve(id, query = {}, options) {
        return this._client.get(path `/organizations/users/${id}`, { query, ...options });
    }
    /**
     * Returns a list of the users in your organization.
     */
    list(query = {}, options) {
        return this._client.getAPIList('/organizations/users', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Returns a report of all users in your organization with their group memberships.
     * This endpoint returns all users without pagination and always includes group
     * information. The report can be retrieved in JSON or CSV format by sending
     * specific content-type headers.
     */
    getGroupsReport(params = {}, options) {
        const { Accept } = params ?? {};
        return this._client.get('/organizations/users/users_groups_report', {
            ...options,
            headers: buildHeaders([
                { ...(Accept?.toString() != null ? { Accept: Accept?.toString() } : undefined) },
                options?.headers,
            ]),
        });
    }
}
Users.Actions = Actions;
