// telnyx/resources/organizations/users/users.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Users = void 0;
const tslib_1 = require("../../../internal/tslib.js");
const resource_1 = require("../../../core/resource.js");
const ActionsAPI = tslib_1.__importStar(require("./actions.js"));
const actions_1 = require("./actions.js");
const pagination_1 = require("../../../core/pagination.js");
const headers_1 = require("../../../internal/headers.js");
const path_1 = require("../../../internal/utils/path.js");
/**
 * Operations related to users in your organization
 */
class Users extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.actions = new ActionsAPI.Actions(this._client);
    }
    /**
     * Returns a user in your organization.
     */
    retrieve(id, query = {}, options) {
        return this._client.get((0, path_1.path) `/organizations/users/${id}`, { query, ...options });
    }
    /**
     * Returns a list of the users in your organization.
     */
    list(query = {}, options) {
        return this._client.getAPIList('/organizations/users', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Returns a report of all users in your organization with their group memberships.
     * This endpoint returns all users without pagination and always includes group
     * information. The report can be retrieved in JSON or CSV format by sending
     * specific content-type headers.
     */
    getGroupsReport(params = {}, options) {
        const { Accept } = params ?? {};
        return this._client.get('/organizations/users/users_groups_report', {
            ...options,
            headers: (0, headers_1.buildHeaders)([
                { ...(Accept?.toString() != null ? { Accept: Accept?.toString() } : undefined) },
                options?.headers,
            ]),
        });
    }
}
exports.Users = Users;
Users.Actions = actions_1.Actions;
