// telnyx/resources/faxes/faxes.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Faxes = void 0;
const tslib_1 = require("../../internal/tslib.js");
const resource_1 = require("../../core/resource.js");
const ActionsAPI = tslib_1.__importStar(require("./actions.js"));
const actions_1 = require("./actions.js");
const pagination_1 = require("../../core/pagination.js");
const headers_1 = require("../../internal/headers.js");
const uploads_1 = require("../../internal/uploads.js");
const path_1 = require("../../internal/utils/path.js");
/**
 * Programmable fax command operations
 */
class Faxes extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.actions = new ActionsAPI.Actions(this._client);
    }
    /**
     * Send a fax. Files have size limits and page count limit validations. If a file
     * is bigger than 50MB or has more than 350 pages it will fail with
     * `file_size_limit_exceeded` and `page_count_limit_exceeded` respectively.
     *
     * **Supported file formats:**
     *
     * - PDF (`application/pdf`)
     * - TIFF (`application/tiff`, `image/tiff`)
     * - JPEG (`image/jpeg`)
     * - PNG (`image/png`)
     * - Microsoft Word `.doc` (`application/msword`)
     * - Microsoft Word `.docx`
     *   (`application/vnd.openxmlformats-officedocument.wordprocessingml.document`)
     * - Rich Text Format `.rtf` (`application/rtf`)
     * - Plain text `.txt` (`text/plain`)
     *
     * **Expected Webhooks:**
     *
     * - `fax.queued`
     * - `fax.media.processed`
     * - `fax.sending.started`
     * - `fax.delivered`
     * - `fax.failed`
     *
     * @example
     * ```ts
     * const fax = await client.faxes.create({
     *   connection_id: '234423',
     *   from: '+13125790015',
     *   to: '+13127367276',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/faxes', (0, uploads_1.maybeMultipartFormRequestOptions)({ body, ...options }, this._client));
    }
    /**
     * View a fax
     *
     * @example
     * ```ts
     * const fax = await client.faxes.retrieve(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/faxes/${id}`, options);
    }
    /**
     * View a list of faxes
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const fax of client.faxes.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/faxes', (pagination_1.DefaultFlatPagination), { query, ...options });
    }
    /**
     * Delete a fax
     *
     * @example
     * ```ts
     * await client.faxes.delete(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/faxes/${id}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.Faxes = Faxes;
Faxes.Actions = actions_1.Actions;
