// telnyx/resources/faxes/faxes.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as ActionsAPI from "./actions.mjs";
import { Actions } from "./actions.mjs";
import { DefaultFlatPagination } from "../../core/pagination.mjs";
import { buildHeaders } from "../../internal/headers.mjs";
import { maybeMultipartFormRequestOptions } from "../../internal/uploads.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * Programmable fax command operations
 */
export class Faxes extends APIResource {
    constructor() {
        super(...arguments);
        this.actions = new ActionsAPI.Actions(this._client);
    }
    /**
     * Send a fax. Files have size limits and page count limit validations. If a file
     * is bigger than 50MB or has more than 350 pages it will fail with
     * `file_size_limit_exceeded` and `page_count_limit_exceeded` respectively.
     *
     * **Supported file formats:**
     *
     * - PDF (`application/pdf`)
     * - TIFF (`application/tiff`, `image/tiff`)
     * - JPEG (`image/jpeg`)
     * - PNG (`image/png`)
     * - Microsoft Word `.doc` (`application/msword`)
     * - Microsoft Word `.docx`
     *   (`application/vnd.openxmlformats-officedocument.wordprocessingml.document`)
     * - Rich Text Format `.rtf` (`application/rtf`)
     * - Plain text `.txt` (`text/plain`)
     *
     * **Expected Webhooks:**
     *
     * - `fax.queued`
     * - `fax.media.processed`
     * - `fax.sending.started`
     * - `fax.delivered`
     * - `fax.failed`
     *
     * @example
     * ```ts
     * const fax = await client.faxes.create({
     *   connection_id: '234423',
     *   from: '+13125790015',
     *   to: '+13127367276',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/faxes', maybeMultipartFormRequestOptions({ body, ...options }, this._client));
    }
    /**
     * View a fax
     *
     * @example
     * ```ts
     * const fax = await client.faxes.retrieve(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/faxes/${id}`, options);
    }
    /**
     * View a list of faxes
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const fax of client.faxes.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/faxes', (DefaultFlatPagination), { query, ...options });
    }
    /**
     * Delete a fax
     *
     * @example
     * ```ts
     * await client.faxes.delete(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/faxes/${id}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
Faxes.Actions = Actions;
