// telnyx/resources/porting/loa-configurations.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoaConfigurations = void 0;
const resource_1 = require("../../core/resource.js");
const pagination_1 = require("../../core/pagination.js");
const headers_1 = require("../../internal/headers.js");
const path_1 = require("../../internal/utils/path.js");
/**
 * Endpoints related to porting orders management.
 */
class LoaConfigurations extends resource_1.APIResource {
    /**
     * Create a LOA configuration.
     *
     * @example
     * ```ts
     * const loaConfiguration =
     *   await client.porting.loaConfigurations.create({
     *     address: {
     *       city: 'Austin',
     *       country_code: 'US',
     *       state: 'TX',
     *       street_address: '600 Congress Avenue',
     *       zip_code: '78701',
     *     },
     *     company_name: 'Telnyx',
     *     contact: {
     *       email: 'testing@telnyx.com',
     *       phone_number: '+12003270001',
     *     },
     *     logo: {
     *       document_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     },
     *     name: 'My LOA Configuration',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/porting/loa_configurations', { body, ...options });
    }
    /**
     * Retrieve a specific LOA configuration.
     *
     * @example
     * ```ts
     * const loaConfiguration =
     *   await client.porting.loaConfigurations.retrieve(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/porting/loa_configurations/${id}`, options);
    }
    /**
     * Update a specific LOA configuration.
     *
     * @example
     * ```ts
     * const loaConfiguration =
     *   await client.porting.loaConfigurations.update(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     {
     *       address: {
     *         city: 'Austin',
     *         country_code: 'US',
     *         state: 'TX',
     *         street_address: '600 Congress Avenue',
     *         zip_code: '78701',
     *       },
     *       company_name: 'Telnyx',
     *       contact: {
     *         email: 'testing@telnyx.com',
     *         phone_number: '+12003270001',
     *       },
     *       logo: {
     *         document_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *       },
     *       name: 'My LOA Configuration',
     *     },
     *   );
     * ```
     */
    update(id, body, options) {
        return this._client.patch((0, path_1.path) `/porting/loa_configurations/${id}`, { body, ...options });
    }
    /**
     * List the LOA configurations.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const portingLoaConfiguration of client.porting.loaConfigurations.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/porting/loa_configurations', (pagination_1.DefaultFlatPagination), { query, ...options });
    }
    /**
     * Delete a specific LOA configuration.
     *
     * @example
     * ```ts
     * await client.porting.loaConfigurations.delete(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/porting/loa_configurations/${id}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Preview the LOA template that would be generated without need to create LOA
     * configuration.
     *
     * @example
     * ```ts
     * const response =
     *   await client.porting.loaConfigurations.preview({
     *     address: {
     *       city: 'Austin',
     *       country_code: 'US',
     *       state: 'TX',
     *       street_address: '600 Congress Avenue',
     *       zip_code: '78701',
     *     },
     *     company_name: 'Telnyx',
     *     contact: {
     *       email: 'testing@telnyx.com',
     *       phone_number: '+12003270001',
     *     },
     *     logo: {
     *       document_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     },
     *     name: 'My LOA Configuration',
     *   });
     *
     * const content = await response.blob();
     * console.log(content);
     * ```
     */
    preview(body, options) {
        return this._client.post('/porting/loa_configurations/preview', {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: 'application/pdf' }, options?.headers]),
            __binaryResponse: true,
        });
    }
    /**
     * Preview the LOA template that would be generated without need to create LOA
     * configuration.
     *
     * @example
     * ```ts
     * const response =
     *   await client.porting.loaConfigurations.preview0({
     *     address: {
     *       city: 'Austin',
     *       country_code: 'US',
     *       state: 'TX',
     *       street_address: '600 Congress Avenue',
     *       zip_code: '78701',
     *     },
     *     company_name: 'Telnyx',
     *     contact: {
     *       email: 'testing@telnyx.com',
     *       phone_number: '+12003270001',
     *     },
     *     logo: {
     *       document_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     },
     *     name: 'My LOA Configuration',
     *   });
     *
     * const content = await response.blob();
     * console.log(content);
     * ```
     */
    preview0(body, options) {
        return this._client.post('/porting/loa_configurations/preview', {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: 'application/pdf' }, options?.headers]),
            __binaryResponse: true,
        });
    }
    /**
     * Preview a specific LOA configuration.
     *
     * @example
     * ```ts
     * const response =
     *   await client.porting.loaConfigurations.preview1(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     *
     * const content = await response.blob();
     * console.log(content);
     * ```
     */
    preview1(id, options) {
        return this._client.get((0, path_1.path) `/porting/loa_configurations/${id}/preview`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: 'application/pdf' }, options?.headers]),
            __binaryResponse: true,
        });
    }
}
exports.LoaConfigurations = LoaConfigurations;
