// telnyx/resources/texml/accounts/accounts.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Accounts = void 0;
const tslib_1 = require("../../../internal/tslib.js");
const resource_1 = require("../../../core/resource.js");
const QueuesAPI = tslib_1.__importStar(require("./queues.js"));
const queues_1 = require("./queues.js");
const CallsAPI = tslib_1.__importStar(require("./calls/calls.js"));
const calls_1 = require("./calls/calls.js");
const ConferencesAPI = tslib_1.__importStar(require("./conferences/conferences.js"));
const conferences_1 = require("./conferences/conferences.js");
const RecordingsAPI = tslib_1.__importStar(require("./recordings/recordings.js"));
const recordings_1 = require("./recordings/recordings.js");
const TranscriptionsAPI = tslib_1.__importStar(require("./transcriptions/transcriptions.js"));
const transcriptions_1 = require("./transcriptions/transcriptions.js");
const path_1 = require("../../../internal/utils/path.js");
/**
 * TeXML REST Commands
 */
class Accounts extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.calls = new CallsAPI.Calls(this._client);
        this.conferences = new ConferencesAPI.Conferences(this._client);
        this.recordings = new RecordingsAPI.Recordings(this._client);
        this.transcriptions = new TranscriptionsAPI.Transcriptions(this._client);
        this.queues = new QueuesAPI.Queues(this._client);
    }
    /**
     * Returns multiple recording resources for an account.
     *
     * @example
     * ```ts
     * const response =
     *   await client.texml.accounts.retrieveRecordingsJson(
     *     'account_sid',
     *   );
     * ```
     */
    retrieveRecordingsJson(accountSid, query = {}, options) {
        return this._client.get((0, path_1.path) `/texml/Accounts/${accountSid}/Recordings.json`, { query, ...options });
    }
    /**
     * Returns multiple recording transcription resources for an account.
     *
     * @example
     * ```ts
     * const response =
     *   await client.texml.accounts.retrieveTranscriptionsJson(
     *     'account_sid',
     *   );
     * ```
     */
    retrieveTranscriptionsJson(accountSid, query = {}, options) {
        return this._client.get((0, path_1.path) `/texml/Accounts/${accountSid}/Transcriptions.json`, { query, ...options });
    }
}
exports.Accounts = Accounts;
Accounts.Calls = calls_1.Calls;
Accounts.Conferences = conferences_1.Conferences;
Accounts.Recordings = recordings_1.Recordings;
Accounts.Transcriptions = transcriptions_1.Transcriptions;
Accounts.Queues = queues_1.Queues;
