// telnyx/resources/texml/accounts/accounts.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../core/resource.mjs";
import * as QueuesAPI from "./queues.mjs";
import { Queues, } from "./queues.mjs";
import * as CallsAPI from "./calls/calls.mjs";
import { Calls, } from "./calls/calls.mjs";
import * as ConferencesAPI from "./conferences/conferences.mjs";
import { Conferences, } from "./conferences/conferences.mjs";
import * as RecordingsAPI from "./recordings/recordings.mjs";
import { Recordings } from "./recordings/recordings.mjs";
import * as TranscriptionsAPI from "./transcriptions/transcriptions.mjs";
import { Transcriptions } from "./transcriptions/transcriptions.mjs";
import { path } from "../../../internal/utils/path.mjs";
/**
 * TeXML REST Commands
 */
export class Accounts extends APIResource {
    constructor() {
        super(...arguments);
        this.calls = new CallsAPI.Calls(this._client);
        this.conferences = new ConferencesAPI.Conferences(this._client);
        this.recordings = new RecordingsAPI.Recordings(this._client);
        this.transcriptions = new TranscriptionsAPI.Transcriptions(this._client);
        this.queues = new QueuesAPI.Queues(this._client);
    }
    /**
     * Returns multiple recording resources for an account.
     *
     * @example
     * ```ts
     * const response =
     *   await client.texml.accounts.retrieveRecordingsJson(
     *     'account_sid',
     *   );
     * ```
     */
    retrieveRecordingsJson(accountSid, query = {}, options) {
        return this._client.get(path `/texml/Accounts/${accountSid}/Recordings.json`, { query, ...options });
    }
    /**
     * Returns multiple recording transcription resources for an account.
     *
     * @example
     * ```ts
     * const response =
     *   await client.texml.accounts.retrieveTranscriptionsJson(
     *     'account_sid',
     *   );
     * ```
     */
    retrieveTranscriptionsJson(accountSid, query = {}, options) {
        return this._client.get(path `/texml/Accounts/${accountSid}/Transcriptions.json`, { query, ...options });
    }
}
Accounts.Calls = Calls;
Accounts.Conferences = Conferences;
Accounts.Recordings = Recordings;
Accounts.Transcriptions = Transcriptions;
Accounts.Queues = Queues;
