// telnyx/resources/texml/accounts/index.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Transcriptions = exports.Recordings = exports.Queues = exports.Conferences = exports.Calls = exports.Accounts = void 0;
var accounts_1 = require("./accounts.js");
Object.defineProperty(exports, "Accounts", { enumerable: true, get: function () { return accounts_1.Accounts; } });
var index_1 = require("./calls/index.js");
Object.defineProperty(exports, "Calls", { enumerable: true, get: function () { return index_1.Calls; } });
var index_2 = require("./conferences/index.js");
Object.defineProperty(exports, "Conferences", { enumerable: true, get: function () { return index_2.Conferences; } });
var queues_1 = require("./queues.js");
Object.defineProperty(exports, "Queues", { enumerable: true, get: function () { return queues_1.Queues; } });
var index_3 = require("./recordings/index.js");
Object.defineProperty(exports, "Recordings", { enumerable: true, get: function () { return index_3.Recordings; } });
var index_4 = require("./transcriptions/index.js");
Object.defineProperty(exports, "Transcriptions", { enumerable: true, get: function () { return index_4.Transcriptions; } });
