// telnyx/resources/texml/accounts/queues.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../core/resource.mjs";
import { DefaultPaginationForQueues, } from "../../../core/pagination.mjs";
import { buildHeaders } from "../../../internal/headers.mjs";
import { path } from "../../../internal/utils/path.mjs";
/**
 * TeXML REST Commands
 */
export class Queues extends APIResource {
    /**
     * Creates a new queue resource.
     *
     * @example
     * ```ts
     * const queue = await client.texml.accounts.queues.create(
     *   'account_sid',
     * );
     * ```
     */
    create(accountSid, body, options) {
        return this._client.post(path `/texml/Accounts/${accountSid}/Queues`, {
            body,
            ...options,
            headers: buildHeaders([{ 'Content-Type': 'application/x-www-form-urlencoded' }, options?.headers]),
        });
    }
    /**
     * Returns a queue resource.
     *
     * @example
     * ```ts
     * const queue = await client.texml.accounts.queues.retrieve(
     *   'queue_sid',
     *   { account_sid: 'account_sid' },
     * );
     * ```
     */
    retrieve(queueSid, params, options) {
        const { account_sid } = params;
        return this._client.get(path `/texml/Accounts/${account_sid}/Queues/${queueSid}`, options);
    }
    /**
     * Updates a queue resource.
     *
     * @example
     * ```ts
     * const queue = await client.texml.accounts.queues.update(
     *   'queue_sid',
     *   { account_sid: 'account_sid' },
     * );
     * ```
     */
    update(queueSid, params, options) {
        const { account_sid, ...body } = params;
        return this._client.post(path `/texml/Accounts/${account_sid}/Queues/${queueSid}`, {
            body,
            ...options,
            headers: buildHeaders([{ 'Content-Type': 'application/x-www-form-urlencoded' }, options?.headers]),
        });
    }
    /**
     * Lists queue resources.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const queueListResponse of client.texml.accounts.queues.list(
     *   'account_sid',
     * )) {
     *   // ...
     * }
     * ```
     */
    list(accountSid, query = {}, options) {
        return this._client.getAPIList(path `/texml/Accounts/${accountSid}/Queues`, (DefaultPaginationForQueues), { query, ...options });
    }
    /**
     * Delete a queue resource.
     *
     * @example
     * ```ts
     * await client.texml.accounts.queues.delete('queue_sid', {
     *   account_sid: 'account_sid',
     * });
     * ```
     */
    delete(queueSid, params, options) {
        const { account_sid } = params;
        return this._client.delete(path `/texml/Accounts/${account_sid}/Queues/${queueSid}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
