// telnyx/resources/texml/accounts/queues.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Queues = void 0;
const resource_1 = require("../../../core/resource.js");
const pagination_1 = require("../../../core/pagination.js");
const headers_1 = require("../../../internal/headers.js");
const path_1 = require("../../../internal/utils/path.js");
/**
 * TeXML REST Commands
 */
class Queues extends resource_1.APIResource {
    /**
     * Creates a new queue resource.
     *
     * @example
     * ```ts
     * const queue = await client.texml.accounts.queues.create(
     *   'account_sid',
     * );
     * ```
     */
    create(accountSid, body, options) {
        return this._client.post((0, path_1.path) `/texml/Accounts/${accountSid}/Queues`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ 'Content-Type': 'application/x-www-form-urlencoded' }, options?.headers]),
        });
    }
    /**
     * Returns a queue resource.
     *
     * @example
     * ```ts
     * const queue = await client.texml.accounts.queues.retrieve(
     *   'queue_sid',
     *   { account_sid: 'account_sid' },
     * );
     * ```
     */
    retrieve(queueSid, params, options) {
        const { account_sid } = params;
        return this._client.get((0, path_1.path) `/texml/Accounts/${account_sid}/Queues/${queueSid}`, options);
    }
    /**
     * Updates a queue resource.
     *
     * @example
     * ```ts
     * const queue = await client.texml.accounts.queues.update(
     *   'queue_sid',
     *   { account_sid: 'account_sid' },
     * );
     * ```
     */
    update(queueSid, params, options) {
        const { account_sid, ...body } = params;
        return this._client.post((0, path_1.path) `/texml/Accounts/${account_sid}/Queues/${queueSid}`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ 'Content-Type': 'application/x-www-form-urlencoded' }, options?.headers]),
        });
    }
    /**
     * Lists queue resources.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const queueListResponse of client.texml.accounts.queues.list(
     *   'account_sid',
     * )) {
     *   // ...
     * }
     * ```
     */
    list(accountSid, query = {}, options) {
        return this._client.getAPIList((0, path_1.path) `/texml/Accounts/${accountSid}/Queues`, (pagination_1.DefaultPaginationForQueues), { query, ...options });
    }
    /**
     * Delete a queue resource.
     *
     * @example
     * ```ts
     * await client.texml.accounts.queues.delete('queue_sid', {
     *   account_sid: 'account_sid',
     * });
     * ```
     */
    delete(queueSid, params, options) {
        const { account_sid } = params;
        return this._client.delete((0, path_1.path) `/texml/Accounts/${account_sid}/Queues/${queueSid}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.Queues = Queues;
