// telnyx/resources/ai/ai.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.AI = void 0;
const tslib_1 = require("../../internal/tslib.js");
const resource_1 = require("../../core/resource.js");
const AudioAPI = tslib_1.__importStar(require("./audio.js"));
const audio_1 = require("./audio.js");
const ChatAPI = tslib_1.__importStar(require("./chat.js"));
const chat_1 = require("./chat.js");
const ClustersAPI = tslib_1.__importStar(require("./clusters.js"));
const clusters_1 = require("./clusters.js");
const McpServersAPI = tslib_1.__importStar(require("./mcp-servers.js"));
const mcp_servers_1 = require("./mcp-servers.js");
const ToolsAPI = tslib_1.__importStar(require("./tools.js"));
const tools_1 = require("./tools.js");
const AssistantsAPI = tslib_1.__importStar(require("./assistants/assistants.js"));
const assistants_1 = require("./assistants/assistants.js");
const ConversationsAPI = tslib_1.__importStar(require("./conversations/conversations.js"));
const conversations_1 = require("./conversations/conversations.js");
const EmbeddingsAPI = tslib_1.__importStar(require("./embeddings/embeddings.js"));
const embeddings_1 = require("./embeddings/embeddings.js");
const FineTuningAPI = tslib_1.__importStar(require("./fine-tuning/fine-tuning.js"));
const fine_tuning_1 = require("./fine-tuning/fine-tuning.js");
const IntegrationsAPI = tslib_1.__importStar(require("./integrations/integrations.js"));
const integrations_1 = require("./integrations/integrations.js");
const MissionsAPI = tslib_1.__importStar(require("./missions/missions.js"));
const missions_1 = require("./missions/missions.js");
const OpenAIAPI = tslib_1.__importStar(require("./openai/openai.js"));
const openai_1 = require("./openai/openai.js");
/**
 * Generate text with LLMs
 */
class AI extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.assistants = new AssistantsAPI.Assistants(this._client);
        this.audio = new AudioAPI.Audio(this._client);
        this.chat = new ChatAPI.Chat(this._client);
        this.clusters = new ClustersAPI.Clusters(this._client);
        this.conversations = new ConversationsAPI.Conversations(this._client);
        this.embeddings = new EmbeddingsAPI.Embeddings(this._client);
        this.fineTuning = new FineTuningAPI.FineTuning(this._client);
        this.integrations = new IntegrationsAPI.Integrations(this._client);
        this.mcpServers = new McpServersAPI.McpServers(this._client);
        this.missions = new MissionsAPI.Missions(this._client);
        this.openai = new OpenAIAPI.OpenAI(this._client);
        this.tools = new ToolsAPI.Tools(this._client);
    }
    /**
     * **Deprecated**: Use `POST /v2/ai/openai/responses` instead. This endpoint is
     * compatible with the
     * [OpenAI Responses API](https://developers.openai.com/api/reference/responses/overview)
     * and may be used with the OpenAI JS or Python SDK. Response id parameter is not
     * supported at the moment. Use the `conversation` parameter with a Telnyx
     * Conversation ID to leverage persistent conversations.
     *
     * @deprecated
     */
    createResponse(params, options) {
        const { input } = params;
        return this._client.post('/ai/responses', { body: input, ...options });
    }
    /**
     * **Deprecated**: Use `GET /v2/ai/openai/models` instead.
     *
     * Returns the same `ModelsResponse` payload as the OpenAI-compatible endpoint —
     * open-source LLMs hosted on Telnyx (e.g. `moonshotai/Kimi-K2.6`,
     * `zai-org/GLM-5.1-FP8`, `MiniMaxAI/MiniMax-M2.7`), embedding models, and
     * fine-tuned models — kept around for backwards compatibility. New integrations
     * should use `/v2/ai/openai/models`.
     *
     * Model ids follow the `{organization}/{model_name}` convention from Hugging Face.
     *
     * @deprecated
     */
    retrieveModels(options) {
        return this._client.get('/ai/models', options);
    }
    /**
     * Generate a summary of a file's contents.
     *
     * Supports the following text formats:
     *
     * - PDF, HTML, txt, json, csv
     *
     * Supports the following media formats (billed for both the transcription and
     * summary):
     *
     * - flac, mp3, mp4, mpeg, mpga, m4a, ogg, wav, or webm
     * - Up to 100 MB
     *
     * @example
     * ```ts
     * const response = await client.ai.summarize({
     *   bucket: 'bucket',
     *   filename: 'filename',
     * });
     * ```
     */
    summarize(body, options) {
        return this._client.post('/ai/summarize', { body, ...options });
    }
}
exports.AI = AI;
AI.Assistants = assistants_1.Assistants;
AI.Audio = audio_1.Audio;
AI.Chat = chat_1.Chat;
AI.Clusters = clusters_1.Clusters;
AI.Conversations = conversations_1.Conversations;
AI.Embeddings = embeddings_1.Embeddings;
AI.FineTuning = fine_tuning_1.FineTuning;
AI.Integrations = integrations_1.Integrations;
AI.McpServers = mcp_servers_1.McpServers;
AI.Missions = missions_1.Missions;
AI.OpenAI = openai_1.OpenAI;
AI.Tools = tools_1.Tools;
