// telnyx/resources/ai/ai.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as AudioAPI from "./audio.mjs";
import { Audio } from "./audio.mjs";
import * as ChatAPI from "./chat.mjs";
import { Chat } from "./chat.mjs";
import * as ClustersAPI from "./clusters.mjs";
import { Clusters, } from "./clusters.mjs";
import * as McpServersAPI from "./mcp-servers.mjs";
import { McpServers, } from "./mcp-servers.mjs";
import * as ToolsAPI from "./tools.mjs";
import { Tools, } from "./tools.mjs";
import * as AssistantsAPI from "./assistants/assistants.mjs";
import { Assistants, } from "./assistants/assistants.mjs";
import * as ConversationsAPI from "./conversations/conversations.mjs";
import { Conversations, } from "./conversations/conversations.mjs";
import * as EmbeddingsAPI from "./embeddings/embeddings.mjs";
import { Embeddings, } from "./embeddings/embeddings.mjs";
import * as FineTuningAPI from "./fine-tuning/fine-tuning.mjs";
import { FineTuning } from "./fine-tuning/fine-tuning.mjs";
import * as IntegrationsAPI from "./integrations/integrations.mjs";
import { Integrations, } from "./integrations/integrations.mjs";
import * as MissionsAPI from "./missions/missions.mjs";
import { Missions, } from "./missions/missions.mjs";
import * as OpenAIAPI from "./openai/openai.mjs";
import { OpenAI, } from "./openai/openai.mjs";
/**
 * Generate text with LLMs
 */
export class AI extends APIResource {
    constructor() {
        super(...arguments);
        this.assistants = new AssistantsAPI.Assistants(this._client);
        this.audio = new AudioAPI.Audio(this._client);
        this.chat = new ChatAPI.Chat(this._client);
        this.clusters = new ClustersAPI.Clusters(this._client);
        this.conversations = new ConversationsAPI.Conversations(this._client);
        this.embeddings = new EmbeddingsAPI.Embeddings(this._client);
        this.fineTuning = new FineTuningAPI.FineTuning(this._client);
        this.integrations = new IntegrationsAPI.Integrations(this._client);
        this.mcpServers = new McpServersAPI.McpServers(this._client);
        this.missions = new MissionsAPI.Missions(this._client);
        this.openai = new OpenAIAPI.OpenAI(this._client);
        this.tools = new ToolsAPI.Tools(this._client);
    }
    /**
     * **Deprecated**: Use `POST /v2/ai/openai/responses` instead. This endpoint is
     * compatible with the
     * [OpenAI Responses API](https://developers.openai.com/api/reference/responses/overview)
     * and may be used with the OpenAI JS or Python SDK. Response id parameter is not
     * supported at the moment. Use the `conversation` parameter with a Telnyx
     * Conversation ID to leverage persistent conversations.
     *
     * @deprecated
     */
    createResponse(params, options) {
        const { input } = params;
        return this._client.post('/ai/responses', { body: input, ...options });
    }
    /**
     * **Deprecated**: Use `GET /v2/ai/openai/models` instead.
     *
     * Returns the same `ModelsResponse` payload as the OpenAI-compatible endpoint —
     * open-source LLMs hosted on Telnyx (e.g. `moonshotai/Kimi-K2.6`,
     * `zai-org/GLM-5.1-FP8`, `MiniMaxAI/MiniMax-M2.7`), embedding models, and
     * fine-tuned models — kept around for backwards compatibility. New integrations
     * should use `/v2/ai/openai/models`.
     *
     * Model ids follow the `{organization}/{model_name}` convention from Hugging Face.
     *
     * @deprecated
     */
    retrieveModels(options) {
        return this._client.get('/ai/models', options);
    }
    /**
     * Generate a summary of a file's contents.
     *
     * Supports the following text formats:
     *
     * - PDF, HTML, txt, json, csv
     *
     * Supports the following media formats (billed for both the transcription and
     * summary):
     *
     * - flac, mp3, mp4, mpeg, mpga, m4a, ogg, wav, or webm
     * - Up to 100 MB
     *
     * @example
     * ```ts
     * const response = await client.ai.summarize({
     *   bucket: 'bucket',
     *   filename: 'filename',
     * });
     * ```
     */
    summarize(body, options) {
        return this._client.post('/ai/summarize', { body, ...options });
    }
}
AI.Assistants = Assistants;
AI.Audio = Audio;
AI.Chat = Chat;
AI.Clusters = Clusters;
AI.Conversations = Conversations;
AI.Embeddings = Embeddings;
AI.FineTuning = FineTuning;
AI.Integrations = Integrations;
AI.McpServers = McpServers;
AI.Missions = Missions;
AI.OpenAI = OpenAI;
AI.Tools = Tools;
