// telnyx/resources/ai/clusters.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Clusters = void 0;
const resource_1 = require("../../core/resource.js");
const pagination_1 = require("../../core/pagination.js");
const headers_1 = require("../../internal/headers.js");
const path_1 = require("../../internal/utils/path.js");
/**
 * Identify common themes and patterns in your embedded documents
 */
class Clusters extends resource_1.APIResource {
    /**
     * Fetch a cluster
     *
     * @example
     * ```ts
     * const cluster = await client.ai.clusters.retrieve(
     *   'task_id',
     * );
     * ```
     */
    retrieve(taskID, query = {}, options) {
        return this._client.get((0, path_1.path) `/ai/clusters/${taskID}`, { query, ...options });
    }
    /**
     * List all clusters
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const clusterListResponse of client.ai.clusters.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/ai/clusters', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete a cluster
     *
     * @example
     * ```ts
     * await client.ai.clusters.delete('task_id');
     * ```
     */
    delete(taskID, options) {
        return this._client.delete((0, path_1.path) `/ai/clusters/${taskID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Starts a background task to compute how the data in an
     * [embedded storage bucket](https://developers.telnyx.com/api-reference/embeddings/embed-documents)
     * is clustered. This helps identify common themes and patterns in the data.
     *
     * @example
     * ```ts
     * const response = await client.ai.clusters.compute({
     *   bucket: 'bucket',
     * });
     * ```
     */
    compute(body, options) {
        return this._client.post('/ai/clusters', { body, ...options });
    }
    /**
     * Fetch a cluster visualization
     *
     * @example
     * ```ts
     * const response = await client.ai.clusters.fetchGraph(
     *   'task_id',
     * );
     *
     * const content = await response.blob();
     * console.log(content);
     * ```
     */
    fetchGraph(taskID, query = {}, options) {
        return this._client.get((0, path_1.path) `/ai/clusters/${taskID}/graph`, {
            query,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: 'image/png' }, options?.headers]),
            __binaryResponse: true,
        });
    }
}
exports.Clusters = Clusters;
