// telnyx/resources/ai/mcp-servers.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import { DefaultFlatPaginationTopLevelArray, } from "../../core/pagination.mjs";
import { buildHeaders } from "../../internal/headers.mjs";
import { path } from "../../internal/utils/path.mjs";
export class McpServers extends APIResource {
    /**
     * Create a new MCP server.
     *
     * @example
     * ```ts
     * const mcpServer = await client.ai.mcpServers.create({
     *   name: 'name',
     *   type: 'type',
     *   url: 'url',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/ai/mcp_servers', { body, ...options });
    }
    /**
     * Retrieve details for a specific MCP server.
     *
     * @example
     * ```ts
     * const mcpServer = await client.ai.mcpServers.retrieve(
     *   'mcp_server_id',
     * );
     * ```
     */
    retrieve(mcpServerID, options) {
        return this._client.get(path `/ai/mcp_servers/${mcpServerID}`, options);
    }
    /**
     * Update an existing MCP server.
     *
     * @example
     * ```ts
     * const mcpServer = await client.ai.mcpServers.update(
     *   'mcp_server_id',
     * );
     * ```
     */
    update(mcpServerID, body, options) {
        return this._client.put(path `/ai/mcp_servers/${mcpServerID}`, { body, ...options });
    }
    /**
     * Retrieve a list of MCP servers.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const mcpServerListResponse of client.ai.mcpServers.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/ai/mcp_servers', (DefaultFlatPaginationTopLevelArray), { query, ...options });
    }
    /**
     * Delete a specific MCP server.
     *
     * @example
     * ```ts
     * await client.ai.mcpServers.delete('mcp_server_id');
     * ```
     */
    delete(mcpServerID, options) {
        return this._client.delete(path `/ai/mcp_servers/${mcpServerID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
