// telnyx/resources/ai/mcp-servers.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.McpServers = void 0;
const resource_1 = require("../../core/resource.js");
const pagination_1 = require("../../core/pagination.js");
const headers_1 = require("../../internal/headers.js");
const path_1 = require("../../internal/utils/path.js");
class McpServers extends resource_1.APIResource {
    /**
     * Create a new MCP server.
     *
     * @example
     * ```ts
     * const mcpServer = await client.ai.mcpServers.create({
     *   name: 'name',
     *   type: 'type',
     *   url: 'url',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/ai/mcp_servers', { body, ...options });
    }
    /**
     * Retrieve details for a specific MCP server.
     *
     * @example
     * ```ts
     * const mcpServer = await client.ai.mcpServers.retrieve(
     *   'mcp_server_id',
     * );
     * ```
     */
    retrieve(mcpServerID, options) {
        return this._client.get((0, path_1.path) `/ai/mcp_servers/${mcpServerID}`, options);
    }
    /**
     * Update an existing MCP server.
     *
     * @example
     * ```ts
     * const mcpServer = await client.ai.mcpServers.update(
     *   'mcp_server_id',
     * );
     * ```
     */
    update(mcpServerID, body, options) {
        return this._client.put((0, path_1.path) `/ai/mcp_servers/${mcpServerID}`, { body, ...options });
    }
    /**
     * Retrieve a list of MCP servers.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const mcpServerListResponse of client.ai.mcpServers.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/ai/mcp_servers', (pagination_1.DefaultFlatPaginationTopLevelArray), { query, ...options });
    }
    /**
     * Delete a specific MCP server.
     *
     * @example
     * ```ts
     * await client.ai.mcpServers.delete('mcp_server_id');
     * ```
     */
    delete(mcpServerID, options) {
        return this._client.delete((0, path_1.path) `/ai/mcp_servers/${mcpServerID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.McpServers = McpServers;
