// telnyx/resources/ai/tools.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tools = void 0;
const resource_1 = require("../../core/resource.js");
const pagination_1 = require("../../core/pagination.js");
const path_1 = require("../../internal/utils/path.js");
/**
 * Configure AI assistant specifications
 */
class Tools extends resource_1.APIResource {
    /**
     * Create Tool
     *
     * @example
     * ```ts
     * const tool = await client.ai.tools.create({
     *   display_name: 'display_name',
     *   type: 'type',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/ai/tools', { body, ...options });
    }
    /**
     * Get Tool
     *
     * @example
     * ```ts
     * const tool = await client.ai.tools.retrieve('tool_id');
     * ```
     */
    retrieve(toolID, options) {
        return this._client.get((0, path_1.path) `/ai/tools/${toolID}`, options);
    }
    /**
     * Update Tool
     *
     * @example
     * ```ts
     * const tool = await client.ai.tools.update('tool_id');
     * ```
     */
    update(toolID, body, options) {
        return this._client.patch((0, path_1.path) `/ai/tools/${toolID}`, { body, ...options });
    }
    /**
     * List Tools
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const toolListResponse of client.ai.tools.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/ai/tools', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete Tool
     *
     * @example
     * ```ts
     * const tool = await client.ai.tools.delete('tool_id');
     * ```
     */
    delete(toolID, options) {
        return this._client.delete((0, path_1.path) `/ai/tools/${toolID}`, options);
    }
}
exports.Tools = Tools;
