// telnyx/resources/ai/tools.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import { DefaultFlatPagination } from "../../core/pagination.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * Configure AI assistant specifications
 */
export class Tools extends APIResource {
    /**
     * Create Tool
     *
     * @example
     * ```ts
     * const tool = await client.ai.tools.create({
     *   display_name: 'display_name',
     *   type: 'type',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/ai/tools', { body, ...options });
    }
    /**
     * Get Tool
     *
     * @example
     * ```ts
     * const tool = await client.ai.tools.retrieve('tool_id');
     * ```
     */
    retrieve(toolID, options) {
        return this._client.get(path `/ai/tools/${toolID}`, options);
    }
    /**
     * Update Tool
     *
     * @example
     * ```ts
     * const tool = await client.ai.tools.update('tool_id');
     * ```
     */
    update(toolID, body, options) {
        return this._client.patch(path `/ai/tools/${toolID}`, { body, ...options });
    }
    /**
     * List Tools
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const toolListResponse of client.ai.tools.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/ai/tools', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete Tool
     *
     * @example
     * ```ts
     * const tool = await client.ai.tools.delete('tool_id');
     * ```
     */
    delete(toolID, options) {
        return this._client.delete(path `/ai/tools/${toolID}`, options);
    }
}
