// telnyx/resources/text-to-speech/text-to-speech.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.TextToSpeech = void 0;
const resource_1 = require("../../core/resource.js");
const headers_1 = require("../../internal/headers.js");
/**
 * Text to speech streaming command operations
 */
class TextToSpeech extends resource_1.APIResource {
    /**
     * Generate synthesized speech audio from text input. Returns audio in the
     * requested format (binary audio stream, base64-encoded JSON, or an audio URL for
     * later retrieval).
     *
     * Authentication is provided via the standard `Authorization: Bearer <API_KEY>`
     * header.
     *
     * The `voice` parameter provides a convenient shorthand to specify provider,
     * model, and voice in a single string (e.g. `telnyx.NaturalHD.Alloy` or
     * `Telnyx.Ultra.<voice_id>`). Alternatively, specify `provider` explicitly along
     * with provider-specific parameters.
     *
     * Supported providers: `aws`, `telnyx`, `azure`, `elevenlabs`, `minimax`, `rime`,
     * `resemble`, `xai`.
     *
     * The Telnyx `Ultra` model supports 44 languages with emotion control, speed
     * adjustment, and volume control. Use the `telnyx` provider-specific parameters to
     * configure these features.
     */
    createSpeech(body, options) {
        return this._client.post('/text-to-speech/speech', { body, ...options });
    }
    /**
     * Open a WebSocket connection to stream text and receive synthesized audio in real
     * time. Authentication is provided via the standard
     * `Authorization: Bearer <API_KEY>` header. Send JSON frames with text to
     * synthesize; receive JSON frames containing base64-encoded audio chunks.
     *
     * Supported providers: `aws`, `telnyx`, `azure`, `murfai`, `minimax`, `rime`,
     * `resemble`, `elevenlabs`, `xai`.
     *
     * **Connection flow:**
     *
     * 1. Open WebSocket with query parameters specifying provider, voice, and model.
     * 2. Send an initial handshake message `{"text": " "}` (single space) with
     *    optional `voice_settings` to initialize the session.
     * 3. Send text messages as `{"text": "Hello world"}`.
     * 4. Receive audio chunks as JSON frames with base64-encoded audio.
     * 5. A final frame with `isFinal: true` indicates the end of audio for the current
     *    text.
     *
     * To interrupt and restart synthesis mid-stream, send `{"force": true}` — the
     * current worker is stopped and a new one is started.
     *
     * **Note:** The Telnyx `Ultra` model is not available over WebSocket. Use the HTTP
     * POST `/text-to-speech/speech` endpoint instead.
     */
    generateSpeech(query = {}, options) {
        return this._client.get('/text-to-speech/speech', {
            query,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Retrieve a list of available voices from one or all TTS providers. When
     * `provider` is specified, returns voices for that provider only. Otherwise,
     * returns voices from all providers.
     *
     * Some providers (ElevenLabs, Resemble) require an API key to list voices.
     */
    listVoices(query = {}, options) {
        return this._client.get('/text-to-speech/voices', { query, ...options });
    }
    /**
     * Open a WebSocket connection to stream text and receive synthesized audio in real
     * time. Authentication is provided via the standard
     * `Authorization: Bearer <API_KEY>` header. Send JSON frames with text to
     * synthesize; receive JSON frames containing base64-encoded audio chunks.
     */
    stream(query = {}, options) {
        return this._client.get('/text-to-speech/speech', {
            query,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.TextToSpeech = TextToSpeech;
