// telnyx/resources/verified-numbers/verified-numbers.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.VerifiedNumbers = void 0;
const tslib_1 = require("../../internal/tslib.js");
const resource_1 = require("../../core/resource.js");
const ActionsAPI = tslib_1.__importStar(require("./actions.js"));
const actions_1 = require("./actions.js");
const pagination_1 = require("../../core/pagination.js");
const path_1 = require("../../internal/utils/path.js");
/**
 * Verified Numbers operations
 */
class VerifiedNumbers extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.actions = new ActionsAPI.Actions(this._client);
    }
    /**
     * Initiates phone number verification procedure. Supports DTMF extension dialing
     * for voice calls to numbers behind IVR systems.
     *
     * @example
     * ```ts
     * const verifiedNumber = await client.verifiedNumbers.create({
     *   phone_number: '+15551234567',
     *   verification_method: 'sms',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/verified_numbers', { body, ...options });
    }
    /**
     * Retrieve a verified number
     *
     * @example
     * ```ts
     * const verifiedNumberDataWrapper =
     *   await client.verifiedNumbers.retrieve('+15551234567');
     * ```
     */
    retrieve(phoneNumber, options) {
        return this._client.get((0, path_1.path) `/verified_numbers/${phoneNumber}`, options);
    }
    /**
     * Gets a paginated list of Verified Numbers.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const verifiedNumber of client.verifiedNumbers.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/verified_numbers', (pagination_1.DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete a verified number
     *
     * @example
     * ```ts
     * const verifiedNumberDataWrapper =
     *   await client.verifiedNumbers.delete('+15551234567');
     * ```
     */
    delete(phoneNumber, options) {
        return this._client.delete((0, path_1.path) `/verified_numbers/${phoneNumber}`, options);
    }
}
exports.VerifiedNumbers = VerifiedNumbers;
VerifiedNumbers.Actions = actions_1.Actions;
