// telnyx/resources/verified-numbers/verified-numbers.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as ActionsAPI from "./actions.mjs";
import { Actions } from "./actions.mjs";
import { DefaultFlatPagination } from "../../core/pagination.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * Verified Numbers operations
 */
export class VerifiedNumbers extends APIResource {
    constructor() {
        super(...arguments);
        this.actions = new ActionsAPI.Actions(this._client);
    }
    /**
     * Initiates phone number verification procedure. Supports DTMF extension dialing
     * for voice calls to numbers behind IVR systems.
     *
     * @example
     * ```ts
     * const verifiedNumber = await client.verifiedNumbers.create({
     *   phone_number: '+15551234567',
     *   verification_method: 'sms',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/verified_numbers', { body, ...options });
    }
    /**
     * Retrieve a verified number
     *
     * @example
     * ```ts
     * const verifiedNumberDataWrapper =
     *   await client.verifiedNumbers.retrieve('+15551234567');
     * ```
     */
    retrieve(phoneNumber, options) {
        return this._client.get(path `/verified_numbers/${phoneNumber}`, options);
    }
    /**
     * Gets a paginated list of Verified Numbers.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const verifiedNumber of client.verifiedNumbers.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/verified_numbers', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Delete a verified number
     *
     * @example
     * ```ts
     * const verifiedNumberDataWrapper =
     *   await client.verifiedNumbers.delete('+15551234567');
     * ```
     */
    delete(phoneNumber, options) {
        return this._client.delete(path `/verified_numbers/${phoneNumber}`, options);
    }
}
VerifiedNumbers.Actions = Actions;
