// telnyx/resources/managed-accounts/managed-accounts.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as ActionsAPI from "./actions.mjs";
import { Actions } from "./actions.mjs";
import { DefaultFlatPagination } from "../../core/pagination.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * Managed Accounts operations
 */
export class ManagedAccounts extends APIResource {
    constructor() {
        super(...arguments);
        this.actions = new ActionsAPI.Actions(this._client);
    }
    /**
     * Create a new managed account owned by the authenticated user. You need to be
     * explictly approved by Telnyx in order to become a manager account.
     *
     * @example
     * ```ts
     * const managedAccount = await client.managedAccounts.create({
     *   business_name: "Larry's Cat Food Inc",
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/managed_accounts', { body, ...options });
    }
    /**
     * Retrieves the details of a single managed account.
     *
     * @example
     * ```ts
     * const managedAccount =
     *   await client.managedAccounts.retrieve('id');
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/managed_accounts/${id}`, options);
    }
    /**
     * Update a single managed account.
     *
     * @example
     * ```ts
     * const managedAccount = await client.managedAccounts.update(
     *   'id',
     * );
     * ```
     */
    update(id, body, options) {
        return this._client.patch(path `/managed_accounts/${id}`, { body, ...options });
    }
    /**
     * Lists the accounts managed by the current user. Users need to be explictly
     * approved by Telnyx in order to become manager accounts.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const managedAccountListResponse of client.managedAccounts.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/managed_accounts', (DefaultFlatPagination), {
            query,
            ...options,
        });
    }
    /**
     * Display information about allocatable global outbound channels for the current
     * user. Only usable by account managers.
     *
     * @example
     * ```ts
     * const response =
     *   await client.managedAccounts.getAllocatableGlobalOutboundChannels();
     * ```
     */
    getAllocatableGlobalOutboundChannels(options) {
        return this._client.get('/managed_accounts/allocatable_global_outbound_channels', options);
    }
    /**
     * Update the amount of allocatable global outbound channels allocated to a
     * specific managed account.
     *
     * @example
     * ```ts
     * const response =
     *   await client.managedAccounts.updateGlobalChannelLimit(
     *     'id',
     *   );
     * ```
     */
    updateGlobalChannelLimit(id, body, options) {
        return this._client.patch(path `/managed_accounts/${id}/update_global_channel_limit`, {
            body,
            ...options,
        });
    }
}
ManagedAccounts.Actions = Actions;
