// telnyx/resources/whatsapp/phone-numbers/phone-numbers.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.PhoneNumbers = void 0;
const tslib_1 = require("../../../internal/tslib.js");
const resource_1 = require("../../../core/resource.js");
const CallingSettingsAPI = tslib_1.__importStar(require("./calling-settings.js"));
const calling_settings_1 = require("./calling-settings.js");
const ProfileAPI = tslib_1.__importStar(require("./profile/profile.js"));
const profile_1 = require("./profile/profile.js");
const pagination_1 = require("../../../core/pagination.js");
const headers_1 = require("../../../internal/headers.js");
const path_1 = require("../../../internal/utils/path.js");
/**
 * Manage Whatsapp phone numbers
 */
class PhoneNumbers extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.callingSettings = new CallingSettingsAPI.CallingSettings(this._client);
        this.profile = new ProfileAPI.Profile(this._client);
    }
    /**
     * List Whatsapp phone numbers
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const phoneNumberListResponse of client.whatsapp.phoneNumbers.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/v2/whatsapp/phone_numbers', (pagination_1.DefaultFlatPagination), { query, ...options });
    }
    /**
     * Delete a Whatsapp phone number
     *
     * @example
     * ```ts
     * await client.whatsapp.phoneNumbers.delete('phone_number');
     * ```
     */
    delete(phoneNumber, options) {
        return this._client.delete((0, path_1.path) `/v2/whatsapp/phone_numbers/${phoneNumber}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Resend verification code
     *
     * @example
     * ```ts
     * await client.whatsapp.phoneNumbers.resendVerification(
     *   'phone_number',
     * );
     * ```
     */
    resendVerification(phoneNumber, body, options) {
        return this._client.post((0, path_1.path) `/v2/whatsapp/phone_numbers/${phoneNumber}/resend_verification`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Submit verification code for a phone number
     *
     * @example
     * ```ts
     * await client.whatsapp.phoneNumbers.verify('phone_number', {
     *   code: 'code',
     * });
     * ```
     */
    verify(phoneNumber, body, options) {
        return this._client.post((0, path_1.path) `/v2/whatsapp/phone_numbers/${phoneNumber}/verify`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.PhoneNumbers = PhoneNumbers;
PhoneNumbers.CallingSettings = calling_settings_1.CallingSettings;
PhoneNumbers.Profile = profile_1.Profile;
