// telnyx/resources/whatsapp/phone-numbers/phone-numbers.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../core/resource.mjs";
import * as CallingSettingsAPI from "./calling-settings.mjs";
import { CallingSettings, } from "./calling-settings.mjs";
import * as ProfileAPI from "./profile/profile.mjs";
import { Profile, } from "./profile/profile.mjs";
import { DefaultFlatPagination, } from "../../../core/pagination.mjs";
import { buildHeaders } from "../../../internal/headers.mjs";
import { path } from "../../../internal/utils/path.mjs";
/**
 * Manage Whatsapp phone numbers
 */
export class PhoneNumbers extends APIResource {
    constructor() {
        super(...arguments);
        this.callingSettings = new CallingSettingsAPI.CallingSettings(this._client);
        this.profile = new ProfileAPI.Profile(this._client);
    }
    /**
     * List Whatsapp phone numbers
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const phoneNumberListResponse of client.whatsapp.phoneNumbers.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/v2/whatsapp/phone_numbers', (DefaultFlatPagination), { query, ...options });
    }
    /**
     * Delete a Whatsapp phone number
     *
     * @example
     * ```ts
     * await client.whatsapp.phoneNumbers.delete('phone_number');
     * ```
     */
    delete(phoneNumber, options) {
        return this._client.delete(path `/v2/whatsapp/phone_numbers/${phoneNumber}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Resend verification code
     *
     * @example
     * ```ts
     * await client.whatsapp.phoneNumbers.resendVerification(
     *   'phone_number',
     * );
     * ```
     */
    resendVerification(phoneNumber, body, options) {
        return this._client.post(path `/v2/whatsapp/phone_numbers/${phoneNumber}/resend_verification`, {
            body,
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Submit verification code for a phone number
     *
     * @example
     * ```ts
     * await client.whatsapp.phoneNumbers.verify('phone_number', {
     *   code: 'code',
     * });
     * ```
     */
    verify(phoneNumber, body, options) {
        return this._client.post(path `/v2/whatsapp/phone_numbers/${phoneNumber}/verify`, {
            body,
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
PhoneNumbers.CallingSettings = CallingSettings;
PhoneNumbers.Profile = Profile;
